Setting up for use of Condor CHTC with Chr 2 hotspot
================
Frederick Boehm
06/08/2019

## Goals

Goal here is to document the setup & organization of files that I need
for using Condor CHTC to do my two-dimensional scans as part of the
analyses of the four Keller hotspots. Note that they reported 5
hotspots.

## Setting

Recall that I have downloaded a Keller 2018 .RData file from DataDryad
that contains many of their results. The .RData file,
“data/Attie\_DO378\_eQTL\_viewer\_v1.Rdata” annotates 139 traits as
belonging to the chromosome 2 hotspot.

Following Karl’s suggestion, I want to look at not only the local trait
that Keller identifies as causal for the hotspot, but also many other
local-ish traits that have reasonable univariate LOD
scores.

## Preparing RDS files for Chr 2 analysis

``` r
load("~/Box Sync/attie/keller2018-chr2-hotspot-chtc/data-to-ignore/Attie_DO378_eQTL_viewer_v1.Rdata")
ls()
```

    ## [1] "dataset.clinical.phenotypes" "dataset.islet.hotspots"     
    ## [3] "dataset.islet.modules"       "dataset.islet.rnaseq"       
    ## [5] "ensembl.version"             "genoprobs"                  
    ## [7] "K"                           "map"                        
    ## [9] "markers"

``` r
library(tidyverse)
```

    ## ── Attaching packages ──────────────────────────────────────────── tidyverse 1.2.1 ──

    ## ✔ ggplot2 3.1.1       ✔ purrr   0.3.2  
    ## ✔ tibble  2.1.1       ✔ dplyr   0.8.0.1
    ## ✔ tidyr   0.8.3       ✔ stringr 1.4.0  
    ## ✔ readr   1.3.1       ✔ forcats 0.4.0

    ## ── Conflicts ─────────────────────────────────────────────── tidyverse_conflicts() ──
    ## ✖ dplyr::filter() masks stats::filter()
    ## ✖ dplyr::lag()    masks stats::lag()

``` r
dataset.islet.rnaseq$expr %>%
  dim
```

    ## [1]   378 21771

``` r
colnames(dataset.islet.rnaseq$expr) %>%
  head
```

    ## [1] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000037"
    ## [4] "ENSMUSG00000000049" "ENSMUSG00000000056" "ENSMUSG00000000058"

We need to get the names of the nonlocal traits that map to Chromosome
13 hotspot.

``` r
library(qtl2)
```

    ## 
    ## Attaching package: 'qtl2'

    ## The following object is masked from 'package:readr':
    ## 
    ##     read_csv

We want the ensembl ids for the traits that map to Chr 13 hotspot.
First, define the boundaries of the hotspot. The supplemental table in
Keller et al. 2018 lists only the midpoint of the hotspot. I believe
that the half-width is 2 Mb. Yes, this is indeed the half-width per the
Rmd file that Dan Gatti prepared.

``` r
hot_chrom <- 2
hot_center <- 165.5# for chr2
hot_lower <- hot_center - 2
hot_upper <- hot_center + 2
```

Now, filter to get those traits that map to the hotspot. Note that we’re
likely to get local traits, too, in this step.

``` r
dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18)
```

    ##               annot.id   marker.id chrom      pos        lod        symbol
    ## 1   ENSMUSG00000038009 2_164027702     2 164.0277  13.722130       Dnajc22
    ## 2   ENSMUSG00000038225 2_165121823     2 165.1218   8.129404       Ccdc111
    ## 3   ENSMUSG00000038370 2_164348452     2 164.3485   9.110261        Pcp4l1
    ## 4   ENSMUSG00000038515 2_164022416     2 164.0224  23.712161         Grtp1
    ## 5   ENSMUSG00000039238 2_163582672     2 163.5827   7.807164        Zfp750
    ## 6   ENSMUSG00000039536 2_166951357     2 166.9514   9.125270         Stau1
    ## 7   ENSMUSG00000039621 2_166754477     2 166.7545  46.907796         Prex1
    ## 8   ENSMUSG00000039725 2_165574583     2 165.5746  67.344051 2810408M09Rik
    ## 9   ENSMUSG00000039849 2_164648869     2 164.6489  10.555842         Pcif1
    ## 10  ENSMUSG00000039873 2_164636687     2 164.6367  64.581279        Neurl2
    ## 11  ENSMUSG00000040420 2_165445995     2 165.4460   7.971777         Cdh18
    ## 12  ENSMUSG00000006373 2_164022416     2 164.0224  15.773971        Pgrmc1
    ## 13  ENSMUSG00000006418 2_167142758     2 167.1428  10.904746        Rnf114
    ## 14  ENSMUSG00000006462 2_167274124     2 167.2741  22.714740 A530013C23Rik
    ## 15  ENSMUSG00000006800 2_166425938     2 166.4259   8.326041         Sulf2
    ## 16  ENSMUSG00000040964 2_165457194     2 165.4572   7.571674     Arhgef10l
    ## 17  ENSMUSG00000041323 2_164314068     2 164.3141  10.187335           Ak7
    ## 18  ENSMUSG00000041372 2_164022416     2 164.0224  21.377480      B4galnt3
    ## 19  ENSMUSG00000041598 2_164255184     2 164.2552   7.584104      Cdc42ep4
    ## 20  ENSMUSG00000042078 2_163582672     2 163.5827   9.728828          Svop
    ## 21  ENSMUSG00000042096 2_164291807     2 164.2918   7.195302           Dao
    ## 22  ENSMUSG00000042444 2_165581587     2 165.5816   8.418006        Fam63b
    ## 23  ENSMUSG00000042476 2_163582672     2 163.5827  15.496483         Abcb4
    ## 24  ENSMUSG00000042616 2_165276843     2 165.2768   7.720894         Oscp1
    ## 25  ENSMUSG00000042638 2_164022416     2 164.0224  11.991941        Gucy2c
    ## 26  ENSMUSG00000042659 2_164022416     2 164.0224   9.215297        Arrdc4
    ## 27  ENSMUSG00000042854 2_166507114     2 166.5071  31.879763       Trp53rk
    ## 28  ENSMUSG00000044362 2_165283031     2 165.2830   7.403028        Ccdc89
    ## 29  ENSMUSG00000044707 2_163582672     2 163.5827   7.184154         Ccnjl
    ## 30  ENSMUSG00000045281 2_166839967     2 166.8400   8.388572         Gpr20
    ## 31  ENSMUSG00000045284 2_164027702     2 164.0277   9.103386      Dcaf12l1
    ## 32  ENSMUSG00000045414 2_165629348     2 165.6293   8.730262 1190002N15Rik
    ## 33  ENSMUSG00000045503 2_164758936     2 164.7589  12.760189          Sys1
    ## 34  ENSMUSG00000048442 2_164022416     2 164.0224  17.914081         Smim5
    ## 35  ENSMUSG00000048490 2_164022416     2 164.0224   9.850746         Nrip1
    ## 36  ENSMUSG00000048905 2_164255184     2 164.2552   8.325836 4930539E08Rik
    ## 37  ENSMUSG00000049091 2_163582672     2 163.5827  27.579864        Sephs2
    ## 38  ENSMUSG00000050373 2_164758936     2 164.7589  20.817931         Snx21
    ## 39  ENSMUSG00000050556 2_166783106     2 166.7831   9.338683         Kcnb1
    ## 40  ENSMUSG00000050822 2_164255184     2 164.2552   7.964625       Slc29a4
    ## 41  ENSMUSG00000050866 2_164022416     2 164.0224  19.365390         Clrn3
    ## 42  ENSMUSG00000051375 2_163582672     2 163.5827  12.570536         Pcdh1
    ## 43  ENSMUSG00000052353 2_165283031     2 165.2830   7.754589 9930013L23Rik
    ## 44  ENSMUSG00000052392 2_164022416     2 164.0224  12.425552         Acot4
    ## 45  ENSMUSG00000053166 2_165053217     2 165.0532  53.841631         Cdh22
    ## 46  ENSMUSG00000053414 2_164027702     2 164.0277  18.238143          Hunk
    ## 47  ENSMUSG00000054455 2_165445995     2 165.4460   8.870223          Vapb
    ## 48  ENSMUSG00000054582 2_164027702     2 164.0277  52.500903       Pabpc1l
    ## 49  ENSMUSG00000054693 2_165276843     2 165.2768   9.916533        Adam10
    ## 50  ENSMUSG00000057228 2_164027702     2 164.0277   8.213972         Aadat
    ## 51  ENSMUSG00000058812 2_163520139     2 163.5201  26.358758 0610039K10Rik
    ## 52  ENSMUSG00000062480 2_166606199     2 166.6062   7.180954         Acat3
    ## 53  ENSMUSG00000063626 2_164027702     2 164.0277   7.268594         Unc5d
    ## 54  ENSMUSG00000063683 2_164064248     2 164.0642  11.175229         Glyat
    ## 55  ENSMUSG00000066687 2_164022416     2 164.0224   7.303748        Zbtb16
    ## 56  ENSMUSG00000068854 2_165876447     2 165.8764   8.997519     Hist2h2be
    ## 57  ENSMUSG00000069520 2_164027702     2 164.0277  20.062717        Tmem19
    ## 58  ENSMUSG00000070047 2_164022416     2 164.0224   8.151938          Fat1
    ## 59  ENSMUSG00000070530 2_164648869     2 164.6489  38.665597        Wfdc16
    ## 60  ENSMUSG00000074063 2_164022416     2 164.0224  13.865735        Osgin1
    ## 61  ENSMUSG00000074240 2_165479592     2 165.4796   8.843977          Cib3
    ## 62  ENSMUSG00000074582 2_166839967     2 166.8400  28.825971       Arfgef2
    ## 63  ENSMUSG00000075316 2_164022416     2 164.0224  14.179958         Scn9a
    ## 64  ENSMUSG00000076431 2_164064248     2 164.0642  14.358836          Sox4
    ## 65  ENSMUSG00000078796 2_164022416     2 164.0224  25.180158        Zfp541
    ## 66  ENSMUSG00000015766 2_164064248     2 164.0642  12.043851          Eps8
    ## 67  ENSMUSG00000078935 2_164291807     2 164.2918  18.535381 1700025C18Rik
    ## 68  ENSMUSG00000079363 2_164027702     2 164.0277   7.340764          Gbp4
    ## 69  ENSMUSG00000079388 2_164022416     2 164.0224   8.261101 2610042L04Rik
    ## 70  ENSMUSG00000081758 2_165485905     2 165.4859  28.474962       Gm11460
    ## 71  ENSMUSG00000082180 2_166422503     2 166.4225   7.564115       Gm12230
    ## 72  ENSMUSG00000082514 2_163507977     2 163.5080  10.951395       Gm11452
    ## 73  ENSMUSG00000082530 2_166181382     2 166.1814   7.967226       Gm12168
    ## 74  ENSMUSG00000017002 2_164605360     2 164.6054  40.503923          Slpi
    ## 75  ENSMUSG00000017009 2_165053217     2 165.0532   8.916415          Sdc4
    ## 76  ENSMUSG00000017453 2_164076561     2 164.0766  16.603570         Pipox
    ## 77  ENSMUSG00000017493 2_165445995     2 165.4460   9.092559        Igfbp4
    ## 78  ENSMUSG00000017664 2_165283031     2 165.2830  24.411361       Slc35c2
    ## 79  ENSMUSG00000017667 2_165445995     2 165.4460  10.023040        Zfp334
    ## 80  ENSMUSG00000017679 2_163582672     2 163.5827  23.612653         Ttpal
    ## 81  ENSMUSG00000017697 2_163582672     2 163.5827  35.875691           Ada
    ## 82  ENSMUSG00000017707 2_163582672     2 163.5827 126.932688       Serinc3
    ## 83  ENSMUSG00000017720 2_164348452     2 164.3485  31.297934      Trp53tg5
    ## 84  ENSMUSG00000017721 2_164636687     2 164.6367  32.836473          Pigt
    ## 85  ENSMUSG00000017723 2_165195982     2 165.1960   8.032489         Wfdc2
    ## 86  ENSMUSG00000017734 2_164291807     2 164.2918   7.516019        Dbndd2
    ## 87  ENSMUSG00000017740 2_165053217     2 165.0532  38.879355       Slc12a5
    ## 88  ENSMUSG00000017754 2_165053217     2 165.0532  13.314428          Pltp
    ## 89  ENSMUSG00000017756 2_163582672     2 163.5827   8.128036       Slc12a7
    ## 90  ENSMUSG00000017760 2_164758936     2 164.7589  23.371606          Ctsa
    ## 91  ENSMUSG00000083518 2_164022416     2 164.0224   8.803603        Gm9429
    ## 92  ENSMUSG00000084099 2_164617986     2 164.6180  41.257656       Gm12929
    ## 93  ENSMUSG00000084840 2_166324274     2 166.3243  18.897196       Gm11468
    ## 94  ENSMUSG00000085007 2_165053217     2 165.0532  10.327628       Gm11549
    ## 95  ENSMUSG00000085436 2_164758936     2 164.7589  17.277533       Gm11458
    ## 96  ENSMUSG00000085760 2_164605360     2 164.6054  10.702363       Gm14302
    ## 97  ENSMUSG00000086098 2_167274124     2 167.2741  73.724492       Gm14291
    ## 98  ENSMUSG00000086549 2_165283031     2 165.2830   7.377476       Gm13648
    ## 99  ENSMUSG00000017897 2_165722231     2 165.7222  98.894768          Eya2
    ## 100 ENSMUSG00000017950 2_164022416     2 164.0224  48.976242         Hnf4a
    ## 101 ENSMUSG00000017969 2_167282638     2 167.2826  56.248197         Ptgis
    ## 102 ENSMUSG00000018126 2_164022416     2 164.0224  19.375404      Baiap2l2
    ## 103 ENSMUSG00000018209 2_164027702     2 164.0277  60.386442          Stk4
    ## 104 ENSMUSG00000018322 2_164076561     2 164.0766  32.428788        Tomm34
    ## 105 ENSMUSG00000018326 2_164022416     2 164.0224  13.956283         Ywhab
    ## 106 ENSMUSG00000086605 2_166839967     2 166.8400  14.234997       Gm14290
    ## 107 ENSMUSG00000086820 2_165930685     2 165.9307  14.406525       Gm11465
    ## 108 ENSMUSG00000089678 2_164022416     2 164.0224  15.081000         Agxt2
    ## 109 ENSMUSG00000089906 2_165892211     2 165.8922   7.701754        Gm6428
    ## 110 ENSMUSG00000090213 2_167293260     2 167.2933  19.014566       Tmem189
    ## 111 ENSMUSG00000090397 2_166951357     2 166.9514  13.075599       Gm17096
    ## 112 ENSMUSG00000090872 2_164076561     2 164.0766   9.411942        Gm3239
    ## 113 ENSMUSG00000090939 2_164022416     2 164.0224  25.219408       Gm17147
    ## 114 ENSMUSG00000000544 2_163582672     2 163.5827  25.805692         Gpa33
    ## 115 ENSMUSG00000091568 2_163582672     2 163.5827  11.367508        Gm8206
    ## 116 ENSMUSG00000091756 2_164032987     2 164.0330  10.907218        Gm3095
    ## 117 ENSMUSG00000093898 2_164255184     2 164.2552   9.681040        Gm3629
    ## 118 ENSMUSG00000094628 2_164255184     2 164.2552   9.906043        Gm3252
    ## 119 ENSMUSG00000095681 2_164022416     2 164.0224   8.250053        Gm8281
    ## 120 ENSMUSG00000096793 2_164255184     2 164.2552   7.450951        Gm3002
    ## 121 ENSMUSG00000019558 2_164022416     2 164.0224   7.365403        Slc6a8
    ## 122 ENSMUSG00000019762 2_164076561     2 164.0766  23.947946           Iyd
    ## 123 ENSMUSG00000097127 2_164032987     2 164.0330   8.890708       Gm26886
    ## 124 ENSMUSG00000097958 2_163582672     2 163.5827  10.350648        Gm8492
    ## 125 ENSMUSG00000020182 2_164064248     2 164.0642  14.051324           Ddc
    ## 126 ENSMUSG00000020346 2_164636687     2 164.6367   9.260118         Mgat1
    ## 127 ENSMUSG00000020614 2_165154046     2 165.1540   7.499019        Fam20a
    ## 128 ENSMUSG00000020623 2_164027702     2 164.0277  18.906731        Map2k6
    ## 129 ENSMUSG00000020627 2_163582672     2 163.5827  10.431452        Klhl29
    ## 130 ENSMUSG00000020733 2_163582672     2 163.5827   8.163711      Slc9a3r1
    ## 131 ENSMUSG00000020752 2_164022416     2 164.0224  31.635676        Recql5
    ## 132 ENSMUSG00000020821 2_163582672     2 163.5827  11.198410         Kif1c
    ## 133 ENSMUSG00000020841 2_164022416     2 164.0224  12.922466           Cpd
    ## 134 ENSMUSG00000021111 2_164032987     2 164.0330  11.707754        Papola
    ## 135 ENSMUSG00000021263 2_164022416     2 164.0224  14.619333         Degs2
    ## 136 ENSMUSG00000021944 2_163582672     2 163.5827   9.168775         Gata4
    ## 137 ENSMUSG00000022383 2_164255184     2 164.2552   8.408770         Ppara
    ## 138 ENSMUSG00000022824 2_164022416     2 164.0224  17.650449         Muc13
    ## 139 ENSMUSG00000023959 2_164022416     2 164.0224  10.395178         Clic5
    ## 140 ENSMUSG00000024029 2_163582672     2 163.5827  16.164473          Tff3
    ## 141 ENSMUSG00000024331 2_165276843     2 165.2768   7.774202          Dsc2
    ## 142 ENSMUSG00000024388 2_163582672     2 163.5827  13.840924         Myo7b
    ## 143 ENSMUSG00000024527 2_164022416     2 164.0224  18.851973        Afg3l2
    ## 144 ENSMUSG00000024553 2_164255184     2 164.2552   9.340722         Galr1
    ## 145 ENSMUSG00000024777 2_163582672     2 163.5827   7.655403       Ppp2r5b
    ## 146 ENSMUSG00000025207 2_164022416     2 164.0224  22.544879        Sema4g
    ## 147 ENSMUSG00000025422 2_164022416     2 164.0224  33.922450         Agap2
    ## 148 ENSMUSG00000026024 2_164032987     2 164.0330  17.867343          Als2
    ## 149 ENSMUSG00000002032 2_163582672     2 163.5827  10.860464        Tmem25
    ## 150 ENSMUSG00000026175 2_163520139     2 163.5201  26.878025          Vil1
    ## 151 ENSMUSG00000026176 2_164255184     2 164.2552   7.445440        Ctdsp1
    ## 152 ENSMUSG00000026640 2_164022416     2 164.0224   8.951271        Plxna2
    ## 153 ENSMUSG00000027173 2_165445995     2 165.4460   7.514199        Depdc7
    ## 154 ENSMUSG00000027199 2_165154046     2 165.1540   7.509499          Gatm
    ## 155 ENSMUSG00000027296 2_164022416     2 164.0224   8.168765         Itpka
    ## 156 ENSMUSG00000027297 2_163582672     2 163.5827   8.464566           Ltk
    ## 157 ENSMUSG00000027610 2_163582672     2 163.5827  14.398048           Gss
    ## 158 ENSMUSG00000027661 2_165700989     2 165.7010  35.973655       Slc2a10
    ## 159 ENSMUSG00000027678 2_165892211     2 165.8922  13.825360         Ncoa3
    ## 160 ENSMUSG00000027931 2_164027702     2 164.0277   9.657224          Npr1
    ## 161 ENSMUSG00000028307 2_164022416     2 164.0224  21.283565         Aldob
    ## 162 ENSMUSG00000028434 2_164022416     2 164.0224   9.384733     Epb4.1l4b
    ## 163 ENSMUSG00000028649 2_165581587     2 165.5816   9.055917         Macf1
    ## 164 ENSMUSG00000028737 2_163582672     2 163.5827   9.532310       Aldh4a1
    ## 165 ENSMUSG00000029134 2_165283031     2 165.2830   8.830237          Plb1
    ## 166 ENSMUSG00000029189 2_164022416     2 164.0224  17.510514        Sel1l3
    ## 167 ENSMUSG00000029269 2_164022416     2 164.0224  17.262349       Sult1b1
    ## 168 ENSMUSG00000029273 2_164032987     2 164.0330  11.381784       Sult1d1
    ## 169 ENSMUSG00000003352 2_164022416     2 164.0224  46.111014        Cacnb3
    ## 170 ENSMUSG00000003360 2_164022416     2 164.0224   7.633528         Ddx23
    ## 171 ENSMUSG00000003585 2_164022416     2 164.0224  12.023064       Sec14l2
    ## 172 ENSMUSG00000003746 2_164255184     2 164.2552   9.218257         Man1a
    ## 173 ENSMUSG00000029919 2_164022416     2 164.0224  30.083235         Hpgds
    ## 174 ENSMUSG00000030200 2_165711827     2 165.7118   7.743620       Bcl2l14
    ## 175 ENSMUSG00000030313 2_166181382     2 166.1814   9.387317       Dennd5b
    ## 176 ENSMUSG00000030314 2_164758936     2 164.7589   7.295713          Atg7
    ## 177 ENSMUSG00000030838 2_164022416     2 164.0224  19.761300         Ush1c
    ## 178 ENSMUSG00000031161 2_164027702     2 164.0277  10.633061         Hdac6
    ## 179 ENSMUSG00000031668 2_164255184     2 164.2552   8.898565       Eif2ak3
    ## 180 ENSMUSG00000031827 2_163582672     2 163.5827  15.765098         Cotl1
    ## 181 ENSMUSG00000031845 2_164022416     2 164.0224  46.208568         Bcmo1
    ## 182 ENSMUSG00000004317 2_164022416     2 164.0224  10.641129         Clcn5
    ## 183 ENSMUSG00000032091 2_163582672     2 163.5827  17.427857       Tmprss4
    ## 184 ENSMUSG00000032246 2_164022416     2 164.0224  11.644430        Calml4
    ## 185 ENSMUSG00000032358 2_164022416     2 164.0224  16.098787        Fam83b
    ## 186 ENSMUSG00000032359 2_164291807     2 164.2918   9.300967          Ctsh
    ## 187 ENSMUSG00000033577 2_164032987     2 164.0330  10.088948          Myo6
    ## 188 ENSMUSG00000033721 2_164027702     2 164.0277  13.405064          Vav3
    ## 189 ENSMUSG00000034295 2_163582672     2 163.5827  17.314782         Fhod3
    ## 190 ENSMUSG00000034427 2_164022416     2 164.0224  69.885409        Myo15b
    ## 191 ENSMUSG00000034570 2_164022416     2 164.0224  10.445749        Inpp5j
    ## 192 ENSMUSG00000034687 2_164064248     2 164.0642  10.093551         Fras1
    ## 193 ENSMUSG00000004748 2_163520139     2 163.5201  16.997184         Mtfp1
    ## 194 ENSMUSG00000034714 2_164022416     2 164.0224  14.220893         Ttyh2
    ## 195 ENSMUSG00000034918 2_164022416     2 164.0224  30.032670         Cdhr2
    ## 196 ENSMUSG00000035000 2_164022416     2 164.0224  25.057609          Dpp4
    ## 197 ENSMUSG00000035226 2_164255184     2 164.2552   9.432654         Rims4
    ## 198 ENSMUSG00000035268 2_163520139     2 163.5201  51.676300          Pkig
    ## 199 ENSMUSG00000035852 2_163582672     2 163.5827   9.225060          Misp
    ## 200 ENSMUSG00000036098 2_164022416     2 164.0224  36.642606          Myrf
    ## 201 ENSMUSG00000036206 2_164022416     2 164.0224   9.071715        Sh3bp4
    ## 202 ENSMUSG00000036526 2_164064248     2 164.0642   7.206373        Card11
    ## 203 ENSMUSG00000036817 2_164255184     2 164.2552   7.856308          Sun1
    ## 204 ENSMUSG00000037295 2_165445995     2 165.4460   7.297532       Ldlrap1
    ## 205 ENSMUSG00000037400 2_164027702     2 164.0277  11.718834        Atp11b
    ## 206 ENSMUSG00000037428 2_164022416     2 164.0224  14.954312           Vgf
    ## 207 ENSMUSG00000037653 2_164255184     2 164.2552   9.476613         Kctd8
    ##     chr      start        end strand     middle nearest.marker.id
    ## 1    15  99.093170  99.104737      1  99.098954       15_99096703
    ## 2     8  46.575579  46.617212     -1  46.596396        8_46539518
    ## 3     1 171.173262 171.196268     -1 171.184765       1_171184444
    ## 4     8  13.176869  13.200624     -1  13.188746        8_13209235
    ## 5    11 121.510978 121.519333     -1 121.515156      11_121517180
    ## 6     2 166.947549 166.996299     -1 166.971924       2_166992184
    ## 7     2 166.566342 166.713832     -1 166.640087       2_166623650
    ## 8     2 165.490110 165.493324     -1 165.491717       2_165495439
    ## 9     2 164.879304 164.894454      1 164.886879       2_164758936
    ## 10    2 164.830732 164.833456     -1 164.832094       2_164758936
    ## 11   15  22.549022  23.474418      1  23.011720       15_23040344
    ## 12    X  36.598206  36.606079      1  36.602142        X_36601896
    ## 13    2 167.492645 167.516173      1 167.504409       2_167523201
    ## 14    2 167.691177 167.729172      1 167.710174       2_167755373
    ## 15    2 166.073089 166.155663     -1 166.114376       2_166172164
    ## 16    4 140.514485 140.666012     -1 140.590248       4_140590537
    ## 17   12 105.705982 105.782447      1 105.744214      12_105722562
    ## 18    6 120.203073 120.294559     -1 120.248816       6_120177600
    ## 19   11 113.726850 113.751881     -1 113.739366      11_113743477
    ## 20    5 114.026910 114.091570     -1 114.059240       5_114111853
    ## 21    5 114.003703 114.025449      1 114.014576       5_114111853
    ## 22    9  70.599014  70.657174     -1  70.628094        9_70614608
    ## 23    5   8.893721   8.959225      1   8.926473         5_8931349
    ## 24    4 126.058565 126.092195      1 126.075380       4_126074510
    ## 25    6 136.697285 136.781765     -1 136.739525       6_136733632
    ## 26    7  68.737045  68.749241     -1  68.743143        7_68751017
    ## 27    2 166.792531 166.799505      1 166.796018       2_166802407
    ## 28    7  90.426312  90.428660      1  90.427486        7_90450625
    ## 29   11  43.528784  43.586997      1  43.557890       11_43613493
    ## 30   15  73.694604  73.707505     -1  73.701054       15_73659987
    ## 31    X  44.786570  44.790197     -1  44.788384        X_44801022
    ## 32    9  94.517864  94.538081     -1  94.527972        9_94495914
    ## 33    2 164.456964 164.479638      1 164.468301       2_164348452
    ## 34   11 115.899966 115.906269      1 115.903118      11_115923957
    ## 35   16  76.287400  76.373827     -1  76.330614       16_76344250
    ## 36   17  28.896397  28.915324     -1  28.905860       17_29049926
    ## 37    7 127.272561 127.273919     -1 127.273240       7_127361439
    ## 38    2 164.785823 164.793816      1 164.789820       2_164758936
    ## 39    2 167.103378 167.190155     -1 167.146766       2_167142758
    ## 40    5 142.702101 142.722490      1 142.712296       5_142726186
    ## 41    7 135.511466 135.528654     -1 135.520060       7_135538209
    ## 42   18  38.185915  38.212053     -1  38.198984       18_38193631
    ## 43    7  83.932857  84.086502     -1  84.009680        7_83978387
    ## 44   12  84.038379  84.044723      1  84.041551       12_84015575
    ## 45    2 165.111507 165.234853     -1 165.173180       2_165163569
    ## 46   16  90.386397  90.499553      1  90.442975       16_90426989
    ## 47    2 173.737511 173.784336      1 173.760924       2_173786235
    ## 48    2 164.025450 164.050538      1 164.037994       2_164032987
    ## 49    9  70.678997  70.780229      1  70.729613        9_70727937
    ## 50    8  60.506124  60.545677      1  60.525900        8_60526333
    ## 51    2 163.644850 163.645801      1 163.645326       2_163582672
    ## 52   17  12.923833  12.940402     -1  12.932118       17_12911544
    ## 53    8  28.646717  29.219338     -1  28.933028        8_29040371
    ## 54   19  12.633308  12.653911      1  12.643610       19_12788036
    ## 55    9  48.654311  48.835945     -1  48.745128        9_48720706
    ## 56    3  96.221121  96.223738      1  96.222430        3_96110858
    ## 57   10 115.340739 115.362262     -1 115.351500      10_115328485
    ## 58    8  44.950208  45.052257      1  45.001232        8_44986539
    ## 59    2 164.634707 164.638796     -1 164.636752       2_164636687
    ## 60    8 119.434124 119.446256      1 119.440190       8_119439023
    ## 61    8  72.204335  72.212837     -1  72.208586        8_72329705
    ## 62    2 166.805588 166.898052      1 166.851820       2_166839967
    ## 63    2  66.480080  66.634952     -1  66.557516        2_66626773
    ## 64   13  28.948919  28.953713     -1  28.951316       13_28960280
    ## 65    7  16.071942  16.096328      1  16.084135        7_16070850
    ## 66    6 137.477245 137.654876     -1 137.566060       6_137496223
    ## 67    2 165.078693 165.090750     -1 165.084722       2_165053217
    ## 68    5 105.115767 105.139586     -1 105.127676       5_105137202
    ## 69   14   4.334763   4.354687      1   4.344725        14_4347733
    ## 70    2 165.666497 165.666857     -1 165.666677       2_165677109
    ## 71   11  54.947435  54.947888      1  54.947662       11_54909911
    ## 72    2 162.907924 162.908613      1 162.908268       2_162907146
    ## 73   11  46.450980  46.451347     -1  46.451164       11_46423983
    ## 74    2 164.354070 164.389095     -1 164.371582       2_164348452
    ## 75    2 164.424247 164.443887     -1 164.434067       2_164348452
    ## 76   11  77.880615  77.894096     -1  77.887356       11_77931903
    ## 77   11  99.041244  99.054392      1  99.047818       11_99061867
    ## 78    2 165.276554 165.287869     -1 165.282212       2_165283031
    ## 79    2 165.374264 165.388259     -1 165.381262       2_165445995
    ## 80    2 163.602314 163.619013      1 163.610664       2_163582672
    ## 81    2 163.726584 163.750239     -1 163.738412       2_163582672
    ## 82    2 163.623272 163.645131     -1 163.634202       2_163582672
    ## 83    2 164.470301 164.473724     -1 164.472012       2_164348452
    ## 84    2 164.497520 164.508301      1 164.502910       2_164605360
    ## 85    2 164.562413 164.568510      1 164.565462       2_164605360
    ## 86    2 164.486106 164.493319      1 164.489712       2_164605360
    ## 87    2 164.960816 164.999727      1 164.980272       2_165053217
    ## 88    2 164.839518 164.857711     -1 164.848614       2_164758936
    ## 89   13  73.763697  73.816743      1  73.790220       13_73735345
    ## 90    2 164.832873 164.841032      1 164.836952       2_164758936
    ## 91    X   8.034212   8.035079      1   8.034646         X_8061743
    ## 92    4 126.476302 126.476671     -1 126.476486       4_126475807
    ## 93    2 166.273908 166.294831      1 166.284370       2_166324274
    ## 94    3  36.515057  36.521813     -1  36.518435        3_36522678
    ## 95    2 164.911354 164.919953      1 164.915654       2_165053217
    ## 96    2 164.459896 164.460955     -1 164.460426       2_164348452
    ## 97    2 167.201576 167.202799      1 167.202188       2_167142758
    ## 98    4 156.052715 156.059802     -1 156.056258       4_156009474
    ## 99    2 165.595032 165.771727      1 165.683380       2_165677109
    ## 100   2 163.506808 163.572910      1 163.539859       2_163520139
    ## 101   2 167.191805 167.240604     -1 167.216204       2_167274124
    ## 102  15  79.258195  79.285537     -1  79.271866       15_79466906
    ## 103   2 164.070322 164.155524      1 164.112923       2_164076561
    ## 104   2 164.053540 164.071169     -1 164.062354       2_164064248
    ## 105   2 163.994960 164.018588      1 164.006774       2_164022416
    ## 106   2 167.098794 167.101826     -1 167.100310       2_167073838
    ## 107   2 165.880025 165.882056      1 165.881040       2_165880388
    ## 108  15  10.358532  10.410153      1  10.384342       15_10382082
    ## 109   1  36.601802  36.602146      1  36.601974        1_36564755
    ## 110   2 167.642608 167.689822     -1 167.666215       2_167647654
    ## 111   2 166.943539 166.944308      1 166.943924       2_166951357
    ## 112  14   4.664192   4.670850      1   4.667521        14_4674456
    ## 113   6  65.128571  65.129002      1  65.128786        6_65185132
    ## 114   1 166.130238 166.166510      1 166.148374       1_166175328
    ## 115  14   6.016759   6.038209     -1   6.027484        14_6022189
    ## 116  14   3.963547   3.970207      1   3.966877        14_3980169
    ## 117  14   6.583959   6.874257     -1   6.729108        14_6716476
    ## 118  14   4.726775   4.747493      1   4.737134        14_4756137
    ## 119  14   5.071040   5.093318      1   5.082179        14_5082860
    ## 120  14   3.810074   3.966518      1   3.888296        14_3898489
    ## 121   X  73.673150  73.682502      1  73.677826        X_73645374
    ## 122  10   3.540240   3.554877      1   3.547558        10_3557254
    ## 123  14   3.822531   3.884087     -1   3.853309        14_3857648
    ## 124  17   9.320395   9.321381     -1   9.320888        17_9336600
    ## 125  11  11.814101  11.898144     -1  11.856122       11_11852460
    ## 126  11  49.244191  49.263030      1  49.253610       11_49307545
    ## 127  11 109.669749 109.722279     -1 109.696014      11_109696100
    ## 128  11 110.399122 110.525522      1 110.462322      11_110473207
    ## 129  12   5.077472   5.375682     -1   5.226577        12_5346973
    ## 130  11 115.163341 115.181181      1 115.172261      11_115173327
    ## 131  11 115.892595 115.933477     -1 115.913036      11_115923957
    ## 132  11  70.700548  70.731964      1  70.716256       11_70728481
    ## 133  11  76.778424  76.847018     -1  76.812721       11_76717210
    ## 134  12 105.784694 105.838944      1 105.811819      12_105822211
    ## 135  12 108.686792 108.702306     -1 108.694549      12_108722686
    ## 136  14  63.198922  63.271692     -1  63.235307       14_63183201
    ## 137  15  85.734983  85.802819      1  85.768901       15_85744532
    ## 138  16  33.794037  33.819934      1  33.806986       16_33822222
    ## 139  17  44.188572  44.280168      1  44.234370       17_44070746
    ## 140  17  31.125306  31.129646     -1  31.127476       17_31127745
    ## 141  18  20.030633  20.059554     -1  20.045094       18_20031679
    ## 142  18  31.959234  32.036961     -1  31.998098       18_31966977
    ## 143  18  67.404767  67.449166     -1  67.426966       18_67390579
    ## 144  18  82.392496  82.406777     -1  82.399636       18_82346839
    ## 145  19   6.227765   6.235872     -1   6.231818        19_6205523
    ## 146  19  44.989101  45.003397      1  44.996249       19_44938797
    ## 147  10 127.078907 127.093169      1 127.086038      10_127086625
    ## 148   1  59.162926  59.237231     -1  59.200078        1_59158177
    ## 149   9  44.793776  44.799307     -1  44.796542        9_44907739
    ## 150   1  74.409376  74.435559      1  74.422468        1_74424810
    ## 151   1  74.391509  74.397285      1  74.394397        1_74388955
    ## 152   1 194.619829 194.816869      1 194.718349       1_194576403
    ## 153   2 104.721784 104.742878     -1 104.732331       2_104752033
    ## 154   2 122.594467 122.611303     -1 122.602885       2_122693523
    ## 155   2 119.742337 119.751263      1 119.746800       2_119619480
    ## 156   2 119.751320 119.760431     -1 119.755876       2_119619480
    ## 157   2 155.563181 155.592810     -1 155.577996       2_155402931
    ## 158   2 165.503787 165.519917      1 165.511852       2_165495439
    ## 159   2 165.992636 166.073242      1 166.032939       2_165930685
    ## 160   3  90.450591  90.465866     -1  90.458228        3_90438761
    ## 161   4  49.535995  49.549546     -1  49.542770        4_49532869
    ## 162   4  56.991972  57.143437     -1  57.067704        4_57069475
    ## 163   4 123.349633 123.684360     -1 123.516996       4_123347784
    ## 164   4 139.622866 139.649690      1 139.636278       4_139638987
    ## 165   5  32.232708  32.366520      1  32.299614        5_32323492
    ## 166   5  53.107084  53.213452     -1  53.160268        5_53154364
    ## 167   5  87.513339  87.538195     -1  87.525767        5_87318825
    ## 168   5  87.554645  87.569027     -1  87.561836        5_87762267
    ## 169  15  98.632220  98.644530      1  98.638375       15_98637786
    ## 170  15  98.645134  98.662889     -1  98.654012       15_98654559
    ## 171  11   4.097039   4.123415     -1   4.110227        11_4152329
    ## 172  10  53.904788  54.075796     -1  53.990292       10_53983416
    ## 173   6  65.117293  65.144908     -1  65.131100        6_65185132
    ## 174   6 134.396318 134.438736      1 134.417527       6_134397323
    ## 175   6 148.988071 149.101680     -1 149.044876       6_149186501
    ## 176   6 114.643097 114.860614      1 114.751856       6_114807823
    ## 177   7  46.195350  46.238503     -1  46.216926        7_46280163
    ## 178   X   7.930120   7.947889     -1   7.939004         X_7941585
    ## 179   6  70.844515  70.905245      1  70.874880        6_70877098
    ## 180   8 119.809225 119.840579     -1 119.824902       8_119824486
    ## 181   8 117.095854 117.133720      1 117.114787       8_117113822
    ## 182   X   7.153810   7.319358     -1   7.236584         X_7250880
    ## 183   9  45.172726  45.204092     -1  45.188409        9_45210468
    ## 184   9  62.858104  62.875918      1  62.867011        9_62944510
    ## 185   9  76.490054  76.567116     -1  76.528585        9_76500010
    ## 186   9  90.054152  90.076089      1  90.065120        9_90024007
    ## 187   9  80.165031  80.311729      1  80.238380        9_80236002
    ## 188   3 109.340653 109.685698      1 109.513176       3_109512586
    ## 189  18  24.709445  25.133500      1  24.921472       18_25080663
    ## 190  11 115.861612 115.892603      1 115.877108      11_115835369
    ## 191  11   3.494375   3.504821     -1   3.499598        11_3454184
    ## 192   5  96.373955  96.784728      1  96.579342        5_96572141
    ## 193  11   4.091480   4.095445     -1   4.093462        11_4059120
    ## 194  11 114.675431 114.720977      1 114.698204      11_114690549
    ## 195  13  54.701461  54.736662      1  54.719062       13_54717917
    ## 196   2  62.330073  62.412231     -1  62.371152        2_62344746
    ## 197   2 163.859751 163.918683     -1 163.889217       2_164022416
    ## 198   2 163.658386 163.726158      1 163.692272       2_163582672
    ## 199  10  79.821021  79.830452      1  79.825736       10_79831010
    ## 200  19  10.208272  10.240748     -1  10.224510       19_10221349
    ## 201   1  89.070415  89.155068      1  89.112742        1_89083210
    ## 202   5 140.872999 141.000596     -1 140.936798       5_140936564
    ## 203   5 139.200637 139.249840      1 139.225238       5_139215508
    ## 204   4 134.744735 134.768024     -1 134.756380       4_134751908
    ## 205   3  35.754134  35.856274      1  35.805204        3_35928964
    ## 206   5 137.030295 137.033351      1 137.031823       5_137013914
    ## 207   5  69.109285  69.341709     -1  69.225497        5_69244496
    ##                  biotype         module hotspot
    ## 1         protein_coding           grey    chr2
    ## 2         protein_coding          green    chr2
    ## 3         protein_coding           grey    chr2
    ## 4         protein_coding          brown    chr2
    ## 5         protein_coding           grey    chr2
    ## 6         protein_coding      royalblue    <NA>
    ## 7         protein_coding           grey    <NA>
    ## 8         protein_coding           grey    <NA>
    ## 9         protein_coding         purple    <NA>
    ## 10        protein_coding         yellow    <NA>
    ## 11        protein_coding    greenyellow    chr2
    ## 12        protein_coding          green    chr2
    ## 13        protein_coding        magenta    chr7
    ## 14               lincRNA           grey    <NA>
    ## 15        protein_coding    yellowgreen    <NA>
    ## 16        protein_coding         yellow    chr2
    ## 17        protein_coding           grey    chr2
    ## 18        protein_coding           grey    chr2
    ## 19        protein_coding           grey    chr7
    ## 20        protein_coding          black    chr2
    ## 21        protein_coding           grey    chr2
    ## 22        protein_coding     lightgreen    chr2
    ## 23        protein_coding          brown    chr2
    ## 24        protein_coding      turquoise    chr2
    ## 25        protein_coding          brown    chr2
    ## 26        protein_coding        magenta    chr5
    ## 27        protein_coding        darkred    <NA>
    ## 28        protein_coding           grey    chr2
    ## 29        protein_coding           grey    chr2
    ## 30        protein_coding           grey    chr2
    ## 31        protein_coding         yellow    chr2
    ## 32        protein_coding      turquoise    chr2
    ## 33        protein_coding         yellow    <NA>
    ## 34        protein_coding         yellow    chr2
    ## 35        protein_coding           grey    chr2
    ## 36        protein_coding      royalblue    chr2
    ## 37        protein_coding         yellow    chr2
    ## 38        protein_coding           blue    <NA>
    ## 39        protein_coding         yellow    <NA>
    ## 40        protein_coding         purple    chr2
    ## 41        protein_coding           grey    chr2
    ## 42        protein_coding          black    chr2
    ## 43        protein_coding     lightgreen    chr2
    ## 44        protein_coding           grey    chr2
    ## 45        protein_coding         purple    <NA>
    ## 46        protein_coding           grey    chr2
    ## 47        protein_coding      steelblue    <NA>
    ## 48        protein_coding           grey    <NA>
    ## 49        protein_coding      royalblue    chr2
    ## 50        protein_coding          brown    chr2
    ## 51             antisense           grey    <NA>
    ## 52        protein_coding      turquoise    chr2
    ## 53        protein_coding    greenyellow    chr2
    ## 54        protein_coding     lightgreen    chr2
    ## 55        protein_coding    greenyellow    chr2
    ## 56        protein_coding          green    chr2
    ## 57        protein_coding           grey    chr2
    ## 58        protein_coding         purple    chr2
    ## 59        protein_coding           grey    <NA>
    ## 60        protein_coding         yellow    chr2
    ## 61        protein_coding           grey    chr2
    ## 62        protein_coding   midnightblue    <NA>
    ## 63        protein_coding           grey    <NA>
    ## 64        protein_coding           blue    chr2
    ## 65        protein_coding           grey    chr2
    ## 66        protein_coding            red    chr2
    ## 67        protein_coding      steelblue    chr5
    ## 68        protein_coding        magenta    chr2
    ## 69        protein_coding          plum1    chr2
    ## 70            pseudogene         yellow    <NA>
    ## 71            pseudogene           grey    chr2
    ## 72            pseudogene           grey    <NA>
    ## 73            pseudogene           blue    chr2
    ## 74        protein_coding           grey    <NA>
    ## 75        protein_coding       skyblue3    chr7
    ## 76        protein_coding    greenyellow    chr2
    ## 77        protein_coding    greenyellow    chr2
    ## 78        protein_coding         yellow    <NA>
    ## 79        protein_coding            red    <NA>
    ## 80        protein_coding     lightgreen    <NA>
    ## 81        protein_coding           grey    <NA>
    ## 82        protein_coding           grey    <NA>
    ## 83        protein_coding           grey    <NA>
    ## 84        protein_coding          black    <NA>
    ## 85        protein_coding           grey    <NA>
    ## 86        protein_coding         orange    <NA>
    ## 87        protein_coding          black    <NA>
    ## 88        protein_coding    lightyellow    <NA>
    ## 89        protein_coding  darkturquoise    chr7
    ## 90        protein_coding         yellow    <NA>
    ## 91            pseudogene           grey    chr2
    ## 92            pseudogene           grey    chr2
    ## 93               lincRNA           grey    <NA>
    ## 94  processed_transcript           grey    chr2
    ## 95             antisense           grey    <NA>
    ## 96             antisense           grey    <NA>
    ## 97             antisense           grey    <NA>
    ## 98  processed_transcript           grey    chr2
    ## 99        protein_coding           grey    <NA>
    ## 100       protein_coding         orange    <NA>
    ## 101       protein_coding           grey    <NA>
    ## 102       protein_coding           grey    chr2
    ## 103       protein_coding          white    <NA>
    ## 104       protein_coding      turquoise    <NA>
    ## 105       protein_coding      turquoise    <NA>
    ## 106              lincRNA         grey60    <NA>
    ## 107            antisense           grey    <NA>
    ## 108       protein_coding           grey    chr2
    ## 109           pseudogene           grey    chr2
    ## 110       protein_coding         yellow    chr5
    ## 111            antisense           grey    <NA>
    ## 112       protein_coding           grey    chr2
    ## 113           pseudogene           grey    chr2
    ## 114       protein_coding           grey    chr2
    ## 115       protein_coding          plum1    chr2
    ## 116       protein_coding           grey    chr2
    ## 117       protein_coding          plum1    chr2
    ## 118       protein_coding          plum1    chr2
    ## 119       protein_coding        magenta    chr2
    ## 120       protein_coding          plum1    chr2
    ## 121       protein_coding            red    chr7
    ## 122       protein_coding       darkgrey    chr2
    ## 123 processed_transcript          plum1    chr2
    ## 124           pseudogene           grey    chr2
    ## 125       protein_coding          brown    chr2
    ## 126       protein_coding         yellow    chr2
    ## 127       protein_coding         yellow    chr2
    ## 128       protein_coding           grey    chr2
    ## 129       protein_coding         yellow    chr2
    ## 130       protein_coding           grey    chr2
    ## 131       protein_coding         purple    chr2
    ## 132       protein_coding          black    chr2
    ## 133       protein_coding           pink    chr2
    ## 134       protein_coding           grey    chr5
    ## 135       protein_coding           grey    chr2
    ## 136       protein_coding       darkgrey    chr2
    ## 137       protein_coding           grey    chr2
    ## 138       protein_coding      royalblue    chr2
    ## 139       protein_coding        magenta    chr2
    ## 140       protein_coding           grey    chr2
    ## 141       protein_coding          brown    chr2
    ## 142       protein_coding    yellowgreen    chr2
    ## 143       protein_coding        darkred    chr2
    ## 144       protein_coding           grey    chr2
    ## 145       protein_coding          black    chr2
    ## 146       protein_coding         yellow    chr2
    ## 147       protein_coding           grey    chr2
    ## 148       protein_coding           grey    chr2
    ## 149       protein_coding         purple    chr2
    ## 150       protein_coding            red    chr2
    ## 151       protein_coding         grey60    chr2
    ## 152       protein_coding    saddlebrown    chr2
    ## 153       protein_coding          brown    chr5
    ## 154       protein_coding          brown    chr7
    ## 155       protein_coding         yellow    <NA>
    ## 156       protein_coding           grey    <NA>
    ## 157       protein_coding           grey    <NA>
    ## 158       protein_coding           grey    <NA>
    ## 159       protein_coding   midnightblue    <NA>
    ## 160       protein_coding          black    chr2
    ## 161       protein_coding           grey    chr2
    ## 162       protein_coding           grey    chr2
    ## 163       protein_coding    saddlebrown    chr2
    ## 164       protein_coding         purple    chr2
    ## 165       protein_coding           grey    chr2
    ## 166       protein_coding         yellow    chr2
    ## 167       protein_coding           grey    chr2
    ## 168       protein_coding           grey    chr2
    ## 169       protein_coding          black    chr2
    ## 170       protein_coding        darkred    chr2
    ## 171       protein_coding           grey    chr2
    ## 172       protein_coding           grey    chr5
    ## 173       protein_coding           grey    chr2
    ## 174       protein_coding           grey    chr2
    ## 175       protein_coding           grey    chr2
    ## 176       protein_coding         grey60    chr2
    ## 177       protein_coding           grey    chr2
    ## 178       protein_coding         purple    chr2
    ## 179       protein_coding darkolivegreen    chr2
    ## 180       protein_coding           blue    chr2
    ## 181       protein_coding           grey    chr5
    ## 182       protein_coding           grey    chr5
    ## 183       protein_coding         yellow    chr2
    ## 184       protein_coding           grey    chr2
    ## 185       protein_coding           grey    chr2
    ## 186       protein_coding           grey    chr2
    ## 187       protein_coding           pink    chr2
    ## 188       protein_coding           grey    chr2
    ## 189       protein_coding           grey    chr2
    ## 190       protein_coding           grey    chr2
    ## 191       protein_coding         yellow    chr2
    ## 192       protein_coding  darkturquoise    chr2
    ## 193       protein_coding      turquoise    chr2
    ## 194       protein_coding          black    chr2
    ## 195       protein_coding    greenyellow    chr2
    ## 196       protein_coding           grey    <NA>
    ## 197       protein_coding  paleturquoise    <NA>
    ## 198       protein_coding           blue    <NA>
    ## 199       protein_coding        magenta    chr2
    ## 200       protein_coding    greenyellow    chr2
    ## 201       protein_coding darkolivegreen    chr2
    ## 202       protein_coding           grey    chr2
    ## 203       protein_coding           pink    chr2
    ## 204       protein_coding         yellow    chr2
    ## 205       protein_coding      turquoise    chr2
    ## 206       protein_coding     orangered4    chr2
    ## 207       protein_coding            tan    chr2

The above tibble has local and nonlocal traits.

``` r
local_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>%
  mutate(local_tx = chr ==hot_chrom) %>%
  filter(local_tx)
```

``` r
nonlocal_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>% #genomewide threshold
  mutate(local_tx = chr == hot_chrom) %>%
  filter(!local_tx) %>% 
  filter(!is.na(hotspot))
```

``` r
dim(nonlocal_annot)
```

    ## [1] 147  16

We’ll use only those nonlocal traits that Keller et al. annotated as
mapping to a hotspot.

We also want to look at the local traits for Chr 2 hotspot. We have 60
local traits. We want to identify a subset of those for two-dimensional
scans. Let’s first limit to those near the hotspot, ie, within 5Mb of
either end.

``` r
local_annot2 <- local_annot %>%
  filter(middle <= hot_upper + 5, middle >= hot_lower - 5) %>%
  arrange(desc(lod))
```

Recall that *Pdx1* is the gene that Keller et al. identified as the key
mediator for Chr 5 hotspot. Its LOD is only 15, which seems to indicate
that I should be looking at a broad set of local traits, not just those,
eg., with LOD \> 40\!

Also, I’m limiting consideration of local genes to those with middle
within 5Mb of the hotspot.

Now, I make the two expression trait matrices that I need for use with
Condor.

``` r
hot_local <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% local_annot2$annot.id]
hot_nonlocal <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% nonlocal_annot$annot.id]
saveRDS(hot_local, "../data/Chr2hot_local.rds") # change this
saveRDS(hot_nonlocal, "../data/Chr2hot_nonlocal.rds") # change this
saveRDS(K$`2`, "../data/Chr2_kinship.rds") # change this - both
saveRDS(genoprobs$`2`, "../data/Chr2_aprobs.rds") # change this - both
```

Lastly, we need to identify the scan region for Chr 2.

We consider the physical
map.

``` r
(start_index <- which(map$`2`== map$`2`[map$`2` > hot_lower][1]) - 50) # change this - all 3!
```

    ## 2_163507977 
    ##        3742

``` r
(stop_index <- which(map$`2`== map$`2`[map$`2` > hot_upper][1]) + 50)
```

    ## 2_167523201 
    ##        3988

``` r
# change this - all 3!
```

``` r
(nsnp <- stop_index - start_index + 1)
```

    ## 2_167523201 
    ##         247

``` r
dim(local_annot2)
```

    ## [1] 52 16

## References

Keller et al. 2018. Genetics. Genetic drivers of pancreatic islet
function.

## Session info

``` r
devtools::session_info()
```

    ## ─ Session info ──────────────────────────────────────────────────────────
    ##  setting  value                       
    ##  version  R version 3.5.3 (2019-03-11)
    ##  os       macOS Mojave 10.14.5        
    ##  system   x86_64, darwin15.6.0        
    ##  ui       X11                         
    ##  language (EN)                        
    ##  collate  en_US.UTF-8                 
    ##  ctype    en_US.UTF-8                 
    ##  tz       America/Chicago             
    ##  date     2019-06-08                  
    ## 
    ## ─ Packages ──────────────────────────────────────────────────────────────
    ##  package     * version date       lib source                        
    ##  assertthat    0.2.1   2019-03-21 [1] CRAN (R 3.5.2)                
    ##  backports     1.1.4   2019-04-10 [1] CRAN (R 3.5.2)                
    ##  bit           1.1-14  2018-05-29 [1] CRAN (R 3.5.0)                
    ##  bit64         0.9-7   2017-05-08 [1] CRAN (R 3.5.0)                
    ##  blob          1.1.1   2018-03-25 [1] CRAN (R 3.5.0)                
    ##  broom         0.5.1   2018-12-05 [1] CRAN (R 3.5.0)                
    ##  callr         3.2.0   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  cellranger    1.1.0   2016-07-27 [1] CRAN (R 3.5.0)                
    ##  cli           1.1.0   2019-03-19 [1] CRAN (R 3.5.3)                
    ##  colorspace    1.4-1   2019-03-18 [1] CRAN (R 3.5.3)                
    ##  crayon        1.3.4   2017-09-16 [1] CRAN (R 3.5.0)                
    ##  data.table    1.12.0  2019-01-13 [1] CRAN (R 3.5.2)                
    ##  DBI           1.0.0   2018-05-02 [1] CRAN (R 3.5.0)                
    ##  desc          1.2.0   2018-05-01 [1] CRAN (R 3.5.0)                
    ##  devtools      2.0.2   2019-04-08 [1] CRAN (R 3.5.2)                
    ##  digest        0.6.19  2019-05-20 [1] CRAN (R 3.5.2)                
    ##  dplyr       * 0.8.0.1 2019-02-15 [1] CRAN (R 3.5.2)                
    ##  evaluate      0.14    2019-05-28 [1] CRAN (R 3.5.3)                
    ##  forcats     * 0.4.0   2019-02-17 [1] CRAN (R 3.5.2)                
    ##  fs            1.3.1   2019-05-06 [1] CRAN (R 3.5.2)                
    ##  generics      0.0.2   2018-11-29 [1] CRAN (R 3.5.0)                
    ##  ggplot2     * 3.1.1   2019-04-07 [1] CRAN (R 3.5.3)                
    ##  glue          1.3.1   2019-03-12 [1] CRAN (R 3.5.2)                
    ##  gtable        0.3.0   2019-03-25 [1] CRAN (R 3.5.2)                
    ##  haven         2.1.0   2019-02-19 [1] CRAN (R 3.5.2)                
    ##  hms           0.4.2   2018-03-10 [1] CRAN (R 3.5.0)                
    ##  htmltools     0.3.6   2017-04-28 [1] CRAN (R 3.5.0)                
    ##  httr          1.4.0   2018-12-11 [1] CRAN (R 3.5.0)                
    ##  jsonlite      1.6     2018-12-07 [1] CRAN (R 3.5.0)                
    ##  knitr         1.23    2019-05-18 [1] CRAN (R 3.5.2)                
    ##  lattice       0.20-38 2018-11-04 [1] CRAN (R 3.5.3)                
    ##  lazyeval      0.2.2   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  lubridate     1.7.4   2018-04-11 [1] CRAN (R 3.5.0)                
    ##  magrittr      1.5     2014-11-22 [1] CRAN (R 3.5.0)                
    ##  memoise       1.1.0   2017-04-21 [1] CRAN (R 3.5.0)                
    ##  modelr        0.1.4   2019-02-18 [1] CRAN (R 3.5.2)                
    ##  munsell       0.5.0   2018-06-12 [1] CRAN (R 3.5.0)                
    ##  nlme          3.1-137 2018-04-07 [1] CRAN (R 3.5.3)                
    ##  pillar        1.3.1   2018-12-15 [1] CRAN (R 3.5.0)                
    ##  pkgbuild      1.0.3   2019-03-20 [1] CRAN (R 3.5.2)                
    ##  pkgconfig     2.0.2   2018-08-16 [1] CRAN (R 3.5.0)                
    ##  pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.5.0)                
    ##  plyr          1.8.4   2016-06-08 [1] CRAN (R 3.5.0)                
    ##  prettyunits   1.0.2   2015-07-13 [1] CRAN (R 3.5.0)                
    ##  processx      3.3.1   2019-05-08 [1] CRAN (R 3.5.2)                
    ##  ps            1.3.0   2018-12-21 [1] CRAN (R 3.5.0)                
    ##  purrr       * 0.3.2   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  qtl2        * 0.18    2019-02-08 [1] local                         
    ##  R6            2.4.0   2019-02-14 [1] CRAN (R 3.5.2)                
    ##  Rcpp          1.0.1.3 2019-05-29 [1] Github (RcppCore/Rcpp@6062d56)
    ##  readr       * 1.3.1   2018-12-21 [1] CRAN (R 3.5.0)                
    ##  readxl        1.3.0   2019-02-15 [1] CRAN (R 3.5.2)                
    ##  remotes       2.0.4   2019-04-10 [1] CRAN (R 3.5.2)                
    ##  rlang         0.3.4   2019-04-07 [1] CRAN (R 3.5.3)                
    ##  rmarkdown     1.13    2019-05-22 [1] CRAN (R 3.5.2)                
    ##  rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.5.0)                
    ##  RSQLite       2.1.1   2018-05-06 [1] CRAN (R 3.5.0)                
    ##  rstudioapi    0.10    2019-03-19 [1] CRAN (R 3.5.2)                
    ##  rvest         0.3.2   2016-06-17 [1] CRAN (R 3.5.0)                
    ##  scales        1.0.0   2018-08-09 [1] CRAN (R 3.5.0)                
    ##  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.5.0)                
    ##  stringi       1.4.3   2019-03-12 [1] CRAN (R 3.5.2)                
    ##  stringr     * 1.4.0   2019-02-10 [1] CRAN (R 3.5.2)                
    ##  testthat      2.1.1   2019-04-23 [1] CRAN (R 3.5.3)                
    ##  tibble      * 2.1.1   2019-03-16 [1] CRAN (R 3.5.3)                
    ##  tidyr       * 0.8.3   2019-03-01 [1] CRAN (R 3.5.2)                
    ##  tidyselect    0.2.5   2018-10-11 [1] CRAN (R 3.5.0)                
    ##  tidyverse   * 1.2.1   2017-11-14 [1] CRAN (R 3.5.0)                
    ##  usethis       1.5.0   2019-04-07 [1] CRAN (R 3.5.2)                
    ##  withr         2.1.2   2018-03-15 [1] CRAN (R 3.5.0)                
    ##  xfun          0.7     2019-05-14 [1] CRAN (R 3.5.2)                
    ##  xml2          1.2.0   2018-01-24 [1] CRAN (R 3.5.0)                
    ##  yaml          2.2.0   2018-07-25 [1] CRAN (R 3.5.0)                
    ## 
    ## [1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library
