---
title: "Setting up for use of Condor CHTC with Chr 2 hotspot"
author: "Frederick Boehm"
date: "06/08/2019"
lastmod: "06/08/2019"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Goals

Goal here is to document the setup & organization of files that I need for using Condor CHTC to do my two-dimensional scans as part of the analyses of the four Keller hotspots. Note that they reported 5 hotspots. 

## Setting

Recall that I have downloaded a Keller 2018 .RData file from DataDryad that contains many of their results. The .RData file, "data/Attie_DO378_eQTL_viewer_v1.Rdata" annotates 139 traits as belonging to the chromosome 2 hotspot.

Following Karl's suggestion, I want to look at not only the local trait that Keller identifies as causal for the hotspot, but also many other local-ish traits that have reasonable univariate LOD scores.


## Preparing RDS files for Chr 2 analysis


```{r, load_attie}
load("~/Box Sync/attie/keller2018-chr2-hotspot-chtc/data-to-ignore/Attie_DO378_eQTL_viewer_v1.Rdata")
ls()
library(tidyverse)
```

```{r checks}
dataset.islet.rnaseq$expr %>%
  dim
colnames(dataset.islet.rnaseq$expr) %>%
  head
```

We need to get the names of the nonlocal traits that map to Chromosome 13 hotspot. 

```{r load_qtl2}
library(qtl2)
```


We want the ensembl ids for the traits that map to Chr 13 hotspot. First, define the boundaries of the hotspot. The supplemental table in Keller et al. 2018 lists only the midpoint of the hotspot. I believe that the half-width is 2 Mb. Yes, this is indeed the half-width per the Rmd file that Dan Gatti prepared.

```{r}
hot_chrom <- 2
hot_center <- 165.5# for chr2
hot_lower <- hot_center - 2
hot_upper <- hot_center + 2
```

Now, filter to get those traits that map to the hotspot. Note that we're likely to get local traits, too, in this step.

```{r}
dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18)
```

The above tibble has local and nonlocal traits.


```{r}
local_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>%
  mutate(local_tx = chr ==hot_chrom) %>%
  filter(local_tx)
```



```{r}
nonlocal_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>% #genomewide threshold
  mutate(local_tx = chr == hot_chrom) %>%
  filter(!local_tx) %>% 
  filter(!is.na(hotspot))
```


```{r}
dim(nonlocal_annot)
```

We'll use only those nonlocal traits that Keller et al. annotated as mapping to a hotspot.

We also want to look at the local traits for Chr `r hot_chrom` hotspot. We have `r nrow(local_annot)` local traits. We want to identify a subset of those for two-dimensional scans. Let's first limit to those near the hotspot, ie, within 5Mb of either end.

```{r}
local_annot2 <- local_annot %>%
  filter(middle <= hot_upper + 5, middle >= hot_lower - 5) %>%
  arrange(desc(lod))
```

Recall that *Pdx1* is the gene that Keller et al. identified as the key mediator for Chr 5 hotspot. Its LOD is only 15, which seems to indicate that I should be looking at a broad set of local traits, not just those, eg., with LOD > 40!

Also, I'm limiting consideration of local genes to those with middle within 5Mb of the hotspot. 

Now, I make the two expression trait matrices that I need for use with Condor.

```{r}
hot_local <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% local_annot2$annot.id]
hot_nonlocal <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% nonlocal_annot$annot.id]
saveRDS(hot_local, "../data/Chr2hot_local.rds") # change this
saveRDS(hot_nonlocal, "../data/Chr2hot_nonlocal.rds") # change this
saveRDS(K$`2`, "../data/Chr2_kinship.rds") # change this - both
saveRDS(genoprobs$`2`, "../data/Chr2_aprobs.rds") # change this - both
```


Lastly, we need to identify the scan region for Chr 2.

We consider the physical map.

```{r}
(start_index <- which(map$`2`== map$`2`[map$`2` > hot_lower][1]) - 50) # change this - all 3!
```

```{r}
(stop_index <- which(map$`2`== map$`2`[map$`2` > hot_upper][1]) + 50)
# change this - all 3!
```

```{r}
(nsnp <- stop_index - start_index + 1)
```


```{r}
dim(local_annot2)
```





## References

Keller et al. 2018. Genetics. Genetic drivers of pancreatic islet function.

## Session info

```{r}
devtools::session_info()
```


