Setting up for use of Condor CHTC with Chr 11 hotspot
================
Frederick Boehm
06/01/2019

## Goals

Goal here is to document the setup & organization of files that I need
for using Condor CHTC to do my two-dimensional scans as part of the
analyses of the four Keller hotspots. Note that they reported 5
hotspots. I previously performed the analyses for Chromosome 2 hotspot.

## Setting

Recall that I have downloaded a Keller 2018 .RData file from DataDryad
that contains many of their results. The .RData file,
“data/Attie\_DO378\_eQTL\_viewer\_v1.Rdata” annotates 139 traits as
belonging to the chromosome 2 hotspot.

Following Karl’s suggestion, I want to look at not only the local trait
that Keller identifies as causal for the hotspot, but also several other
local-ish traits that have reasonable univariate LOD
scores.

## Preparing RDS files for Chr 11 analysis

``` r
load("~/Box Sync/attie/keller2018-chr2-hotspot-chtc/data-to-ignore/Attie_DO378_eQTL_viewer_v1.Rdata")
ls()
```

    ## [1] "dataset.clinical.phenotypes" "dataset.islet.hotspots"     
    ## [3] "dataset.islet.modules"       "dataset.islet.rnaseq"       
    ## [5] "ensembl.version"             "genoprobs"                  
    ## [7] "K"                           "map"                        
    ## [9] "markers"

``` r
library(tidyverse)
```

    ## ── Attaching packages ────────────────────────────────────────────────── tidyverse 1.2.1 ──

    ## ✔ ggplot2 3.1.1       ✔ purrr   0.3.2  
    ## ✔ tibble  2.1.1       ✔ dplyr   0.8.0.1
    ## ✔ tidyr   0.8.3       ✔ stringr 1.4.0  
    ## ✔ readr   1.3.1       ✔ forcats 0.4.0

    ## ── Conflicts ───────────────────────────────────────────────────── tidyverse_conflicts() ──
    ## ✖ dplyr::filter() masks stats::filter()
    ## ✖ dplyr::lag()    masks stats::lag()

``` r
dataset.islet.rnaseq$expr %>%
  dim
```

    ## [1]   378 21771

``` r
colnames(dataset.islet.rnaseq$expr) %>%
  head
```

    ## [1] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000037"
    ## [4] "ENSMUSG00000000049" "ENSMUSG00000000056" "ENSMUSG00000000058"

We need to get the names of the nonlocal traits that map to Chromosome 7
hotspot.

``` r
library(qtl2)
```

    ## 
    ## Attaching package: 'qtl2'

    ## The following object is masked from 'package:readr':
    ## 
    ##     read_csv

I forgot that the LOD peaks for all traits are saved in
`dataset.islet.rnaseq$lod.peaks`. It has nearly 40,000 peaks recorded. I
really just need to look at it. That is, for now, I don’t really need
the output of `scan1()`.

We want the ensembl ids for the traits that map to Chr 5 hotspot. First,
define the boundaries of the hotspot. The supplemental table in Keller
et al. 2018 lists only the midpoint of the hotspot. I believe that the
half-width is 2 Mb. Yes, this is indeed the half-width per the Rmd file
that Dan Gatti prepared.

``` r
hot_chrom <- 11
hot_center <- 71# for chr11
hot_lower <- hot_center - 2
hot_upper <- hot_center + 2
```

Now, filter to get those traits that map to the hotspot. Note that we’re
likely to get local traits, too, in this step.

``` r
dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18)
```

    ##               annot.id   marker.id chrom      pos        lod        symbol
    ## 1   ENSMUSG00000038011 11_72996574    11 72.99657   8.085680        Dnah10
    ## 2   ENSMUSG00000038195 11_71859513    11 71.85951  22.038343          Rilp
    ## 3   ENSMUSG00000038379 11_69505599    11 69.50560   7.287127           Ttk
    ## 4   ENSMUSG00000038393 11_72601385    11 72.60138   7.374881         Txnip
    ## 5   ENSMUSG00000038520 11_69829176    11 69.82918   9.330975       Tbc1d17
    ## 6   ENSMUSG00000039050 11_69810287    11 69.81029   8.683959        Osbpl2
    ## 7   ENSMUSG00000039347 11_71693132    11 71.69313   7.673404      Atp6v0e2
    ## 8   ENSMUSG00000040296 11_69829176    11 69.82918   7.674002         Ddx58
    ## 9   ENSMUSG00000040483 11_72314268    11 72.31427  52.999651          Xaf1
    ## 10  ENSMUSG00000040554 11_72040688    11 72.04069  18.837109         Aipl1
    ## 11  ENSMUSG00000040599 11_71418821    11 71.41882  38.560032         Mis12
    ## 12  ENSMUSG00000040667 11_71354568    11 71.35457  21.371028         Nup88
    ## 13  ENSMUSG00000006586 11_71693132    11 71.69313   9.091885       Runx1t1
    ## 14  ENSMUSG00000040746 11_70750402    11 70.75040  40.707077        Rnf167
    ## 15  ENSMUSG00000040904 11_71120660    11 71.12066  13.954750       Gm21988
    ## 16  ENSMUSG00000040938 11_70616235    11 70.61624  21.419463      Slc16a11
    ## 17  ENSMUSG00000041135 11_69829176    11 69.82918   8.318676         Ripk2
    ## 18  ENSMUSG00000041165 11_72512588    11 72.51259  14.412941         Spem1
    ## 19  ENSMUSG00000041287 11_69064645    11 69.06464   9.351414         Sox15
    ## 20  ENSMUSG00000041329 11_69796313    11 69.79631  13.450483        Atp1b2
    ## 21  ENSMUSG00000041346 11_69542556    11 69.54256  21.965771        Wrap53
    ## 22  ENSMUSG00000041624 11_72959270    11 72.95927   7.187932       Gucy1a2
    ## 23  ENSMUSG00000041801 11_72210521    11 72.21052   9.362473        Phlda3
    ## 24  ENSMUSG00000042628 11_71276599    11 71.27660   8.655518        Zfyve1
    ## 25  ENSMUSG00000042726 11_69829176    11 69.82918   7.951001        Trafd1
    ## 26  ENSMUSG00000043091 11_71510771    11 71.51077   7.777950        Tuba1c
    ## 27  ENSMUSG00000043419 11_69231034    11 69.23103  20.135740 A030009H04Rik
    ## 28  ENSMUSG00000043602 11_70750402    11 70.75040   8.038654          Zfp3
    ## 29  ENSMUSG00000044162 11_71354568    11 71.35457   7.523762         Tnip3
    ## 30  ENSMUSG00000044367 11_70609170    11 70.60917  14.453236      Slc16a13
    ## 31  ENSMUSG00000007867 11_69810287    11 69.81029   9.928308         Ift43
    ## 32  ENSMUSG00000008035 11_71354568    11 71.35457   7.978847       Mid1ip1
    ## 33  ENSMUSG00000008200 11_69837051    11 69.83705   7.474816         Fnbp4
    ## 34  ENSMUSG00000044783 11_69224642    11 69.22464   8.709552         Hjurp
    ## 35  ENSMUSG00000045251 11_69876917    11 69.87692   7.533249        Zfp688
    ## 36  ENSMUSG00000045377 11_69416362    11 69.41636  16.008807        Tmem88
    ## 37  ENSMUSG00000046675 11_71693132    11 71.69313   8.503942       Tmem251
    ## 38  ENSMUSG00000046731 11_69924127    11 69.92413  24.173013        Kctd11
    ## 39  ENSMUSG00000046879 11_71813729    11 71.81373   7.765917         Irgm1
    ## 40  ENSMUSG00000046897 11_71354568    11 71.35457   9.157349        Zfp740
    ## 41  ENSMUSG00000047180 11_70772323    11 70.77232   8.446305        Neurl3
    ## 42  ENSMUSG00000047284 11_72989681    11 72.98968   7.856942        Neurl4
    ## 43  ENSMUSG00000047787 11_71510771    11 71.51077   7.246068         Flrt1
    ## 44  ENSMUSG00000047921 11_72662670    11 72.66267   7.491731       Trappc9
    ## 45  ENSMUSG00000048379 11_71354568    11 71.35457   8.126786         Socs4
    ## 46  ENSMUSG00000048834 11_70750402    11 70.75040  10.506679        Vstm2a
    ## 47  ENSMUSG00000048922 11_69829176    11 69.82918   7.201449         Cdca2
    ## 48  ENSMUSG00000049734 11_69837051    11 69.83705   9.965929         Trex1
    ## 49  ENSMUSG00000050270 11_72040688    11 72.04069   9.860951       Tmem220
    ## 50  ENSMUSG00000050565 11_71693132    11 71.69313   8.728723      Tor1aip2
    ## 51  ENSMUSG00000050675 11_71354568    11 71.35457  43.000217         Gp1ba
    ## 52  ENSMUSG00000051378 11_69810287    11 69.81029   7.297316        Kif18b
    ## 53  ENSMUSG00000051864 11_70785072    11 70.78507   7.471321      Tbc1d22a
    ## 54  ENSMUSG00000052776 11_69796313    11 69.79631   7.947120         Oas1a
    ## 55  ENSMUSG00000053574 11_72210521    11 72.21052  40.827446 4930563E22Rik
    ## 56  ENSMUSG00000053929 11_71354568    11 71.35457   7.962114         Cyhr1
    ## 57  ENSMUSG00000055670 11_72601385    11 72.60138  14.770236         Zzef1
    ## 58  ENSMUSG00000010505 11_71354568    11 71.35457   8.370282          Myt1
    ## 59  ENSMUSG00000056310 11_71354568    11 71.35457   9.807189          Tyw1
    ## 60  ENSMUSG00000057054 11_69810287    11 69.81029  20.916548         Inca1
    ## 61  ENSMUSG00000057135 11_69978753    11 69.97875  11.966446         Scimp
    ## 62  ENSMUSG00000057778 11_72944065    11 72.94406  36.253118        Cyb5d2
    ## 63  ENSMUSG00000059796 11_69592546    11 69.59255  24.995852        Eif4a1
    ## 64  ENSMUSG00000060216 11_71198629    11 71.19863  19.656292         Arrb2
    ## 65  ENSMUSG00000060550 11_69829176    11 69.82918   7.899009         H2-Q7
    ## 66  ENSMUSG00000060600 11_69978753    11 69.97875  56.825370          Eno3
    ## 67  ENSMUSG00000060802 11_69829176    11 69.82918   8.086660           B2m
    ## 68  ENSMUSG00000060862 11_71782692    11 71.78269   8.959168        Zbtb40
    ## 69  ENSMUSG00000061232 11_69829176    11 69.82918   7.834306         H2-K1
    ## 70  ENSMUSG00000014773 11_71354568    11 71.35457  10.717858          Dll1
    ## 71  ENSMUSG00000063681 11_69810287    11 69.81029   8.657519          Crb1
    ## 72  ENSMUSG00000064360 11_71782692    11 71.78269   7.387915        mt-Nd3
    ## 73  ENSMUSG00000067567 11_70782062    11 70.78206   7.581051         Hdac8
    ## 74  ENSMUSG00000068457 11_72450663    11 72.45066   7.999850           Uty
    ## 75  ENSMUSG00000069830 11_71120660    11 71.12066 130.829473        Nlrp1a
    ## 76  ENSMUSG00000069835 11_69768364    11 69.76836  54.752196          Sat2
    ## 77  ENSMUSG00000070390 11_69978753    11 69.97875  11.205844        Nlrp1b
    ## 78  ENSMUSG00000070394 11_69924127    11 69.92413   9.059992       Tmem256
    ## 79  ENSMUSG00000070697 11_71693132    11 71.69313   7.231247          Utp3
    ## 80  ENSMUSG00000015316 11_72040688    11 72.04069   7.449085        Slamf1
    ## 81  ENSMUSG00000015354 11_69837051    11 69.83705   8.191339       Pcolce2
    ## 82  ENSMUSG00000072720 11_71354568    11 71.35457  10.434181        Myo18b
    ## 83  ENSMUSG00000072809 11_69064645    11 69.06464  76.214094 9330160F10Rik
    ## 84  ENSMUSG00000073096 11_69416362    11 69.41636   7.579636        Lrrc61
    ## 85  ENSMUSG00000073409 11_69837051    11 69.83705   7.866018         H2-Q6
    ## 86  ENSMUSG00000073705 11_72210521    11 72.21052   7.248966        Apitd1
    ## 87  ENSMUSG00000073968 11_72040688    11 72.04069   8.040305        Trim68
    ## 88  ENSMUSG00000074890 11_71813729    11 71.81373   7.849553         Lcmt2
    ## 89  ENSMUSG00000078429 11_71429405    11 71.42941   9.299295        Ctdsp2
    ## 90  ENSMUSG00000078493 11_71354568    11 71.35457   7.495058 A930039A15Rik
    ## 91  ENSMUSG00000015846 11_71354568    11 71.35457   7.630420          Rxra
    ## 92  ENSMUSG00000016529 11_71693132    11 71.69313   7.198759          Il10
    ## 93  ENSMUSG00000082209 11_69733033    11 69.73303  25.923634     Eif4a-ps4
    ## 94  ENSMUSG00000084139 11_71793079    11 71.79308  16.754042       Gm12322
    ## 95  ENSMUSG00000085178 11_69484146    11 69.48415  14.780695 1700064E03Rik
    ## 96  ENSMUSG00000086517 11_70785072    11 70.78507   7.201667       Gm12534
    ## 97  ENSMUSG00000086591 11_69227838    11 69.22784  20.847870 6030455H03Rik
    ## 98  ENSMUSG00000018286 11_69829176    11 69.82918  12.644749         Psmb6
    ## 99  ENSMUSG00000018442 11_71354568    11 71.35457  24.331853         Derl2
    ## 100 ENSMUSG00000018446 11_71937774    11 71.93777  13.821042         C1qbp
    ## 101 ENSMUSG00000018449 11_71443287    11 71.44329  15.495242         Rpain
    ## 102 ENSMUSG00000018451 11_71120660    11 71.12066  66.532115 6330403K07Rik
    ## 103 ENSMUSG00000018559 11_69663280    11 69.66328   8.892792       Ctdnep1
    ## 104 ENSMUSG00000018565 11_70728481    11 70.72848  10.131916          Elp5
    ## 105 ENSMUSG00000018566 11_69876397    11 69.87640  17.154645        Slc2a4
    ## 106 ENSMUSG00000086976 11_69810287    11 69.81029   9.754274       Gm12316
    ## 107 ENSMUSG00000087523 11_71782692    11 71.78269   7.895661       Gm12319
    ## 108 ENSMUSG00000089669 11_69131341    11 69.13134  13.547150       Tnfsf13
    ## 109 ENSMUSG00000089876 11_69484146    11 69.48415  14.365592       Tmem102
    ## 110 ENSMUSG00000018570 11_69876397    11 69.87640  10.841222 2810408A11Rik
    ## 111 ENSMUSG00000018740 11_69505599    11 69.50560   9.305293      Slc25a35
    ## 112 ENSMUSG00000018750 11_70785072    11 70.78507  10.780532         Zbtb4
    ## 113 ENSMUSG00000018752 11_69484146    11 69.48415  19.739891      BC096441
    ## 114 ENSMUSG00000018761 11_69796313    11 69.79631  13.644976         Mpdu1
    ## 115 ENSMUSG00000018765 11_69810287    11 69.81029  11.424120          Fxr2
    ## 116 ENSMUSG00000018923 11_71449172    11 71.44917  10.540510         Med11
    ## 117 ENSMUSG00000018925 11_69542556    11 69.54256   7.451503       Gm11435
    ## 118 ENSMUSG00000000194 11_69876917    11 69.87692   7.703307        Gpr107
    ## 119 ENSMUSG00000000320 11_69844926    11 69.84493  29.387372        Alox12
    ## 120 ENSMUSG00000092286 11_71449172    11 71.44917   8.959010       Gm20448
    ## 121 ENSMUSG00000093989 11_71429405    11 71.42941  24.860846        Rnasek
    ## 122 ENSMUSG00000094483 11_71693132    11 71.69313   7.816143          Purb
    ## 123 ENSMUSG00000094845 11_69231034    11 69.23103  10.939589        Tmem95
    ## 124 ENSMUSG00000097328 11_69796313    11 69.79631  25.915258       Tnfsf12
    ## 125 ENSMUSG00000097386 11_69227838    11 69.22784  10.690643 9130213A22Rik
    ## 126 ENSMUSG00000097418 11_71354568    11 71.35457   7.231001       Gm26667
    ## 127 ENSMUSG00000097523 11_69484146    11 69.48415   7.307315 1700022N22Rik
    ## 128 ENSMUSG00000097772 11_69214855    11 69.21486   7.790326 5430416N02Rik
    ## 129 ENSMUSG00000097797 11_71693132    11 71.69313   7.675529       Gm26901
    ## 130 ENSMUSG00000019900 11_71693132    11 71.69313   8.747553          Rfx6
    ## 131 ENSMUSG00000020547 11_72974476    11 72.97448   8.181821          Bzw2
    ## 132 ENSMUSG00000020572 11_69829176    11 69.82918   7.643970         Nampt
    ## 133 ENSMUSG00000020794 11_72040688    11 72.04069   9.550537        Ube2g1
    ## 134 ENSMUSG00000020799 11_72366456    11 72.36646  73.157007         Tekt1
    ## 135 ENSMUSG00000020801 11_72210521    11 72.21052  28.775964         Med31
    ## 136 ENSMUSG00000020803 11_71859513    11 71.85951  12.298288       Txndc17
    ## 137 ENSMUSG00000020808 11_71782692    11 71.78269   7.709167        Fam64a
    ## 138 ENSMUSG00000020811 11_71782692    11 71.78269  28.561910         Wscd1
    ## 139 ENSMUSG00000020817 11_71354568    11 71.35457  11.510839        Rabep1
    ## 140 ENSMUSG00000020827 11_70661194    11 70.66119  18.670941         Mink1
    ## 141 ENSMUSG00000020828 11_71354568    11 71.35457  25.935898          Pld2
    ## 142 ENSMUSG00000020884 11_70785072    11 70.78507  10.521628         Asgr1
    ## 143 ENSMUSG00000020891 11_69131341    11 69.13134  68.881743         Alox8
    ## 144 ENSMUSG00000020892 11_69064645    11 69.06464   8.654159        Aloxe3
    ## 145 ENSMUSG00000020894 11_69131341    11 69.13134   8.221452         Vamp2
    ## 146 ENSMUSG00000020895 11_69131341    11 69.13134  18.238940       Tmem107
    ## 147 ENSMUSG00000020897 11_69234230    11 69.23423   8.878828         Aurkb
    ## 148 ENSMUSG00000020904 11_69064645    11 69.06464  12.015604         Wdr16
    ## 149 ENSMUSG00000021025 11_70750402    11 70.75040   8.487403        Nfkbia
    ## 150 ENSMUSG00000021176 11_69592546    11 69.59255   8.091173       Efcab11
    ## 151 ENSMUSG00000021185 11_72040688    11 72.04069   8.873081 9030617O03Rik
    ## 152 ENSMUSG00000000686 11_71693132    11 71.69313   7.306978        Abhd15
    ## 153 ENSMUSG00000000982 11_69837051    11 69.83705   7.747335          Ccl3
    ## 154 ENSMUSG00000001156 11_70785072    11 70.78507   7.693250          Mxd1
    ## 155 ENSMUSG00000022216 11_69829176    11 69.82918   8.161038         Psme1
    ## 156 ENSMUSG00000022360 11_69829176    11 69.82918   7.306121         Atad2
    ## 157 ENSMUSG00000022385 11_69810287    11 69.81029   8.978982         Gtse1
    ## 158 ENSMUSG00000022440 11_72012709    11 72.01271   7.758218       C1qtnf6
    ## 159 ENSMUSG00000022442 11_72601385    11 72.60138   8.636583         Ttll1
    ## 160 ENSMUSG00000022637 11_69810287    11 69.81029  10.906303          Cblb
    ## 161 ENSMUSG00000023015 11_69829176    11 69.82918   8.322247       Racgap1
    ## 162 ENSMUSG00000023030 11_69810287    11 69.81029   8.958892       Slc11a2
    ## 163 ENSMUSG00000023393 11_71937774    11 71.93777   7.324025       Slc17a9
    ## 164 ENSMUSG00000001583 11_69837051    11 69.83705 103.526684          Tnk1
    ## 165 ENSMUSG00000001829 11_71693132    11 71.69313   8.122187          Clpb
    ## 166 ENSMUSG00000025265 11_71693132    11 71.69313   7.979446          Fgd1
    ## 167 ENSMUSG00000025474 11_71782692    11 71.78269   7.577438       Tubgcp2
    ## 168 ENSMUSG00000001942 11_69810287    11 69.81029   7.987770          Siae
    ## 169 ENSMUSG00000026238 11_69876917    11 69.87692   7.295952          Ptma
    ## 170 ENSMUSG00000026473 11_71354568    11 71.35457   8.480388          Glul
    ## 171 ENSMUSG00000026479 11_71354568    11 71.35457   8.660927         Lamc2
    ## 172 ENSMUSG00000026670 11_71276599    11 71.27660   7.982495          Uap1
    ## 173 ENSMUSG00000026981 11_71658293    11 71.65829   8.690513         Il1rn
    ## 174 ENSMUSG00000027306 11_69829176    11 69.82918   7.940493        Nusap1
    ## 175 ENSMUSG00000027613 11_70661194    11 70.66119   7.320860          Eif6
    ## 176 ENSMUSG00000027674 11_71834380    11 71.83438   7.335886         Pex5l
    ## 177 ENSMUSG00000027698 11_70785072    11 70.78507   7.375834         Nceh1
    ## 178 ENSMUSG00000002728 11_69829176    11 69.82918   9.173427         Naa20
    ## 179 ENSMUSG00000028266 11_69542556    11 69.54256   8.120862          Lmo4
    ## 180 ENSMUSG00000028312 11_69810287    11 69.81029   7.303971          Smc2
    ## 181 ENSMUSG00000028655 11_71354568    11 71.35457   8.030973        Mfsd2a
    ## 182 ENSMUSG00000028793 11_71813729    11 71.81373   8.229299        Rnf19b
    ## 183 ENSMUSG00000028919 11_72989681    11 72.98968   7.198013      Arhgef19
    ## 184 ENSMUSG00000029130 11_72989681    11 72.98968   8.971665         Rnf32
    ## 185 ENSMUSG00000029757 11_70661194    11 70.66119   8.352589       Dync1i1
    ## 186 ENSMUSG00000029817 11_71354568    11 71.35457   7.216908         Tra2a
    ## 187 ENSMUSG00000003382 11_71449172    11 71.44917   7.272108          Etv3
    ## 188 ENSMUSG00000030083 11_71693132    11 71.69313   8.674318         Abtb1
    ## 189 ENSMUSG00000030110 11_71354568    11 71.35457   9.032825           Ret
    ## 190 ENSMUSG00000030654 11_71813729    11 71.81373   7.340505       Arl6ip1
    ## 191 ENSMUSG00000030867 11_69810287    11 69.81029   7.253676          Plk1
    ## 192 ENSMUSG00000031119 11_71418821    11 71.41882   9.185994          Gpc4
    ## 193 ENSMUSG00000031370 11_71859513    11 71.85951   7.967773         Zrsr2
    ## 194 ENSMUSG00000031762 11_71354568    11 71.35457  12.546048           Mt2
    ## 195 ENSMUSG00000031765 11_71354568    11 71.35457  10.572808           Mt1
    ## 196 ENSMUSG00000032064 11_72989681    11 72.98968   7.501906        Dixdc1
    ## 197 ENSMUSG00000032369 11_69810287    11 69.81029   7.799061        Plscr1
    ## 198 ENSMUSG00000032698 11_71120660    11 71.12066   9.466752          Lmo2
    ## 199 ENSMUSG00000032807 11_69227838    11 69.22784  27.739946       Alox12b
    ## 200 ENSMUSG00000034066 11_71354568    11 71.35457   9.117575         Farp2
    ## 201 ENSMUSG00000034245 11_71693132    11 71.69313   7.948531        Hdac11
    ## 202 ENSMUSG00000034422 11_69829176    11 69.82918   7.646596        Parp14
    ## 203 ENSMUSG00000034457 11_69810287    11 69.81029   8.686231         Eda2r
    ## 204 ENSMUSG00000034522 11_71675712    11 71.67571   8.538309        Zfp395
    ## 205 ENSMUSG00000005198 11_69796313    11 69.79631  10.102583        Polr2a
    ## 206 ENSMUSG00000005204 11_69782339    11 69.78234  13.126534         Senp3
    ## 207 ENSMUSG00000005237 11_69227838    11 69.22784  17.740281         Dnah2
    ## 208 ENSMUSG00000035246 11_72996574    11 72.99657   8.881481        Pcyt1b
    ## 209 ENSMUSG00000035273 11_69829176    11 69.82918   8.117479          Hpse
    ## 210 ENSMUSG00000035505 11_72996574    11 72.99657   7.747425         Cox18
    ## 211 ENSMUSG00000035929 11_69542556    11 69.54256   7.351569         H2-Q4
    ## 212 ENSMUSG00000035944 11_70785072    11 70.78507   8.187956         Ttc38
    ## 213 ENSMUSG00000036086 11_72974476    11 72.97448   9.123252        Zranb3
    ## 214 ENSMUSG00000036501 11_71834380    11 71.83438   7.364448        Fam13b
    ## 215 ENSMUSG00000037235 11_71354568    11 71.35457   9.388531          Mxd4
    ## 216 ENSMUSG00000037300 11_72999600    11 72.99960  10.346201         Ttc13
    ## 217 ENSMUSG00000037434 11_71813729    11 71.81373   8.004838       Slc30a1
    ## 218 ENSMUSG00000037685 11_72314268    11 72.31427   8.806341        Atp8a1
    ##     chr      start        end strand     middle nearest.marker.id
    ## 1     5 124.725085 124.834308      1 124.779696       5_124781618
    ## 2    11  75.510094  75.513168      1  75.511631       11_75466326
    ## 3     9  83.834689  83.872390      1  83.853540        9_83866054
    ## 4     3  96.557957  96.561883      1  96.559920        3_96523481
    ## 5     7  44.840901  44.849079     -1  44.844990        7_44838742
    ## 6     2 180.119306 180.162680      1 180.140993       2_180122688
    ## 7     6  48.537569  48.541798      1  48.539684        6_48551052
    ## 8     4  40.203773  40.239828     -1  40.221800        4_40238373
    ## 9    11  72.301629  72.313733      1  72.307681       11_72314268
    ## 10   11  72.027963  72.037509     -1  72.032736       11_72040688
    ## 11   11  71.019611  71.027371      1  71.023491       11_71120660
    ## 12   11  70.943058  70.969973     -1  70.956516       11_71120660
    ## 13    4  13.743436  13.893649      1  13.818542        4_13797671
    ## 14   11  70.647235  70.651421      1  70.649328       11_70661194
    ## 15   11  70.235206  70.239830     -1  70.237518       11_70196497
    ## 16   11  70.212752  70.216417      1  70.214584       11_70196497
    ## 17    4  16.122733  16.163647     -1  16.143190        4_16172555
    ## 18   11  69.820876  69.822180     -1  69.821528       11_69829176
    ## 19   11  69.655314  69.656727      1  69.656020       11_69663280
    ## 20   11  69.599736  69.605942     -1  69.602839       11_69592546
    ## 21   11  69.561758  69.580255     -1  69.571006       11_69592546
    ## 22    9   3.532354   3.897342      1   3.714848         9_3522909
    ## 23    1 135.766119 135.769136      1 135.767628       1_135798134
    ## 24   12  83.546941  83.597147     -1  83.572044       12_83594674
    ## 25    5 121.371725 121.385632     -1 121.378678       5_121381571
    ## 26   15  99.029891  99.038105      1  99.033998       15_99035604
    ## 27   11  69.340461  69.342641      1  69.341551       11_69416362
    ## 28   11  70.764209  70.772928      1  70.768568       11_70772323
    ## 29    6  65.590398  65.634040      1  65.612219        6_65644380
    ## 30   11  70.216789  70.220994     -1  70.218892       11_70196497
    ## 31   12  86.082561  86.162458      1  86.122510       12_86121929
    ## 32    X  10.715013  10.719690      1  10.717352        X_10712974
    ## 33    2  90.745370  90.781021      1  90.763196        2_90760336
    ## 34    1  88.262471  88.277633     -1  88.270052        1_88292250
    ## 35    7 127.418967 127.422068     -1 127.420518       7_127419189
    ## 36   11  69.396519  69.398234     -1  69.397376       11_69416362
    ## 37   12 102.743760 102.747561      1 102.745660      12_102744991
    ## 38   11  69.878265  69.880985     -1  69.879625       11_69876917
    ## 39   11  48.861968  48.871683     -1  48.866826       11_48841752
    ## 40   15 102.203648 102.215606      1 102.209627      15_102215134
    ## 41    1  36.264602  36.273425     -1  36.269014        1_36365564
    ## 42   11  69.901072  69.913820      1  69.907446       11_69924127
    ## 43   19   7.092014   7.105729     -1   7.098872        19_7142297
    ## 44   15  72.589621  73.061204     -1  72.825412       15_72785157
    ## 45   14  47.276931  47.296098      1  47.286514       14_47280585
    ## 46   11  16.257724  16.427310      1  16.342517       11_16300316
    ## 47   14  67.676331  67.715841     -1  67.696086       14_67686290
    ## 48    9 109.057933 109.060338     -1 109.059136       9_109050738
    ## 49   11  67.025154  67.035312      1  67.030233       11_67078257
    ## 50    1 156.035403 156.068861      1 156.052132       1_156039141
    ## 51   11  70.639122  70.642036      1  70.640579       11_70661194
    ## 52   11 102.905529 102.925124     -1 102.915326      11_102924705
    ## 53   15  86.214459  86.498503      1  86.356481       15_86351671
    ## 54    5 120.896256 120.907521     -1 120.901888       5_120845230
    ## 55   11  72.215138  72.218450      1  72.216794       11_72210521
    ## 56   15  76.643395  76.660117     -1  76.651756       15_76640377
    ## 57   11  72.796226  72.927120      1  72.861673       11_72883243
    ## 58    2 181.763332 181.827797      1 181.795564       2_181761974
    ## 59    5 130.255619 130.341563      1 130.298591       5_130342578
    ## 60   11  70.688361  70.700155     -1  70.694258       11_70661194
    ## 61   11  70.790932  70.812586     -1  70.801759       11_70785072
    ## 62   11  72.777232  72.796142     -1  72.786687       11_72771429
    ## 63   11  69.666936  69.672423     -1  69.669680       11_69663280
    ## 64   11  70.432635  70.440828      1  70.436732       11_70441181
    ## 65   17  35.439155  35.443773      1  35.441464       17_35438833
    ## 66   11  70.657202  70.662513      1  70.659858       11_70661194
    ## 67    2 122.147686 122.153083      1 122.150384       2_122156188
    ## 68    4 136.979732 137.048801     -1 137.014266       4_137016523
    ## 69   17  33.996017  34.000333     -1  33.998175       17_33920457
    ## 70   17  15.367354  15.376872     -1  15.372113       17_15378796
    ## 71    1 139.197056 139.377100     -1 139.287078       1_139284729
    ## 72   MT   0.009459   0.009806      1   0.009632              <NA>
    ## 73    X 102.284639 102.505359     -1 102.394999       X_102313977
    ## 74    Y   1.096861   1.245759     -1   1.171310              <NA>
    ## 75   11  71.092236  71.144704     -1  71.118470       11_71120660
    ## 76   11  69.622024  69.623870      1  69.622947       11_69592546
    ## 77   11  71.153102  71.230733     -1  71.191918       11_71198629
    ## 78   11  69.838514  69.839618      1  69.839066       11_69837051
    ## 79    5  88.554483  88.556083      1  88.555283        5_88657556
    ## 80    1 171.767127 171.801184      1 171.784156       1_171778290
    ## 81    9  95.637601  95.698096      1  95.667848        9_95754414
    ## 82    5 112.688876 112.896362     -1 112.792619       5_112794659
    ## 83   11  69.057718  69.060483     -1  69.059100       11_69064645
    ## 84    6  48.554796  48.570722      1  48.562759        6_48566527
    ## 85   17  35.424850  35.430055      1  35.427452       17_35438833
    ## 86    4 149.127121 149.137629     -1 149.132375       4_149124954
    ## 87    7 102.677580 102.687326     -1 102.682453       7_102538773
    ## 88    2 121.128307 121.140698     -1 121.134502       2_121139013
    ## 89   10 126.978717 126.999975      1 126.989346      10_127014380
    ## 90    1 153.905301 153.916910      1 153.911106       1_153913246
    ## 91    2  27.676440  27.762957      1  27.719698        2_27775319
    ## 92    1 131.019845 131.024974      1 131.022410       1_130914982
    ## 93    1  60.703918  60.705656      1  60.704787        1_60816518
    ## 94   11  70.909257  70.909645     -1  70.909451       11_70785072
    ## 95   11  69.401513  69.402700      1  69.402106       11_69416362
    ## 96    3  37.231333  37.232469      1  37.231901        3_37230854
    ## 97   11  69.448548  69.453676      1  69.451112       11_69484146
    ## 98   11  70.525367  70.527858      1  70.526612       11_70609170
    ## 99   11  71.007164  71.019841     -1  71.013502       11_71120660
    ## 100  11  70.977836  70.983026     -1  70.980431       11_71120660
    ## 101  11  70.970213  70.977833      1  70.974023       11_71120660
    ## 102  11  71.031941  71.033513     -1  71.032727       11_71120660
    ## 103  11  69.981156  69.990601      1  69.985878       11_69978753
    ## 104  11  69.968222  69.982517     -1  69.975370       11_69978753
    ## 105  11  69.942539  69.948188     -1  69.945364       11_69924127
    ## 106  11  70.611212  70.613211     -1  70.612212       11_70612702
    ## 107  11  70.654741  70.656493     -1  70.655617       11_70661194
    ## 108  11  69.682846  69.685784     -1  69.684315       11_69663280
    ## 109  11  69.803603  69.805624     -1  69.804614       11_69810287
    ## 110  11  69.897352  69.900987     -1  69.899170       11_69876917
    ## 111  11  68.968131  68.974365      1  68.971248       11_68987403
    ## 112  11  69.765912  69.784023      1  69.774968       11_69768364
    ## 113  11  69.682577  69.696098     -1  69.689338       11_69663280
    ## 114  11  69.656697  69.662642     -1  69.659670       11_69663280
    ## 115  11  69.632990  69.653297      1  69.643144       11_69663280
    ## 116  11  70.451919  70.453727      1  70.452823       11_70441181
    ## 117  11  83.511736  83.522175     -1  83.516956       11_83554638
    ## 118   2  31.152316  31.219548      1  31.185932        2_31262542
    ## 119  11  70.241457  70.255353     -1  70.248405       11_70196497
    ## 120  12   3.752543   3.762146     -1   3.757344        12_3798393
    ## 121  11  70.238124  70.239842     -1  70.238983       11_70196497
    ## 122  11   6.467599   6.476076     -1   6.471838        11_6462540
    ## 123  11  69.876684  69.878018     -1  69.877351       11_69876917
    ## 124  11  69.682577  69.696098     -1  69.689338       11_69663280
    ## 125  11  69.120054  69.122589     -1  69.121322       11_69131341
    ## 126  16  84.713023  84.715242      1  84.714132       16_84722155
    ## 127  17  25.985815  25.988469      1  25.987142       17_25934908
    ## 128   5 100.420842 100.429516     -1 100.425179       5_100397159
    ## 129   1   7.349406   7.397801     -1   7.373604         1_7511318
    ## 130  10  51.677756  51.730429      1  51.704092       10_51680496
    ## 131  12  36.091846  36.156825     -1  36.124336       12_36115728
    ## 132  12  32.820335  32.853369      1  32.836852       12_32944094
    ## 133  11  72.607283  72.686481      1  72.646882       11_72642242
    ## 134  11  72.344722  72.362442     -1  72.353582       11_72366456
    ## 135  11  72.211724  72.215592     -1  72.213658       11_72210521
    ## 136  11  72.207547  72.210015      1  72.208781       11_72210521
    ## 137  11  72.042032  72.047370      1  72.044701       11_72040688
    ## 138  11  71.749920  71.789647      1  71.769784       11_71782692
    ## 139  11  70.844778  70.943105      1  70.893942       11_70785072
    ## 140  11  70.562881  70.614483      1  70.588682       11_70609170
    ## 141  11  70.540064  70.558110      1  70.549087       11_70609170
    ## 142  11  70.054085  70.057894      1  70.055990       11_70044434
    ## 143  11  69.183932  69.197843     -1  69.190888       11_69198036
    ## 144  11  69.125896  69.149115      1  69.137506       11_69131341
    ## 145  11  69.088490  69.092384      1  69.090437       11_69064645
    ## 146  11  69.070806  69.073293      1  69.072050       11_69064645
    ## 147  11  69.045647  69.051664      1  69.048656       11_69064645
    ## 148  11  67.924806  67.965651     -1  67.945228       11_67946874
    ## 149  12  55.489411  55.492647     -1  55.491029       12_55544744
    ## 150  12  99.717531  99.883442     -1  99.800486       12_99780940
    ## 151  12 100.779057 100.896981      1 100.838019      12_100809117
    ## 152  11  77.515121  77.538607      1  77.526864       11_77539214
    ## 153  11  83.647844  83.649355     -1  83.648600       11_83647144
    ## 154   6  86.647044  86.669159     -1  86.658102        6_86655127
    ## 155  14  55.578123  55.581529      1  55.579826       14_55574243
    ## 156  15  58.094047  58.135082     -1  58.114564       15_58114543
    ## 157  15  85.859707  85.876572      1  85.868140       15_85744532
    ## 158  15  78.523346  78.529625     -1  78.526486       15_78483573
    ## 159  15  83.483772  83.510893     -1  83.497332       15_83568143
    ## 160  16  52.031549  52.208047      1  52.119798       16_52098764
    ## 161  15  99.620496  99.651656     -1  99.636076       15_99636295
    ## 162  15 100.387898 100.425072     -1 100.406485      15_100424363
    ## 163   2 180.725263 180.742280      1 180.733772       2_180745048
    ## 164  11  69.851005  69.858730     -1  69.854868       11_69844926
    ## 165   7 101.663633 101.790168      1 101.726900       7_101719131
    ## 166   X 151.046150 151.089521      1 151.067836       X_151054214
    ## 167   7 139.995955 140.036350     -1 140.016152       7_140027791
    ## 168   9  37.613847  37.648318      1  37.631082        9_37683669
    ## 169   1  86.526736  86.530698      1  86.528717        1_86642528
    ## 170   1 153.899944 153.909723      1 153.904834       1_153913246
    ## 171   1 153.122756 153.186447     -1 153.154602       1_153186664
    ## 172   1 170.141938 170.174957     -1 170.158448       1_170300796
    ## 173   2  24.336853  24.351494      1  24.344174        2_24338978
    ## 174   2 119.618298 119.651244      1 119.634771       2_119619480
    ## 175   2 155.819832 155.826925     -1 155.823378       2_155776844
    ## 176   3  32.949408  33.143247     -1  33.046328        3_33098584
    ## 177   3  27.182965  27.284608      1  27.233786        3_27238815
    ## 178   2 145.902099 145.916425      1 145.909262       2_145906957
    ## 179   3 144.188530 144.205220     -1 144.196875       3_144248421
    ## 180   4  52.439243  52.488260      1  52.463752        4_52484329
    ## 181   4 122.946850 122.961188     -1 122.954019       4_122944932
    ## 182   4 129.058271 129.085886      1 129.072078       4_129047148
    ## 183   4 141.239499 141.257564      1 141.248532       4_141243055
    ## 184   5  29.195992  29.225524      1  29.210758        5_29213824
    ## 185   6   5.725639   6.028039      1   5.876839         6_5878061
    ## 186   6  49.243921  49.264052     -1  49.253986        6_49275304
    ## 187   3  87.525407  87.540156      1  87.532782        3_87524447
    ## 188   6  88.835915  88.841935     -1  88.838925        6_88848755
    ## 189   6 118.151748 118.197744     -1 118.174746       6_118192991
    ## 190   7 118.118891 118.129625     -1 118.124258       7_118237771
    ## 191   7 122.159437 122.169875      1 122.164656       7_122188931
    ## 192   X  52.053021  52.165252     -1  52.109136        X_52082200
    ## 193   X 163.935443 163.958661     -1 163.947052       X_163990072
    ## 194   8  94.172618  94.173567      1  94.173092        8_94171846
    ## 195   8  94.179089  94.180325      1  94.179707        8_94171846
    ## 196   9  50.662752  50.739517     -1  50.701134        9_50740446
    ## 197   9  92.250057  92.271975      1  92.261016        9_92257673
    ## 198   2 103.957986 103.981878      1 103.969932       2_103951663
    ## 199  11  69.156989  69.169792      1  69.163390       11_69131341
    ## 200   1  93.512079  93.621976      1  93.567028        1_93501778
    ## 201   6  91.156665  91.174692      1  91.165678        6_91170456
    ## 202  16  35.832874  35.871544     -1  35.852209       16_35793887
    ## 203   X  97.333840  97.377216     -1  97.355528        X_97373574
    ## 204  14  65.358676  65.398930      1  65.378803       14_65362324
    ## 205  11  69.733997  69.758637     -1  69.746317       11_69733033
    ## 206  11  69.673115  69.682084     -1  69.677600       11_69663280
    ## 207  11  69.420809  69.549110     -1  69.484960       11_69484146
    ## 208   X  93.654863  93.749951      1  93.702407        X_93702485
    ## 209   5 100.679484 100.719716     -1 100.699600       5_100598593
    ## 210   5  90.214724  90.224001     -1  90.219362        5_90211015
    ## 211  17  35.379617  35.385290      1  35.382454       17_35369431
    ## 212  15  85.832306  85.858822      1  85.845564       15_85744532
    ## 213   1 127.954184 128.103047     -1 128.028616       1_127864215
    ## 214  18  34.442352  34.506823     -1  34.474588       18_34463724
    ## 215   5  34.173887  34.187720     -1  34.180804        5_34211315
    ## 216   8 124.671335 124.721983     -1 124.696659       8_124505654
    ## 217   1 191.906767 191.913247      1 191.910007       1_191909192
    ## 218   5  67.618140  67.847432     -1  67.732786        5_67728801
    ##                  biotype         module hotspot
    ## 1         protein_coding         purple   chr11
    ## 2         protein_coding           grey    <NA>
    ## 3         protein_coding         violet   chr11
    ## 4         protein_coding          brown   chr11
    ## 5         protein_coding         yellow   chr11
    ## 6         protein_coding          brown   chr11
    ## 7         protein_coding          brown   chr11
    ## 8         protein_coding        magenta   chr11
    ## 9         protein_coding        magenta    <NA>
    ## 10        protein_coding         purple    <NA>
    ## 11        protein_coding      turquoise    <NA>
    ## 12        protein_coding      turquoise    <NA>
    ## 13        protein_coding         yellow   chr11
    ## 14        protein_coding         yellow    <NA>
    ## 15        protein_coding           blue    <NA>
    ## 16        protein_coding      lightcyan    <NA>
    ## 17        protein_coding        magenta   chr11
    ## 18        protein_coding        darkred    <NA>
    ## 19        protein_coding           grey    <NA>
    ## 20        protein_coding         purple    <NA>
    ## 21        protein_coding            tan    <NA>
    ## 22        protein_coding         yellow   chr11
    ## 23        protein_coding          brown   chr11
    ## 24        protein_coding          brown   chr11
    ## 25        protein_coding        magenta   chr11
    ## 26        protein_coding       skyblue3   chr11
    ## 27  processed_transcript          green    <NA>
    ## 28        protein_coding      turquoise    <NA>
    ## 29        protein_coding       skyblue3   chr11
    ## 30        protein_coding           grey    <NA>
    ## 31        protein_coding      darkgreen   chr11
    ## 32        protein_coding           blue   chr11
    ## 33        protein_coding        magenta   chr11
    ## 34        protein_coding         yellow   chr11
    ## 35        protein_coding         yellow   chr11
    ## 36        protein_coding     orangered4    <NA>
    ## 37        protein_coding      darkgreen   chr11
    ## 38        protein_coding         yellow    <NA>
    ## 39        protein_coding        magenta    <NA>
    ## 40        protein_coding         yellow   chr11
    ## 41        protein_coding       skyblue3   chr11
    ## 42        protein_coding         purple    <NA>
    ## 43        protein_coding         grey60   chr11
    ## 44        protein_coding          black   chr11
    ## 45        protein_coding      darkgreen   chr11
    ## 46        protein_coding          brown    <NA>
    ## 47        protein_coding         violet   chr11
    ## 48        protein_coding        magenta   chr11
    ## 49        protein_coding            tan    <NA>
    ## 50        protein_coding        magenta   chr11
    ## 51        protein_coding           pink    <NA>
    ## 52        protein_coding         violet    <NA>
    ## 53        protein_coding          brown   chr11
    ## 54        protein_coding        magenta   chr11
    ## 55  processed_transcript           grey    <NA>
    ## 56        protein_coding         grey60   chr11
    ## 57        protein_coding    saddlebrown    <NA>
    ## 58        protein_coding            tan   chr11
    ## 59        protein_coding          brown   chr11
    ## 60        protein_coding           grey    <NA>
    ## 61        protein_coding     darkorange    <NA>
    ## 62        protein_coding          brown    <NA>
    ## 63        protein_coding      turquoise    <NA>
    ## 64        protein_coding         grey60    <NA>
    ## 65        protein_coding       skyblue3   chr11
    ## 66        protein_coding         yellow    <NA>
    ## 67        protein_coding        magenta   chr11
    ## 68        protein_coding         grey60   chr11
    ## 69        protein_coding        magenta   chr11
    ## 70        protein_coding       skyblue3   chr11
    ## 71        protein_coding          brown   chr11
    ## 72        protein_coding      turquoise    <NA>
    ## 73        protein_coding            tan   chr11
    ## 74        protein_coding         yellow    <NA>
    ## 75        protein_coding           grey    <NA>
    ## 76        protein_coding          brown    <NA>
    ## 77        protein_coding           grey    <NA>
    ## 78        protein_coding           blue    <NA>
    ## 79        protein_coding      darkgreen   chr11
    ## 80        protein_coding        magenta   chr11
    ## 81        protein_coding            tan   chr11
    ## 82        protein_coding      royalblue   chr11
    ## 83        protein_coding           grey    <NA>
    ## 84        protein_coding         yellow   chr11
    ## 85        protein_coding        magenta   chr11
    ## 86        protein_coding           grey   chr11
    ## 87        protein_coding          brown   chr11
    ## 88        protein_coding            red   chr11
    ## 89        protein_coding         yellow   chr11
    ## 90        protein_coding     lightgreen    <NA>
    ## 91        protein_coding         yellow   chr11
    ## 92        protein_coding       skyblue3   chr11
    ## 93            pseudogene      turquoise    <NA>
    ## 94            pseudogene           grey    <NA>
    ## 95             antisense     lightgreen    <NA>
    ## 96             antisense           grey   chr11
    ## 97             antisense           grey    <NA>
    ## 98        protein_coding      turquoise    <NA>
    ## 99        protein_coding          green    <NA>
    ## 100       protein_coding        darkred    <NA>
    ## 101       protein_coding           grey    <NA>
    ## 102              lincRNA           grey    <NA>
    ## 103       protein_coding      royalblue    <NA>
    ## 104       protein_coding       skyblue3    <NA>
    ## 105       protein_coding           grey    <NA>
    ## 106            antisense           grey    <NA>
    ## 107              lincRNA           blue    <NA>
    ## 108       protein_coding           grey    <NA>
    ## 109       protein_coding     orangered4    <NA>
    ## 110       protein_coding         yellow    <NA>
    ## 111       protein_coding         grey60    <NA>
    ## 112       protein_coding         yellow    <NA>
    ## 113       protein_coding           grey    <NA>
    ## 114       protein_coding            red    <NA>
    ## 115       protein_coding         purple    <NA>
    ## 116       protein_coding           blue    <NA>
    ## 117       protein_coding        magenta    <NA>
    ## 118       protein_coding         yellow   chr11
    ## 119       protein_coding        magenta    <NA>
    ## 120            antisense          brown   chr11
    ## 121       protein_coding         yellow    <NA>
    ## 122       protein_coding           pink    <NA>
    ## 123       protein_coding           grey    <NA>
    ## 124       protein_coding         yellow    <NA>
    ## 125       protein_coding           grey    <NA>
    ## 126              lincRNA       skyblue3   chr11
    ## 127              lincRNA           grey   chr11
    ## 128              lincRNA      turquoise   chr11
    ## 129              lincRNA           grey   chr11
    ## 130       protein_coding            tan   chr11
    ## 131       protein_coding      darkgreen   chr11
    ## 132       protein_coding        magenta   chr11
    ## 133       protein_coding          brown    <NA>
    ## 134       protein_coding           grey    <NA>
    ## 135       protein_coding      turquoise    <NA>
    ## 136       protein_coding         salmon    <NA>
    ## 137       protein_coding         violet    <NA>
    ## 138       protein_coding           grey    chr5
    ## 139       protein_coding           pink    <NA>
    ## 140       protein_coding          black    <NA>
    ## 141       protein_coding           grey    <NA>
    ## 142       protein_coding           grey    <NA>
    ## 143       protein_coding           grey    <NA>
    ## 144       protein_coding            red    <NA>
    ## 145       protein_coding         purple    <NA>
    ## 146       protein_coding           blue    <NA>
    ## 147       protein_coding         violet    <NA>
    ## 148       protein_coding      turquoise    <NA>
    ## 149       protein_coding       skyblue3   chr11
    ## 150       protein_coding           grey   chr11
    ## 151       protein_coding          brown   chr11
    ## 152       protein_coding         grey60    <NA>
    ## 153       protein_coding       skyblue3    <NA>
    ## 154       protein_coding       skyblue3   chr11
    ## 155       protein_coding        magenta   chr11
    ## 156       protein_coding         violet   chr11
    ## 157       protein_coding         violet   chr11
    ## 158       protein_coding         yellow   chr11
    ## 159       protein_coding            tan   chr11
    ## 160       protein_coding  darkturquoise   chr11
    ## 161       protein_coding         violet   chr11
    ## 162       protein_coding       skyblue3   chr11
    ## 163       protein_coding          brown   chr11
    ## 164       protein_coding         purple    <NA>
    ## 165       protein_coding         yellow   chr11
    ## 166       protein_coding         grey60   chr11
    ## 167       protein_coding         yellow   chr11
    ## 168       protein_coding            tan   chr11
    ## 169       protein_coding      lightcyan   chr11
    ## 170       protein_coding     lightgreen   chr11
    ## 171       protein_coding       skyblue3   chr11
    ## 172       protein_coding      turquoise   chr11
    ## 173       protein_coding     darkorange   chr11
    ## 174       protein_coding         violet   chr11
    ## 175       protein_coding       skyblue3   chr11
    ## 176       protein_coding          brown   chr11
    ## 177       protein_coding           grey   chr11
    ## 178       protein_coding      turquoise   chr11
    ## 179       protein_coding    greenyellow   chr11
    ## 180       protein_coding         violet   chr11
    ## 181       protein_coding       skyblue3   chr11
    ## 182       protein_coding        magenta   chr11
    ## 183       protein_coding          brown   chr11
    ## 184       protein_coding          brown   chr11
    ## 185       protein_coding          brown   chr11
    ## 186       protein_coding      darkgreen   chr11
    ## 187       protein_coding darkolivegreen   chr11
    ## 188       protein_coding          brown   chr11
    ## 189       protein_coding         grey60   chr11
    ## 190       protein_coding      turquoise   chr11
    ## 191       protein_coding         violet   chr11
    ## 192       protein_coding         yellow   chr11
    ## 193       protein_coding      darkgreen   chr11
    ## 194       protein_coding            red   chr11
    ## 195       protein_coding            red   chr11
    ## 196       protein_coding          brown   chr11
    ## 197       protein_coding      royalblue   chr13
    ## 198       protein_coding           grey   chr11
    ## 199       protein_coding       skyblue3    <NA>
    ## 200       protein_coding         yellow   chr11
    ## 201       protein_coding          brown   chr11
    ## 202       protein_coding        magenta   chr11
    ## 203       protein_coding           grey   chr11
    ## 204       protein_coding         grey60   chr11
    ## 205       protein_coding  darkturquoise    <NA>
    ## 206       protein_coding        darkred    <NA>
    ## 207       protein_coding         purple    <NA>
    ## 208       protein_coding          brown   chr11
    ## 209       protein_coding      turquoise   chr11
    ## 210       protein_coding           grey   chr11
    ## 211       protein_coding        magenta   chr11
    ## 212       protein_coding          brown   chr11
    ## 213       protein_coding            tan   chr11
    ## 214       protein_coding        skyblue   chr11
    ## 215       protein_coding          brown   chr11
    ## 216       protein_coding            tan   chr11
    ## 217       protein_coding            red   chr11
    ## 218       protein_coding           grey   chr11

The above tibble has local and nonlocal traits.

``` r
local_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>%
  mutate(local_tx = chr ==hot_chrom) %>%
  filter(local_tx)
```

``` r
nonlocal_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>% #genomewide threshold
  mutate(local_tx = chr == hot_chrom) %>%
  filter(!local_tx) %>% 
  filter(!is.na(hotspot))
```

``` r
dim(nonlocal_annot)
```

    ## [1] 126  16

We’ll use only those nonlocal traits that Keller et al. annotated as
mapping to a hotspot.

We also want to look at the local traits for Chr 11 hotspot. We have 88
local traits. We want to identify a subset of those for two-dimensional
scans. Let’s first limit to those near the hotspot, ie, within 5Mb of
either end.

``` r
local_annot2 <- local_annot %>%
  filter(middle <= hot_upper + 5, middle >= hot_lower - 5) %>%
  arrange(desc(lod))
```

Recall that *Pdx1* is the gene that Keller et al. identified as the key
mediator for Chr 5 hotspot. Its LOD is only 15, which seems to indicate
that I should be looking at a broad set of local traits, not just those,
eg., with LOD \> 40\!

Also, I’m limiting consideration of local genes to those with middle
within 5Mb of the hotspot.

Now, I make the two expression trait matrices that I need for use with
Condor.

``` r
hot_local <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% local_annot2$annot.id]
hot_nonlocal <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% nonlocal_annot$annot.id]
saveRDS(hot_local, "../data/Chr11hot_local.rds") # change this
saveRDS(hot_nonlocal, "../data/Chr11hot_nonlocal.rds") # change this
saveRDS(K$`11`, "../data/Chr11_kinship.rds") # change this - both
saveRDS(genoprobs$`11`, "../data/Chr11_aprobs.rds") # change this - both
saveRDS(map, "../data/map.rds")
```

Lastly, we need to identify the scan region for Chr 11.

We consider the physical
map.

``` r
(start_index <- which(map$`11`== map$`11`[map$`11` > hot_lower][1]) - 50) # change this - all 3!
```

    ## 11_69064645 
    ##        1868

``` r
(stop_index <- which(map$`11`== map$`11`[map$`11` > hot_upper][1]) + 50)
```

    ## 11_73002626 
    ##        2084

``` r
# change this - all 3!
```

``` r
(nsnp <- stop_index - start_index + 1)
```

    ## 11_73002626 
    ##         217

``` r
dim(local_annot2)
```

    ## [1] 82 16

## References

Keller et al. 2018. Genetics. Genetic drivers of pancreatic islet
function.

## Session info

``` r
devtools::session_info()
```

    ## ─ Session info ──────────────────────────────────────────────────────────
    ##  setting  value                       
    ##  version  R version 3.5.3 (2019-03-11)
    ##  os       macOS Mojave 10.14.5        
    ##  system   x86_64, darwin15.6.0        
    ##  ui       X11                         
    ##  language (EN)                        
    ##  collate  en_US.UTF-8                 
    ##  ctype    en_US.UTF-8                 
    ##  tz       America/Chicago             
    ##  date     2019-06-02                  
    ## 
    ## ─ Packages ──────────────────────────────────────────────────────────────
    ##  package     * version date       lib source                        
    ##  assertthat    0.2.1   2019-03-21 [1] CRAN (R 3.5.2)                
    ##  backports     1.1.4   2019-04-10 [1] CRAN (R 3.5.2)                
    ##  bit           1.1-14  2018-05-29 [1] CRAN (R 3.5.0)                
    ##  bit64         0.9-7   2017-05-08 [1] CRAN (R 3.5.0)                
    ##  blob          1.1.1   2018-03-25 [1] CRAN (R 3.5.0)                
    ##  broom         0.5.1   2018-12-05 [1] CRAN (R 3.5.0)                
    ##  callr         3.2.0   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  cellranger    1.1.0   2016-07-27 [1] CRAN (R 3.5.0)                
    ##  cli           1.1.0   2019-03-19 [1] CRAN (R 3.5.3)                
    ##  colorspace    1.4-1   2019-03-18 [1] CRAN (R 3.5.3)                
    ##  crayon        1.3.4   2017-09-16 [1] CRAN (R 3.5.0)                
    ##  data.table    1.12.0  2019-01-13 [1] CRAN (R 3.5.2)                
    ##  DBI           1.0.0   2018-05-02 [1] CRAN (R 3.5.0)                
    ##  desc          1.2.0   2018-05-01 [1] CRAN (R 3.5.0)                
    ##  devtools      2.0.2   2019-04-08 [1] CRAN (R 3.5.2)                
    ##  digest        0.6.19  2019-05-20 [1] CRAN (R 3.5.2)                
    ##  dplyr       * 0.8.0.1 2019-02-15 [1] CRAN (R 3.5.2)                
    ##  evaluate      0.14    2019-05-28 [1] CRAN (R 3.5.3)                
    ##  forcats     * 0.4.0   2019-02-17 [1] CRAN (R 3.5.2)                
    ##  fs            1.3.1   2019-05-06 [1] CRAN (R 3.5.2)                
    ##  generics      0.0.2   2018-11-29 [1] CRAN (R 3.5.0)                
    ##  ggplot2     * 3.1.1   2019-04-07 [1] CRAN (R 3.5.3)                
    ##  glue          1.3.1   2019-03-12 [1] CRAN (R 3.5.2)                
    ##  gtable        0.3.0   2019-03-25 [1] CRAN (R 3.5.2)                
    ##  haven         2.1.0   2019-02-19 [1] CRAN (R 3.5.2)                
    ##  hms           0.4.2   2018-03-10 [1] CRAN (R 3.5.0)                
    ##  htmltools     0.3.6   2017-04-28 [1] CRAN (R 3.5.0)                
    ##  httr          1.4.0   2018-12-11 [1] CRAN (R 3.5.0)                
    ##  jsonlite      1.6     2018-12-07 [1] CRAN (R 3.5.0)                
    ##  knitr         1.23    2019-05-18 [1] CRAN (R 3.5.2)                
    ##  lattice       0.20-38 2018-11-04 [1] CRAN (R 3.5.3)                
    ##  lazyeval      0.2.2   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  lubridate     1.7.4   2018-04-11 [1] CRAN (R 3.5.0)                
    ##  magrittr      1.5     2014-11-22 [1] CRAN (R 3.5.0)                
    ##  memoise       1.1.0   2017-04-21 [1] CRAN (R 3.5.0)                
    ##  modelr        0.1.4   2019-02-18 [1] CRAN (R 3.5.2)                
    ##  munsell       0.5.0   2018-06-12 [1] CRAN (R 3.5.0)                
    ##  nlme          3.1-137 2018-04-07 [1] CRAN (R 3.5.3)                
    ##  pillar        1.3.1   2018-12-15 [1] CRAN (R 3.5.0)                
    ##  pkgbuild      1.0.3   2019-03-20 [1] CRAN (R 3.5.2)                
    ##  pkgconfig     2.0.2   2018-08-16 [1] CRAN (R 3.5.0)                
    ##  pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.5.0)                
    ##  plyr          1.8.4   2016-06-08 [1] CRAN (R 3.5.0)                
    ##  prettyunits   1.0.2   2015-07-13 [1] CRAN (R 3.5.0)                
    ##  processx      3.3.1   2019-05-08 [1] CRAN (R 3.5.2)                
    ##  ps            1.3.0   2018-12-21 [1] CRAN (R 3.5.0)                
    ##  purrr       * 0.3.2   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  qtl2        * 0.18    2019-02-08 [1] local                         
    ##  R6            2.4.0   2019-02-14 [1] CRAN (R 3.5.2)                
    ##  Rcpp          1.0.1.3 2019-05-29 [1] Github (RcppCore/Rcpp@6062d56)
    ##  readr       * 1.3.1   2018-12-21 [1] CRAN (R 3.5.0)                
    ##  readxl        1.3.0   2019-02-15 [1] CRAN (R 3.5.2)                
    ##  remotes       2.0.4   2019-04-10 [1] CRAN (R 3.5.2)                
    ##  rlang         0.3.4   2019-04-07 [1] CRAN (R 3.5.3)                
    ##  rmarkdown     1.13    2019-05-22 [1] CRAN (R 3.5.2)                
    ##  rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.5.0)                
    ##  RSQLite       2.1.1   2018-05-06 [1] CRAN (R 3.5.0)                
    ##  rstudioapi    0.10    2019-03-19 [1] CRAN (R 3.5.2)                
    ##  rvest         0.3.2   2016-06-17 [1] CRAN (R 3.5.0)                
    ##  scales        1.0.0   2018-08-09 [1] CRAN (R 3.5.0)                
    ##  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.5.0)                
    ##  stringi       1.4.3   2019-03-12 [1] CRAN (R 3.5.2)                
    ##  stringr     * 1.4.0   2019-02-10 [1] CRAN (R 3.5.2)                
    ##  testthat      2.1.1   2019-04-23 [1] CRAN (R 3.5.3)                
    ##  tibble      * 2.1.1   2019-03-16 [1] CRAN (R 3.5.3)                
    ##  tidyr       * 0.8.3   2019-03-01 [1] CRAN (R 3.5.2)                
    ##  tidyselect    0.2.5   2018-10-11 [1] CRAN (R 3.5.0)                
    ##  tidyverse   * 1.2.1   2017-11-14 [1] CRAN (R 3.5.0)                
    ##  usethis       1.5.0   2019-04-07 [1] CRAN (R 3.5.2)                
    ##  withr         2.1.2   2018-03-15 [1] CRAN (R 3.5.0)                
    ##  xfun          0.7     2019-05-14 [1] CRAN (R 3.5.2)                
    ##  xml2          1.2.0   2018-01-24 [1] CRAN (R 3.5.0)                
    ##  yaml          2.2.0   2018-07-25 [1] CRAN (R 3.5.0)                
    ## 
    ## [1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library
