Setting up for use of Condor CHTC with Chr 7 hotspot
================
Frederick Boehm
05/31/2019

## Goals

Goal here is to document the setup & organization of files that I need
for using Condor CHTC to do my two-dimensional scans as part of the
analyses of the four Keller hotspots. Note that they reported 5
hotspots. I previously performed the analyses for Chromosome 2 hotspot.

## Setting

Recall that I have downloaded a Keller 2018 .RData file from DataDryad
that contains many of their results. The .RData file,
“data/Attie\_DO378\_eQTL\_viewer\_v1.Rdata” annotates 139 traits as
belonging to the chromosome 2 hotspot.

Following Karl’s suggestion, I want to look at not only the local trait
that Keller identifies as causal for the hotspot, but also several other
local-ish traits that have reasonable univariate LOD
scores.

## Preparing RDS files for Chr 7 analysis

``` r
load("~/Box Sync/attie/keller2018-chr2-hotspot-chtc/data-to-ignore/Attie_DO378_eQTL_viewer_v1.Rdata")
ls()
```

    ## [1] "dataset.clinical.phenotypes" "dataset.islet.hotspots"     
    ## [3] "dataset.islet.modules"       "dataset.islet.rnaseq"       
    ## [5] "ensembl.version"             "genoprobs"                  
    ## [7] "K"                           "map"                        
    ## [9] "markers"

``` r
library(tidyverse)
```

    ## ── Attaching packages ───────────────────────────────────────── tidyverse 1.2.1 ──

    ## ✔ ggplot2 3.1.1       ✔ purrr   0.3.2  
    ## ✔ tibble  2.1.1       ✔ dplyr   0.8.0.1
    ## ✔ tidyr   0.8.3       ✔ stringr 1.4.0  
    ## ✔ readr   1.3.1       ✔ forcats 0.4.0

    ## ── Conflicts ──────────────────────────────────────────── tidyverse_conflicts() ──
    ## ✖ dplyr::filter() masks stats::filter()
    ## ✖ dplyr::lag()    masks stats::lag()

``` r
dataset.islet.rnaseq$expr %>%
  dim
```

    ## [1]   378 21771

``` r
colnames(dataset.islet.rnaseq$expr) %>%
  head
```

    ## [1] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000037"
    ## [4] "ENSMUSG00000000049" "ENSMUSG00000000056" "ENSMUSG00000000058"

We need to get the names of the nonlocal traits that map to Chromosome 7
hotspot.

``` r
library(qtl2)
```

    ## 
    ## Attaching package: 'qtl2'

    ## The following object is masked from 'package:readr':
    ## 
    ##     read_csv

I forgot that the LOD peaks for all traits are saved in
`dataset.islet.rnaseq$lod.peaks`. It has nearly 40,000 peaks recorded. I
really just need to look at it. That is, for now, I don’t really need
the output of `scan1()`.

We want the ensembl ids for the traits that map to Chr 5 hotspot. First,
define the boundaries of the hotspot. The supplemental table in Keller
et al. 2018 lists only the midpoint of the hotspot. I believe that the
half-width is 2 Mb. Yes, this is indeed the half-width per the Rmd file
that Dan Gatti prepared.

``` r
hot_chrom <- 7
hot_center <- 46# for chr7
hot_lower <- hot_center - 2
hot_upper <- hot_center + 2
```

Now, filter to get those traits that map to the hotspot. Note that we’re
likely to get local traits, too, in this step.

``` r
dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18)
```

    ##               annot.id  marker.id chrom      pos        lod        symbol
    ## 1   ENSMUSG00000038079 7_47021298     7 47.02130   8.664238       Tmem237
    ## 2   ENSMUSG00000038132 7_46358463     7 46.35846   8.773876         Rbm24
    ## 3   ENSMUSG00000038239 7_45270542     7 45.27054   9.328109           Hrc
    ## 4   ENSMUSG00000038292 7_45270542     7 45.27054  22.993516       Ccdc155
    ## 5   ENSMUSG00000038387 7_44838742     7 44.83874  29.296385          Rras
    ## 6   ENSMUSG00000038520 7_44819852     7 44.81985  10.982534       Tbc1d17
    ## 7   ENSMUSG00000038630 7_44780832     7 44.78083   8.157070      Zkscan16
    ## 8   ENSMUSG00000038644 7_44551043     7 44.55104  17.462611         Pold1
    ## 9   ENSMUSG00000038738 7_44445619     7 44.44562  18.873961        Shank1
    ## 10  ENSMUSG00000039114 7_45462287     7 45.46229   7.689855          Nrn1
    ## 11  ENSMUSG00000039501 7_44896497     7 44.89650   8.220185         Znfx1
    ## 12  ENSMUSG00000039911 7_45499645     7 45.49965  13.535817         Spsb1
    ## 13  ENSMUSG00000040033 7_46297732     7 46.29773   7.466062         Stat2
    ## 14  ENSMUSG00000040046 7_46557864     7 46.55786  86.804339          Tph1
    ## 15  ENSMUSG00000040136 7_46000542     7 46.00054  64.580925         Abcc8
    ## 16  ENSMUSG00000040189 7_46000542     7 46.00054  40.701471       Ccdc114
    ## 17  ENSMUSG00000040212 7_46000542     7 46.00054  21.331489          Emp3
    ## 18  ENSMUSG00000040364 7_44896497     7 44.89650   8.347217          Sec1
    ## 19  ENSMUSG00000040428 7_45622192     7 45.62219 122.235876       Plekha4
    ## 20  ENSMUSG00000006418 7_45689416     7 45.68942   7.832925        Rnf114
    ## 21  ENSMUSG00000006763 7_46610522     7 46.61052 133.567597         Saal1
    ## 22  ENSMUSG00000041598 7_46000542     7 46.00054   8.735513      Cdc42ep4
    ## 23  ENSMUSG00000041827 7_45455539     7 45.45554   9.567588         Oasl1
    ## 24  ENSMUSG00000041912 7_46326762     7 46.32676   9.165411         Tdrkh
    ## 25  ENSMUSG00000042557 7_45499645     7 45.49965   7.244968         Sin3a
    ## 26  ENSMUSG00000042812 7_45499645     7 45.49965   8.661229         Foxf1
    ## 27  ENSMUSG00000042918 7_45622192     7 45.62219   7.225122        Mamstr
    ## 28  ENSMUSG00000043065 7_45622192     7 45.62219   7.374143        Spice1
    ## 29  ENSMUSG00000043099 7_47160161     7 47.16016   7.402522          Hic1
    ## 30  ENSMUSG00000043262 7_47035721     7 47.03572  52.834830         Uevld
    ## 31  ENSMUSG00000043858 7_44632571     7 44.63257   8.301696         Nup62
    ## 32  ENSMUSG00000043953 7_46297732     7 46.29773   7.180139         Ccrl2
    ## 33  ENSMUSG00000044562 7_45334220     7 45.33422  14.692374        Rasip1
    ## 34  ENSMUSG00000007783 7_44940291     7 44.94029  67.586286         Cpt1c
    ## 35  ENSMUSG00000007833 7_45165000     7 45.16500  95.442630      Aldh16a1
    ## 36  ENSMUSG00000008461 7_45206863     7 45.20686  20.607460          Fut1
    ## 37  ENSMUSG00000046718 7_45689416     7 45.68942   7.823869          Bst2
    ## 38  ENSMUSG00000047098 7_46315301     7 46.31530   8.038777         Rnf31
    ## 39  ENSMUSG00000048279 7_46000542     7 46.00054   7.862093          Sacs
    ## 40  ENSMUSG00000048304 7_46000542     7 46.00054  12.366305       Slitrk3
    ## 41  ENSMUSG00000049502 7_44838742     7 44.83874   7.691770         Dtx3l
    ## 42  ENSMUSG00000049661 7_45270542     7 45.27054  19.827636         Mtag2
    ## 43  ENSMUSG00000009487 7_46280163     7 46.28016  40.817119          Otog
    ## 44  ENSMUSG00000050708 7_45622192     7 45.62219   8.939663          Ftl1
    ## 45  ENSMUSG00000050908 7_44364954     7 44.36495   7.189574        Tvp23a
    ## 46  ENSMUSG00000051113 7_44551043     7 44.55104  29.921393       Fam71e1
    ## 47  ENSMUSG00000052005 7_45334220     7 45.33422   8.011056        Gm9864
    ## 48  ENSMUSG00000052374 7_46358463     7 46.35846   7.575153         Actn2
    ## 49  ENSMUSG00000054161 7_45705169     7 45.70517  43.249414        Fam83e
    ## 50  ENSMUSG00000054720 7_44729808     7 44.72981   9.214808        Lrrc8c
    ## 51  ENSMUSG00000010307 7_46828517     7 46.82852  16.579693       Tmem86a
    ## 52  ENSMUSG00000011382 7_45720922     7 45.72092  75.782155          Dhdh
    ## 53  ENSMUSG00000011658 7_44874623     7 44.87462  62.001388           Fuz
    ## 54  ENSMUSG00000057342 7_45705169     7 45.70517   9.627080         Sphk2
    ## 55  ENSMUSG00000058542 7_45455539     7 45.45554  31.773908       Gm15590
    ## 56  ENSMUSG00000058743 7_45434499     7 45.43450  39.525025        Kcnj14
    ## 57  ENSMUSG00000058975 7_46368799     7 46.36880   9.237981         Kcnc1
    ## 58  ENSMUSG00000062044 7_44625756     7 44.62576  17.860818         Lmtk3
    ## 59  ENSMUSG00000062382 7_45622192     7 45.62219  36.491280       Gm10116
    ## 60  ENSMUSG00000014418 7_46838536     7 46.83854  40.496259          Hps5
    ## 61  ENSMUSG00000014550 7_45504585     7 45.50458   9.188581       Zfyve20
    ## 62  ENSMUSG00000062944 7_45463938     7 45.46394   8.237227 9130023H24Rik
    ## 63  ENSMUSG00000063268 7_45720922     7 45.72092   7.401914        Parp10
    ## 64  ENSMUSG00000063558 7_44588908     7 44.58891   7.273580          Aox1
    ## 65  ENSMUSG00000064158 7_46000542     7 46.00054  16.676429        Izumo1
    ## 66  ENSMUSG00000066026 7_45689416     7 45.68942   7.398933         Dhrs3
    ## 67  ENSMUSG00000066513 7_44431929     7 44.43193  16.938547        Klk1b4
    ## 68  ENSMUSG00000068245 7_45689416     7 45.68942   7.409084        Phf11d
    ## 69  ENSMUSG00000070498 7_46921126     7 46.92113   9.207246      Tmem132b
    ## 70  ENSMUSG00000070564 7_45270542     7 45.27054  22.007266          Ntn5
    ## 71  ENSMUSG00000070570 7_45420117     7 45.42012   7.539681       Slc17a7
    ## 72  ENSMUSG00000070730 7_46288948     7 46.28895   7.606435         Rmdn3
    ## 73  ENSMUSG00000071107 7_46280163     7 46.28016   8.186508 4732465J04Rik
    ## 74  ENSMUSG00000015222 7_46358463     7 46.35846   7.236585          Map2
    ## 75  ENSMUSG00000074129 7_45051425     7 45.05143  29.850315        Rpl13a
    ## 76  ENSMUSG00000074141 7_44896497     7 44.89650  17.811144         Il4i1
    ## 77  ENSMUSG00000016619 7_44940291     7 44.94029   7.380143         Nup50
    ## 78  ENSMUSG00000079109 7_45504585     7 45.50458   8.299377          Pms2
    ## 79  ENSMUSG00000079442 7_44819852     7 44.81985   8.842680    St6galnac4
    ## 80  ENSMUSG00000081142 7_46687297     7 46.68730   8.205158       Gm15497
    ## 81  ENSMUSG00000081251 7_46000542     7 46.00054  19.662409       Gm12164
    ## 82  ENSMUSG00000082062 7_45499645     7 45.49965  65.507662          Ftl2
    ## 83  ENSMUSG00000017009 7_45689416     7 45.68942   7.598037          Sdc4
    ## 84  ENSMUSG00000017756 7_45720922     7 45.72092   7.294622       Slc12a7
    ## 85  ENSMUSG00000084128 7_46750658     7 46.75066   7.773178         Esrp2
    ## 86  ENSMUSG00000084519 7_45106404     7 45.10640  13.219479       Gm22478
    ## 87  ENSMUSG00000084882 7_45689416     7 45.68942   8.625476       Gm16047
    ## 88  ENSMUSG00000085698 7_45484823     7 45.48482  66.022765       Gm16022
    ## 89  ENSMUSG00000018387 7_47026120     7 47.02612   7.494430       Shroom1
    ## 90  ENSMUSG00000087138 7_44940291     7 44.94029  59.304562       Gm15545
    ## 91  ENSMUSG00000087298 7_46828517     7 46.82852  13.316774        Gm9392
    ## 92  ENSMUSG00000090035 7_46315301     7 46.31530   7.870430        Galnt4
    ## 93  ENSMUSG00000090272 7_45455539     7 45.45554   7.253361         Mndal
    ## 94  ENSMUSG00000090523 7_45504585     7 45.50458   8.819789          Gypc
    ## 95  ENSMUSG00000018604 7_44721343     7 44.72134   7.197939          Tbx3
    ## 96  ENSMUSG00000000275 7_45101619     7 45.10162   9.902184        Trim25
    ## 97  ENSMUSG00000000359 7_44609389     7 44.60939   8.436092          Rem1
    ## 98  ENSMUSG00000091649 7_45144068     7 45.14407   7.308243        Phf11b
    ## 99  ENSMUSG00000096146 7_45705169     7 45.70517  12.105623        Kcnj11
    ## 100 ENSMUSG00000096755 7_45455539     7 45.45554  41.406606       Gm20746
    ## 101 ENSMUSG00000019539 7_45111189     7 45.11119  14.933531          Rcn3
    ## 102 ENSMUSG00000019558 7_46326762     7 46.32676  10.551710        Slc6a8
    ## 103 ENSMUSG00000020739 7_46358463     7 46.35846   8.518015         Nup85
    ## 104 ENSMUSG00000021418 7_45504585     7 45.50458   7.453778         Rpp40
    ## 105 ENSMUSG00000021470 7_46000542     7 46.00054   7.241501       Ercc6l2
    ## 106 ENSMUSG00000001123 7_46000542     7 46.00054   7.611372        Lgals9
    ## 107 ENSMUSG00000021708 7_46297732     7 46.29773   7.823481       Rasgrf2
    ## 108 ENSMUSG00000021745 7_44819852     7 44.81985   7.319465         Ptprg
    ## 109 ENSMUSG00000022019 7_45470001     7 45.47000   7.782115         Tdrd3
    ## 110 ENSMUSG00000022100 7_45499645     7 45.49965   8.282669          Xpo7
    ## 111 ENSMUSG00000022661 7_45434499     7 45.43450   7.818033         Cd200
    ## 112 ENSMUSG00000022780 7_46364052     7 46.36405   8.451017          Mfi2
    ## 113 ENSMUSG00000022861 7_45720922     7 45.72092   8.797565          Dgkg
    ## 114 ENSMUSG00000022892 7_46000542     7 46.00054  10.244252           App
    ## 115 ENSMUSG00000023010 7_46000542     7 46.00054   7.297829        Tmbim6
    ## 116 ENSMUSG00000023066 7_45622192     7 45.62219   9.103043          Rttn
    ## 117 ENSMUSG00000023087 7_45499645     7 45.49965   9.980165        Ccrn4l
    ## 118 ENSMUSG00000024014 7_45499645     7 45.49965   8.310795          Pim1
    ## 119 ENSMUSG00000024059 7_45455539     7 45.45554   7.416343         Clip4
    ## 120 ENSMUSG00000024269 7_46365522     7 46.36552   7.717114         Tpgs2
    ## 121 ENSMUSG00000024593 7_46326762     7 46.32676   8.960606        Megf10
    ## 122 ENSMUSG00000024953 7_45705169     7 45.70517   8.116757         Prdx5
    ## 123 ENSMUSG00000025492 7_45499645     7 45.49965   7.590536        Ifitm3
    ## 124 ENSMUSG00000025498 7_45455539     7 45.45554   7.446133          Irf7
    ## 125 ENSMUSG00000025909 7_45499645     7 45.49965   8.557393         Sntg1
    ## 126 ENSMUSG00000026080 7_45689416     7 45.68942   8.673487        Chst10
    ## 127 ENSMUSG00000002205 7_44761322     7 44.76132  14.796072          Vrk3
    ## 128 ENSMUSG00000002307 7_45689416     7 45.68942   7.695174          Daxx
    ## 129 ENSMUSG00000002325 7_46297732     7 46.29773   7.194969          Irf9
    ## 130 ENSMUSG00000026222 7_45144068     7 45.14407   7.293672         Sp100
    ## 131 ENSMUSG00000026437 7_47836298     7 47.83630   8.341530         Cdk18
    ## 132 ENSMUSG00000026707 7_44053303     7 44.05330   7.514586         Nsun6
    ## 133 ENSMUSG00000026764 7_46333584     7 46.33358   9.645653         Kif5c
    ## 134 ENSMUSG00000027199 7_46358463     7 46.35846  11.110193          Gatm
    ## 135 ENSMUSG00000027204 7_44721343     7 44.72134   7.584970          Fbn1
    ## 136 ENSMUSG00000027580 7_46297732     7 46.29773   7.652718         Helz2
    ## 137 ENSMUSG00000027650 7_46330173     7 46.33017   7.194538          Tti1
    ## 138 ENSMUSG00000002602 7_44940291     7 44.94029   8.041996           Axl
    ## 139 ENSMUSG00000002771 7_46000542     7 46.00054  13.297862        Grin2d
    ## 140 ENSMUSG00000002778 7_46000542     7 46.00054  12.618395        Kdelr1
    ## 141 ENSMUSG00000002781 7_46000542     7 46.00054  25.865946       Tmem143
    ## 142 ENSMUSG00000002881 7_46610522     7 46.61052   7.770303          Nab1
    ## 143 ENSMUSG00000002963 7_44780832     7 44.78083  73.413066          Pnkp
    ## 144 ENSMUSG00000028331 7_44761322     7 44.76132   7.455919 5830415F09Rik
    ## 145 ENSMUSG00000028391 7_46000542     7 46.00054  18.822973         Wdr31
    ## 146 ENSMUSG00000028517 7_46336995     7 46.33700   8.022758        Ppap2b
    ## 147 ENSMUSG00000028864 7_46000542     7 46.00054   7.347039           Hgf
    ## 148 ENSMUSG00000029475 7_45720922     7 45.72092   7.773247         Kdm2b
    ## 149 ENSMUSG00000029695 7_45720922     7 45.72092  10.033089          Aass
    ## 150 ENSMUSG00000029815 7_45622192     7 45.62219   8.730002        Malsu1
    ## 151 ENSMUSG00000003184 7_44819852     7 44.81985  29.392904          Irf3
    ## 152 ENSMUSG00000003190 7_45334220     7 45.33422  15.763956       Bcl2l12
    ## 153 ENSMUSG00000003269 7_45720922     7 45.72092  14.737735         Cyth2
    ## 154 ENSMUSG00000003271 7_44940291     7 44.94029  11.416083       Sult2b1
    ## 155 ENSMUSG00000003273 7_45689416     7 45.68942  59.674520         Car11
    ## 156 ENSMUSG00000003420 7_44940291     7 44.94029  11.274391         Fcgrt
    ## 157 ENSMUSG00000003421 7_44940291     7 44.94029  24.299416         Nosip
    ## 158 ENSMUSG00000003423 7_45270542     7 45.27054  93.687445        Pih1d1
    ## 159 ENSMUSG00000003429 7_45101619     7 45.10162  22.965145         Rps11
    ## 160 ENSMUSG00000030051 7_46297732     7 46.29773   7.969230          Aplf
    ## 161 ENSMUSG00000030107 7_45689416     7 45.68942   7.459232         Usp18
    ## 162 ENSMUSG00000030468 7_45144068     7 45.14407  16.914172       Siglecg
    ## 163 ENSMUSG00000030471 7_47430616     7 47.43062   8.811965       Zdhhc13
    ## 164 ENSMUSG00000030731 7_44053303     7 44.05330   7.914945          Syt3
    ## 165 ENSMUSG00000030739 7_44761322     7 44.76132  40.321801         Myh14
    ## 166 ENSMUSG00000030790 7_45270542     7 45.27054   8.397668           Adm
    ## 167 ENSMUSG00000030792 7_46280163     7 46.28016  13.000053         Dkkl1
    ## 168 ENSMUSG00000030824 7_45689416     7 45.68942  12.309167         Nucb1
    ## 169 ENSMUSG00000030827 7_45504585     7 45.50458   7.664410         Fgf21
    ## 170 ENSMUSG00000030838 7_46000542     7 46.00054  36.388014         Ush1c
    ## 171 ENSMUSG00000030839 7_46358463     7 46.35846  27.293819        Sergef
    ## 172 ENSMUSG00000031447 7_44431929     7 44.43193   8.861541         Lamp1
    ## 173 ENSMUSG00000031697 7_47971525     7 47.97152   7.805342          Orc6
    ## 174 ENSMUSG00000031818 7_45689416     7 45.68942   9.558762        Cox4i1
    ## 175 ENSMUSG00000003863 7_45334220     7 45.33422  21.133588        Ppfia3
    ## 176 ENSMUSG00000003865 7_45441513     7 45.44151   8.358699          Gys1
    ## 177 ENSMUSG00000003872 7_45420117     7 45.42012   9.889125         Lin7b
    ## 178 ENSMUSG00000003929 7_45720922     7 45.72092   7.269718         Zfp81
    ## 179 ENSMUSG00000004383 7_45720922     7 45.72092   7.388481         Large
    ## 180 ENSMUSG00000004609 7_44438774     7 44.43877  13.122583          Cd33
    ## 181 ENSMUSG00000032194 7_46315301     7 46.31530   7.635065         Kank2
    ## 182 ENSMUSG00000032481 7_44819852     7 44.81985   9.176663       Smarcc1
    ## 183 ENSMUSG00000033152 7_46280163     7 46.28016  10.030771        Podxl2
    ## 184 ENSMUSG00000033486 7_44995858     7 44.99586   7.526738      Catsper2
    ## 185 ENSMUSG00000033579 7_44874623     7 44.87462   7.264635          Fa2h
    ## 186 ENSMUSG00000034566 7_45689416     7 45.68942   7.288814         Atp5h
    ## 187 ENSMUSG00000004791 7_46000542     7 46.00054  10.412794           Pgf
    ## 188 ENSMUSG00000005142 7_47971525     7 47.97152   7.399253        Man2b1
    ## 189 ENSMUSG00000005267 7_45499645     7 45.49965   9.398486        Zfp287
    ## 190 ENSMUSG00000034708 7_44780832     7 44.78083   9.564299           Grn
    ## 191 ENSMUSG00000035042 7_45720922     7 45.72092   9.285916          Ccl5
    ## 192 ENSMUSG00000035673 7_46326762     7 46.32676   7.810045         Sbno2
    ## 193 ENSMUSG00000035692 7_45720922     7 45.72092   7.785958         Isg15
    ## 194 ENSMUSG00000035910 7_45499645     7 45.49965   7.550905        Dcdc2a
    ## 195 ENSMUSG00000036097 7_45504585     7 45.50458   7.507471       Fam178a
    ## 196 ENSMUSG00000036198 7_46288948     7 46.28895   9.934129      Arhgap36
    ## 197 ENSMUSG00000036368 7_47701070     7 47.70107   8.111344         Rmdn2
    ## 198 ENSMUSG00000036986 7_45705169     7 45.70517   8.312078           Pml
    ## 199 ENSMUSG00000037035 7_46326762     7 46.32676   8.602552         Inhbb
    ## 200 ENSMUSG00000037106 7_45622192     7 45.62219  10.289214        Fer1l6
    ## 201 ENSMUSG00000037321 7_45720922     7 45.72092   7.451845          Tap1
    ## 202 ENSMUSG00000037416 7_46000542     7 46.00054   7.182409         Dmxl1
    ##     chr      start        end strand     middle nearest.marker.id
    ## 1     1  59.100594  59.120096     -1  59.110345        1_59151364
    ## 2    13  46.418300  46.431090      1  46.424695       13_46463035
    ## 3     7  45.335290  45.338973      1  45.337132        7_45334220
    ## 4     7  45.183624  45.204892     -1  45.194258        7_45183584
    ## 5     7  45.017978  45.021646      1  45.019812        7_44995858
    ## 6     7  44.840901  44.849079     -1  44.844990        7_44838742
    ## 7     4  58.943628  58.958355      1  58.950992        4_58861443
    ## 8     7  44.532746  44.548849     -1  44.540798        7_44551043
    ## 9     7  44.310253  44.358351      1  44.334302        7_44341174
    ## 10   13  36.725625  36.734477     -1  36.730051       13_36759342
    ## 11    2 167.035793 167.063015     -1 167.049404       2_167033011
    ## 12    4 149.896283 149.955043     -1 149.925663       4_149925626
    ## 13   10 128.270576 128.292849      1 128.281712      10_128282522
    ## 14    7  46.647130  46.672537     -1  46.659834        7_46687297
    ## 15    7  46.104523  46.180033     -1  46.142278        7_46280163
    ## 16    7  45.924072  45.948956      1  45.936514        7_46000542
    ## 17    7  45.918023  45.921426     -1  45.919724        7_46000542
    ## 18    7  45.677689  45.694402     -1  45.686046        7_45689416
    ## 19    7  45.526330  45.554229      1  45.540280        7_45504585
    ## 20    2 167.492645 167.516173      1 167.504409       2_167523201
    ## 21    7  46.686108  46.710680     -1  46.698394        7_46687297
    ## 22   11 113.726850 113.751881     -1 113.739366      11_113743477
    ## 23    5 114.923240 114.937915      1 114.930578       5_114947913
    ## 24    3  94.413296  94.432633      1  94.422964        3_94423357
    ## 25    9  57.072040  57.128366      1  57.100203        9_56954968
    ## 26    8 121.084386 121.088144      1 121.086265       8_121091806
    ## 27    7  45.639977  45.646521      1  45.643249        7_45622192
    ## 28   16  44.347121  44.388497      1  44.367809       16_44332550
    ## 29   11  75.164565  75.169519     -1  75.167042       11_75099555
    ## 30    7  46.923539  46.958527     -1  46.941033        7_46921126
    ## 31    7  44.816088  44.830809      1  44.823448        7_44819852
    ## 32    9 111.054835 111.057270     -1 111.056052       9_111045299
    ## 33    7  45.627488  45.639093      1  45.633290        7_45622192
    ## 34    7  44.959373  44.974851     -1  44.967112        7_44940291
    ## 35    7  45.140684  45.154584     -1  45.147634        7_45144068
    ## 36    7  45.617606  45.621059      1  45.619332        7_45622192
    ## 37    8  71.534264  71.537437     -1  71.535850        8_71546739
    ## 38   14  55.591708  55.603693      1  55.597700       14_55600734
    ## 39   14  61.138457  61.240695      1  61.189576       14_61198806
    ## 40    3  73.048126  73.056943     -1  73.052534        3_73056409
    ## 41   16  35.926511  35.939151     -1  35.932831       16_35971529
    ## 42    7  45.366163  45.367948      1  45.367056        7_45334220
    ## 43    7  46.240987  46.311434      1  46.276210        7_46280163
    ## 44    7  45.457944  45.459884     -1  45.458914        7_45458985
    ## 45   16  10.420557  10.447362     -1  10.433960       16_10457509
    ## 46    7  44.496588  44.501134      1  44.498861        7_44482294
    ## 47    2 107.595630 107.600539     -1 107.598084       2_107376882
    ## 48   13  12.269426  12.340760     -1  12.305093       13_12162118
    ## 49    7  45.721212  45.729492      1  45.725352        7_45720922
    ## 50    5 105.519388 105.613018      1 105.566203       5_105603380
    ## 51    7  47.050640  47.054776      1  47.052708        7_47035721
    ## 52    7  45.472956  45.488805     -1  45.480880        7_45479882
    ## 53    7  44.896079  44.902631      1  44.899355        7_44896497
    ## 54    7  45.709467  45.718002     -1  45.713734        7_45720922
    ## 55   17   6.131854   6.132405     -1   6.132130        17_6124675
    ## 56    7  45.816461  45.824756     -1  45.820608        7_45720922
    ## 57    7  46.396497  46.438375      1  46.417436        7_46368799
    ## 58    7  45.783738  45.804144      1  45.793941        7_45720922
    ## 59   13  74.406396  74.407317      1  74.406856       13_74227165
    ## 60    7  46.760466  46.796064     -1  46.778265        7_46750658
    ## 61    6  92.186712  92.214925     -1  92.200818        6_92229302
    ## 62    7 128.234451 128.238031     -1 128.236241       7_128266981
    ## 63   15  76.231174  76.243441     -1  76.237308       15_76237397
    ## 64    1  58.029931  58.106413      1  58.068172        1_58149196
    ## 65    7  45.621811  45.628823      1  45.625317        7_45622192
    ## 66    4 144.892827 144.928209      1 144.910518       4_144912814
    ## 67    7  44.207435  44.211754      1  44.209594        7_44138903
    ## 68   14  59.347407  59.365470     -1  59.356438       14_59350137
    ## 69    5 125.531774 125.792583      1 125.662178       5_125693419
    ## 70    7  45.684022  45.694556      1  45.689289        7_45689416
    ## 71    7  45.163921  45.176138      1  45.170030        7_45174292
    ## 72    2 119.137001 119.157034     -1 119.147018       2_119152658
    ## 73   10  95.792571  95.794980      1  95.793776       10_95847015
    ## 74    1  66.175273  66.442583      1  66.308928        1_66331797
    ## 75    7  45.125558  45.136231     -1  45.130894        7_45144068
    ## 76    7  44.816370  44.840809      1  44.828590        7_44819852
    ## 77   15  84.923428  84.942955      1  84.933192       15_84929393
    ## 78    5 143.909964 143.937309      1 143.923636       5_143926063
    ## 79    2  32.587095  32.599698      1  32.593396        2_32582163
    ## 80    5 138.610739 138.611289      1 138.611014       5_138732772
    ## 81   11  45.829319  45.829869      1  45.829594       11_45796447
    ## 82    4 125.504288 125.504839      1 125.504564       4_125504939
    ## 83    2 164.424247 164.443887     -1 164.434067       2_164348452
    ## 84   13  73.763697  73.816743      1  73.790220       13_73735345
    ## 85    8 106.130319 106.136974     -1 106.133646       8_106121403
    ## 86    7  45.388531  45.388684      1  45.388608        7_45420117
    ## 87    7  45.630664  45.633077     -1  45.631870        7_45622192
    ## 88    7  45.550534  45.551615     -1  45.551074        7_45504585
    ## 89   11  53.457205  53.467766      1  53.462486       11_53492983
    ## 90    7  44.986900  44.994597      1  44.990748        7_44995858
    ## 91    7  46.820267  46.822048     -1  46.821158        7_46815826
    ## 92   10  99.108135  99.113245      1  99.110690       10_99055927
    ## 93    1 173.857221 173.942491     -1 173.899856       1_173809881
    ## 94   18  32.528322  32.560034     -1  32.544178       18_32496893
    ## 95    5 119.670669 119.684724      1 119.677696       5_119678319
    ## 96   11  88.999376  89.020293      1  89.009834       11_89003558
    ## 97    2 152.626951 152.635198      1 152.631074       2_152630199
    ## 98   14  59.320964  59.341335     -1  59.331150       14_59324652
    ## 99    7  46.098725  46.099897     -1  46.099311        7_46000542
    ## 100  10 130.592154 130.592705     -1 130.592430      10_130573968
    ## 101   7  45.082913  45.092221     -1  45.087567        7_45101619
    ## 102   X  73.673150  73.682502      1  73.677826        X_73645374
    ## 103  11 115.564434 115.583985      1 115.574210      11_115572062
    ## 104  13  35.893472  35.906359     -1  35.899916       13_35883708
    ## 105  13  63.815240  63.900302      1  63.857771       13_63834249
    ## 106  11  78.962974  78.984946     -1  78.973960       11_78973614
    ## 107  13  91.880400  92.131656     -1  92.006028       13_92011130
    ## 108  14  11.553532  12.242041      1  11.897786       14_12040183
    ## 109  14  87.416639  87.545373      1  87.481006       14_87479865
    ## 110  14  70.654246  70.766628     -1  70.710437       14_70747575
    ## 111  16  45.382135  45.409053     -1  45.395594       16_45437425
    ## 112  16  31.878810  31.899020      1  31.888915       16_31858099
    ## 113  16  22.468461  22.657221     -1  22.562841       16_22579959
    ## 114  16  84.954440  85.173707     -1  85.064074       16_85205378
    ## 115  15  99.392882  99.410049      1  99.401466       15_99393594
    ## 116  18  88.971790  89.131013      1  89.051402       18_89263452
    ## 117   3  51.224447  51.251644      1  51.238046        3_51227805
    ## 118  17  29.490812  29.495445      1  29.493128       17_29398852
    ## 119  17  71.781947  71.858351      1  71.820149       17_71788146
    ## 120  18  25.127223  25.169007     -1  25.148115       18_25149250
    ## 121  18  57.133090  57.297467      1  57.215278       18_57222632
    ## 122  19   6.906697   6.910106     -1   6.908402        19_6957328
    ## 123   7 141.009586 141.010770     -1 141.010178       7_141008500
    ## 124   7 141.262706 141.266481     -1 141.264594       7_141247814
    ## 125   1   8.361475   9.299878     -1   8.830676         1_8951236
    ## 126   1  38.863877  38.898160     -1  38.881018        1_38770397
    ## 127   7  44.748413  44.777515      1  44.762964        7_44761322
    ## 128  17  33.909414  33.915589      1  33.912502       17_33920457
    ## 129  14  55.603571  55.610030      1  55.606800       14_55609082
    ## 130   1  85.649988  85.709998      1  85.679993        1_85661923
    ## 131   1 132.113550 132.139685     -1 132.126618       1_132199611
    ## 132   2  14.995131  15.054884     -1  15.025008        2_15027005
    ## 133   2  49.619298  49.774778      1  49.697038        2_49650224
    ## 134   2 122.594467 122.611303     -1 122.602885       2_122693523
    ## 135   2 125.300594 125.507993     -1 125.404294       2_125396374
    ## 136   2 181.227615 181.242027     -1 181.234821       2_181203462
    ## 137   2 157.981803 158.028433     -1 158.005118       2_157983121
    ## 138   7  25.757273  25.788705     -1  25.772989        7_25668949
    ## 139   7  45.831883  45.870928     -1  45.851406        7_45720922
    ## 140   7  45.872772  45.883731      1  45.878252        7_46000542
    ## 141   7  45.896941  45.917413      1  45.907177        7_46000542
    ## 142   1  52.457294  52.500679     -1  52.478986        1_52479702
    ## 143   7  44.857139  44.862925      1  44.860032        7_44856652
    ## 144   4  46.376505  46.389437     -1  46.382971        4_46392881
    ## 145   4  62.448653  62.470896     -1  62.459774        4_62448296
    ## 146   4 105.157347 105.232764      1 105.195056       4_105191975
    ## 147   5  16.553550  16.619439      1  16.586494        5_16550184
    ## 148   5 122.870668 123.015080     -1 122.942874       5_122944168
    ## 149   6  23.072173  23.132986     -1  23.102580        6_23121321
    ## 150   6  49.073795  49.086751      1  49.080273        6_49116341
    ## 151   7  44.997648  45.002848      1  45.000248        7_44995858
    ## 152   7  44.991222  44.998712     -1  44.994967        7_44995858
    ## 153   7  45.806637  45.814581     -1  45.810609        7_45720922
    ## 154   7  45.729983  45.759612     -1  45.744798        7_45720922
    ## 155   7  45.699843  45.704682      1  45.702262        7_45705169
    ## 156   7  45.092993  45.103822     -1  45.098408        7_45101619
    ## 157   7  45.062429  45.078503      1  45.070466        7_45051425
    ## 158   7  45.154303  45.160069      1  45.157186        7_45165000
    ## 159   7  45.122388  45.124389     -1  45.123388        7_45111189
    ## 160   6  87.628429  87.672168     -1  87.650298        6_87719413
    ## 161   6 121.245906 121.270916      1 121.258411       6_121173514
    ## 162   7  43.408204  43.418358      1  43.413281        7_43416832
    ## 163   7  48.789003  48.827440      1  48.808222        7_48805261
    ## 164   7  44.384102  44.400187      1  44.392144        7_44364954
    ## 165   7  44.605803  44.670843     -1  44.638323        7_44632571
    ## 166   7 110.627669 110.629819      1 110.628744       7_110628503
    ## 167   7  45.207525  45.211883     -1  45.209704        7_45206863
    ## 168   7  45.492874  45.510406     -1  45.501640        7_45499645
    ## 169   7  45.613907  45.615490     -1  45.614698        7_45622192
    ## 170   7  46.195350  46.238503     -1  46.216926        7_46280163
    ## 171   7  46.443168  46.639807     -1  46.541488        7_46557864
    ## 172   8  13.159135  13.175338      1  13.167236        8_13209235
    ## 173   8  85.299632  85.308278      1  85.303955        8_85285466
    ## 174   8 120.668222 120.674207      1 120.671214       8_120681261
    ## 175   7  45.339126  45.367019     -1  45.353072        7_45334220
    ## 176   7  45.434844  45.456619      1  45.445732        7_45445019
    ## 177   7  45.367891  45.370583     -1  45.369237        7_45334220
    ## 178  17  33.329342  33.358878     -1  33.344110       17_33389925
    ## 179   8  72.814599  73.353540     -1  73.084070        8_72965724
    ## 180   7  43.527456  43.533171     -1  43.530314        7_43545216
    ## 181   9  21.766773  21.798546     -1  21.782660        9_21878582
    ## 182   9 110.132037 110.240178      1 110.186108       9_110205720
    ## 183   6  88.842558  88.875044     -1  88.858801        6_88852959
    ## 184   2 121.392631 121.413792     -1 121.403212       2_121538726
    ## 185   8 111.345135 111.393824     -1 111.369480       8_111390175
    ## 186  11 115.415689 115.419962     -1 115.417826      11_115406199
    ## 187  12  85.166639  85.177296     -1  85.171968       12_85201674
    ## 188   8  85.083269  85.098739      1  85.091004        8_85007380
    ## 189  11  62.700356  62.731905     -1  62.716130       11_62706563
    ## 190  11 102.430315 102.437048      1 102.433682      11_102371824
    ## 191  11  83.525778  83.530518     -1  83.528148       11_83554638
    ## 192  10  80.057416  80.102698     -1  80.080057       10_80053544
    ## 193   4 156.199424 156.200818     -1 156.200121       4_156216260
    ## 194  13  25.056004  25.210706      1  25.133355       13_25119107
    ## 195  19  44.931119  44.983787      1  44.957453       19_44938797
    ## 196   X  49.463945  49.500244      1  49.482094        X_49481750
    ## 197  17  79.614900  79.682152      1  79.648526       17_79725162
    ## 198   9  58.218076  58.249786     -1  58.233931        9_58264792
    ## 199   1 119.415465 119.422248     -1 119.418856       1_119435251
    ## 200  15  58.510048  58.665092      1  58.587570       15_58589562
    ## 201  17  34.187553  34.197225      1  34.192389       17_34195963
    ## 202  18  49.832670  49.965473      1  49.899072       18_49809849
    ##                  biotype        module hotspot
    ## 1         protein_coding         green    chr7
    ## 2         protein_coding          cyan    chr7
    ## 3         protein_coding          grey    <NA>
    ## 4         protein_coding          grey    <NA>
    ## 5         protein_coding          grey    <NA>
    ## 6         protein_coding        yellow   chr11
    ## 7         protein_coding         white    chr7
    ## 8         protein_coding         black    <NA>
    ## 9         protein_coding        purple    <NA>
    ## 10        protein_coding          grey    chr7
    ## 11        protein_coding       magenta    chr7
    ## 12        protein_coding       magenta    chr7
    ## 13        protein_coding       magenta    chr7
    ## 14        protein_coding          grey    <NA>
    ## 15        protein_coding         black    <NA>
    ## 16        protein_coding          grey    <NA>
    ## 17        protein_coding          cyan    <NA>
    ## 18        protein_coding     royalblue    <NA>
    ## 19        protein_coding          grey    <NA>
    ## 20        protein_coding       magenta    chr7
    ## 21        protein_coding          grey    <NA>
    ## 22        protein_coding          grey    chr7
    ## 23        protein_coding       magenta    chr7
    ## 24        protein_coding          grey    chr7
    ## 25        protein_coding         white    chr7
    ## 26        protein_coding   lightyellow    chr7
    ## 27        protein_coding     royalblue    <NA>
    ## 28        protein_coding     turquoise    chr7
    ## 29        protein_coding        yellow    chr7
    ## 30        protein_coding           tan    <NA>
    ## 31        protein_coding          blue    <NA>
    ## 32        protein_coding       magenta    chr7
    ## 33        protein_coding   lightyellow    <NA>
    ## 34        protein_coding          grey    <NA>
    ## 35        protein_coding          grey    <NA>
    ## 36        protein_coding       skyblue    <NA>
    ## 37        protein_coding       magenta    chr7
    ## 38        protein_coding       magenta    chr7
    ## 39        protein_coding          pink    chr7
    ## 40        protein_coding          grey    chr7
    ## 41        protein_coding       magenta    chr7
    ## 42  processed_transcript          blue    <NA>
    ## 43        protein_coding          grey    <NA>
    ## 44        protein_coding          grey    <NA>
    ## 45        protein_coding     turquoise    chr7
    ## 46        protein_coding          blue    <NA>
    ## 47            pseudogene          grey    chr7
    ## 48        protein_coding         brown    chr7
    ## 49        protein_coding          grey    <NA>
    ## 50        protein_coding   lightyellow    chr7
    ## 51        protein_coding         brown   chr13
    ## 52        protein_coding          grey    <NA>
    ## 53        protein_coding         black    <NA>
    ## 54        protein_coding         black    <NA>
    ## 55            pseudogene          grey    chr7
    ## 56        protein_coding       darkred    <NA>
    ## 57        protein_coding           red    <NA>
    ## 58        protein_coding        yellow    <NA>
    ## 59        protein_coding          grey    chr7
    ## 60        protein_coding          pink    <NA>
    ## 61        protein_coding          pink    chr7
    ## 62        protein_coding       skyblue    <NA>
    ## 63        protein_coding       magenta    chr7
    ## 64        protein_coding          grey    chr7
    ## 65        protein_coding       skyblue    <NA>
    ## 66        protein_coding         brown    chr7
    ## 67        protein_coding      darkgrey    <NA>
    ## 68        protein_coding       magenta    chr7
    ## 69        protein_coding   saddlebrown    chr7
    ## 70        protein_coding          grey    <NA>
    ## 71        protein_coding          grey    <NA>
    ## 72        protein_coding       magenta    chr7
    ## 73        protein_coding          grey    <NA>
    ## 74        protein_coding          pink    chr7
    ## 75        protein_coding       sienna3    <NA>
    ## 76        protein_coding         black    <NA>
    ## 77        protein_coding          grey    chr7
    ## 78        protein_coding     turquoise    chr7
    ## 79        protein_coding       magenta    chr7
    ## 80            pseudogene          grey    chr7
    ## 81            pseudogene          grey    chr7
    ## 82            pseudogene          grey    chr7
    ## 83        protein_coding      skyblue3    chr7
    ## 84        protein_coding darkturquoise    chr7
    ## 85        protein_coding    lightgreen    chr7
    ## 86                snoRNA          grey    <NA>
    ## 87             antisense          grey    <NA>
    ## 88             antisense          grey    <NA>
    ## 89        protein_coding        yellow    chr7
    ## 90  processed_transcript        yellow    <NA>
    ## 91            pseudogene          grey    <NA>
    ## 92        protein_coding          grey    chr7
    ## 93        protein_coding       magenta    chr7
    ## 94        protein_coding          cyan    chr7
    ## 95        protein_coding   lightyellow    chr7
    ## 96        protein_coding       magenta    chr7
    ## 97        protein_coding          cyan    chr7
    ## 98        protein_coding       magenta    chr7
    ## 99        protein_coding  midnightblue    <NA>
    ## 100           pseudogene          grey    chr7
    ## 101       protein_coding   greenyellow    <NA>
    ## 102       protein_coding           red    chr7
    ## 103       protein_coding       darkred    chr7
    ## 104       protein_coding         white    chr7
    ## 105       protein_coding     turquoise    chr7
    ## 106       protein_coding       magenta    chr7
    ## 107       protein_coding        yellow    chr7
    ## 108       protein_coding   lightyellow    chr7
    ## 109       protein_coding         green    chr7
    ## 110       protein_coding   saddlebrown    chr7
    ## 111       protein_coding         brown    chr7
    ## 112       protein_coding          grey    chr7
    ## 113       protein_coding     steelblue    chr7
    ## 114       protein_coding   greenyellow    chr7
    ## 115       protein_coding        yellow    chr7
    ## 116       protein_coding         white    chr7
    ## 117       protein_coding          grey    chr7
    ## 118       protein_coding   lightyellow    chr7
    ## 119       protein_coding          grey    chr7
    ## 120       protein_coding          grey    chr7
    ## 121       protein_coding     turquoise    chr7
    ## 122       protein_coding          blue    chr7
    ## 123       protein_coding       magenta    <NA>
    ## 124       protein_coding       magenta    <NA>
    ## 125       protein_coding     turquoise    chr7
    ## 126       protein_coding     steelblue    chr7
    ## 127       protein_coding     royalblue    <NA>
    ## 128       protein_coding       magenta    chr7
    ## 129       protein_coding       magenta    chr7
    ## 130       protein_coding       magenta    chr7
    ## 131       protein_coding        yellow    chr7
    ## 132       protein_coding     turquoise    chr7
    ## 133       protein_coding          pink    chr7
    ## 134       protein_coding         brown    chr7
    ## 135       protein_coding          cyan    chr7
    ## 136       protein_coding       magenta    chr7
    ## 137       protein_coding         white    chr7
    ## 138       protein_coding          cyan    <NA>
    ## 139       protein_coding         black    <NA>
    ## 140       protein_coding        yellow    chr5
    ## 141       protein_coding        yellow    <NA>
    ## 142       protein_coding          grey    chr7
    ## 143       protein_coding         black    <NA>
    ## 144       protein_coding     turquoise    chr7
    ## 145       protein_coding          grey    chr7
    ## 146       protein_coding   lightyellow    chr7
    ## 147       protein_coding          cyan    chr7
    ## 148       protein_coding          grey    chr7
    ## 149       protein_coding         brown    chr7
    ## 150       protein_coding       sienna3    chr7
    ## 151       protein_coding          blue    <NA>
    ## 152       protein_coding          grey    <NA>
    ## 153       protein_coding          grey    <NA>
    ## 154       protein_coding        yellow    <NA>
    ## 155       protein_coding          grey    <NA>
    ## 156       protein_coding   lightyellow    <NA>
    ## 157       protein_coding          blue    <NA>
    ## 158       protein_coding          blue    <NA>
    ## 159       protein_coding        salmon    <NA>
    ## 160       protein_coding          grey    chr7
    ## 161       protein_coding       magenta    chr7
    ## 162       protein_coding          grey    <NA>
    ## 163       protein_coding     turquoise    <NA>
    ## 164       protein_coding        yellow    <NA>
    ## 165       protein_coding          grey    <NA>
    ## 166       protein_coding   lightyellow    <NA>
    ## 167       protein_coding          grey    <NA>
    ## 168       protein_coding        yellow    <NA>
    ## 169       protein_coding       skyblue    <NA>
    ## 170       protein_coding          grey    chr2
    ## 171       protein_coding          grey    <NA>
    ## 172       protein_coding          grey    chr7
    ## 173       protein_coding     turquoise    chr7
    ## 174       protein_coding       sienna3    chr7
    ## 175       protein_coding         black    <NA>
    ## 176       protein_coding       skyblue    <NA>
    ## 177       protein_coding          blue    <NA>
    ## 178       protein_coding     turquoise    chr7
    ## 179       protein_coding        yellow    chr7
    ## 180       protein_coding     darkgreen    <NA>
    ## 181       protein_coding        yellow    chr7
    ## 182       protein_coding   saddlebrown    chr7
    ## 183       protein_coding          blue    chr7
    ## 184       protein_coding          grey    chr7
    ## 185       protein_coding        yellow    chr7
    ## 186       protein_coding        salmon    chr7
    ## 187       protein_coding         brown    chr7
    ## 188       protein_coding        yellow    chr7
    ## 189       protein_coding         white    chr7
    ## 190       protein_coding        yellow    chr7
    ## 191       protein_coding       magenta    chr7
    ## 192       protein_coding     royalblue    chr7
    ## 193       protein_coding       magenta    chr7
    ## 194       protein_coding         brown    chr7
    ## 195       protein_coding          pink    chr7
    ## 196       protein_coding          grey    chr7
    ## 197       protein_coding     turquoise    chr7
    ## 198       protein_coding       magenta    chr7
    ## 199       protein_coding   lightyellow    chr7
    ## 200       protein_coding          grey    chr7
    ## 201       protein_coding       magenta    chr7
    ## 202       protein_coding          pink    chr7

The above tibble has local and nonlocal traits.

``` r
local_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>%
  mutate(local_tx = chr ==hot_chrom) %>%
  filter(local_tx)
```

``` r
dim(local_annot)
```

    ## [1] 78 16

``` r
nonlocal_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chrom, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>% #genomewide threshold
  mutate(local_tx = chr == hot_chrom) %>%
  filter(!local_tx) %>% 
  filter(!is.na(hotspot))
```

``` r
dim(nonlocal_annot)
```

    ## [1] 123  16

We’ll use only those nonlocal traits that Keller et al. annotated as
mapping to a hotspot.

We also want to look at the local traits for Chr 7 hotspot. We have 78
local traits. We want to identify a subset of those for two-dimensional
scans. Let’s first limit to those near the hotspot, ie, within 5Mb of
either end.

``` r
local_annot2 <- local_annot %>%
  filter(middle <= hot_upper + 5, middle >= hot_lower - 5) %>%
  arrange(desc(lod))
```

Recall that *Pdx1* is the gene that Keller et al. identified as the key
mediator for Chr 5 hotspot. Its LOD is only 15, which seems to indicate
that I should be looking at a broad set of local traits, not just those,
eg., with LOD \> 40\!

Also, I’m limiting consideration of local genes to those with middle
within 5Mb of the hotspot.

Now, I make the two expression trait matrices that I need for use with
Condor.

``` r
hot_local <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% local_annot2$annot.id]
hot_nonlocal <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% nonlocal_annot$annot.id]
saveRDS(hot_local, "../data/Chr7hot_local.rds")
saveRDS(hot_nonlocal, "../data/Chr7hot_nonlocal.rds")
saveRDS(K$`7`, "../data/Chr7_kinship.rds")
saveRDS(genoprobs$`7`, "../data/Chr7_aprobs.rds")
```

Lastly, we need to identify the scan region for Chr 7.

We consider the physical map.

``` r
(start_index <- which(map$`7`== map$`7`[map$`7` > hot_lower][1]) - 50)
```

    ## 7_44020771 
    ##        927

``` r
(stop_index <- which(map$`7`== map$`7`[map$`7` > hot_upper][1]) + 50)
```

    ## 7_48106752 
    ##       1168

``` r
(nsnp <- stop_index - start_index + 1)
```

    ## 7_48106752 
    ##        242

``` r
dim(local_annot2)
```

    ## [1] 73 16

## References

Keller et al. 2018. Genetics. Genetic drivers of pancreatic islet
function.

## Session info

``` r
devtools::session_info()
```

    ## ─ Session info ──────────────────────────────────────────────────────────
    ##  setting  value                       
    ##  version  R version 3.5.3 (2019-03-11)
    ##  os       macOS Mojave 10.14.5        
    ##  system   x86_64, darwin15.6.0        
    ##  ui       X11                         
    ##  language (EN)                        
    ##  collate  en_US.UTF-8                 
    ##  ctype    en_US.UTF-8                 
    ##  tz       America/Chicago             
    ##  date     2019-05-31                  
    ## 
    ## ─ Packages ──────────────────────────────────────────────────────────────
    ##  package     * version date       lib source                        
    ##  assertthat    0.2.1   2019-03-21 [1] CRAN (R 3.5.2)                
    ##  backports     1.1.4   2019-04-10 [1] CRAN (R 3.5.2)                
    ##  bit           1.1-14  2018-05-29 [1] CRAN (R 3.5.0)                
    ##  bit64         0.9-7   2017-05-08 [1] CRAN (R 3.5.0)                
    ##  blob          1.1.1   2018-03-25 [1] CRAN (R 3.5.0)                
    ##  broom         0.5.1   2018-12-05 [1] CRAN (R 3.5.0)                
    ##  callr         3.2.0   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  cellranger    1.1.0   2016-07-27 [1] CRAN (R 3.5.0)                
    ##  cli           1.1.0   2019-03-19 [1] CRAN (R 3.5.3)                
    ##  colorspace    1.4-1   2019-03-18 [1] CRAN (R 3.5.3)                
    ##  crayon        1.3.4   2017-09-16 [1] CRAN (R 3.5.0)                
    ##  data.table    1.12.0  2019-01-13 [1] CRAN (R 3.5.2)                
    ##  DBI           1.0.0   2018-05-02 [1] CRAN (R 3.5.0)                
    ##  desc          1.2.0   2018-05-01 [1] CRAN (R 3.5.0)                
    ##  devtools      2.0.2   2019-04-08 [1] CRAN (R 3.5.2)                
    ##  digest        0.6.19  2019-05-20 [1] CRAN (R 3.5.2)                
    ##  dplyr       * 0.8.0.1 2019-02-15 [1] CRAN (R 3.5.2)                
    ##  evaluate      0.14    2019-05-28 [1] CRAN (R 3.5.3)                
    ##  forcats     * 0.4.0   2019-02-17 [1] CRAN (R 3.5.2)                
    ##  fs            1.3.1   2019-05-06 [1] CRAN (R 3.5.2)                
    ##  generics      0.0.2   2018-11-29 [1] CRAN (R 3.5.0)                
    ##  ggplot2     * 3.1.1   2019-04-07 [1] CRAN (R 3.5.3)                
    ##  glue          1.3.1   2019-03-12 [1] CRAN (R 3.5.2)                
    ##  gtable        0.3.0   2019-03-25 [1] CRAN (R 3.5.2)                
    ##  haven         2.1.0   2019-02-19 [1] CRAN (R 3.5.2)                
    ##  hms           0.4.2   2018-03-10 [1] CRAN (R 3.5.0)                
    ##  htmltools     0.3.6   2017-04-28 [1] CRAN (R 3.5.0)                
    ##  httr          1.4.0   2018-12-11 [1] CRAN (R 3.5.0)                
    ##  jsonlite      1.6     2018-12-07 [1] CRAN (R 3.5.0)                
    ##  knitr         1.23    2019-05-18 [1] CRAN (R 3.5.2)                
    ##  lattice       0.20-38 2018-11-04 [1] CRAN (R 3.5.3)                
    ##  lazyeval      0.2.2   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  lubridate     1.7.4   2018-04-11 [1] CRAN (R 3.5.0)                
    ##  magrittr      1.5     2014-11-22 [1] CRAN (R 3.5.0)                
    ##  memoise       1.1.0   2017-04-21 [1] CRAN (R 3.5.0)                
    ##  modelr        0.1.4   2019-02-18 [1] CRAN (R 3.5.2)                
    ##  munsell       0.5.0   2018-06-12 [1] CRAN (R 3.5.0)                
    ##  nlme          3.1-137 2018-04-07 [1] CRAN (R 3.5.3)                
    ##  pillar        1.3.1   2018-12-15 [1] CRAN (R 3.5.0)                
    ##  pkgbuild      1.0.3   2019-03-20 [1] CRAN (R 3.5.2)                
    ##  pkgconfig     2.0.2   2018-08-16 [1] CRAN (R 3.5.0)                
    ##  pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.5.0)                
    ##  plyr          1.8.4   2016-06-08 [1] CRAN (R 3.5.0)                
    ##  prettyunits   1.0.2   2015-07-13 [1] CRAN (R 3.5.0)                
    ##  processx      3.3.1   2019-05-08 [1] CRAN (R 3.5.2)                
    ##  ps            1.3.0   2018-12-21 [1] CRAN (R 3.5.0)                
    ##  purrr       * 0.3.2   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  qtl2        * 0.18    2019-02-08 [1] local                         
    ##  R6            2.4.0   2019-02-14 [1] CRAN (R 3.5.2)                
    ##  Rcpp          1.0.1.3 2019-05-29 [1] Github (RcppCore/Rcpp@6062d56)
    ##  readr       * 1.3.1   2018-12-21 [1] CRAN (R 3.5.0)                
    ##  readxl        1.3.0   2019-02-15 [1] CRAN (R 3.5.2)                
    ##  remotes       2.0.4   2019-04-10 [1] CRAN (R 3.5.2)                
    ##  rlang         0.3.4   2019-04-07 [1] CRAN (R 3.5.3)                
    ##  rmarkdown     1.13    2019-05-22 [1] CRAN (R 3.5.2)                
    ##  rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.5.0)                
    ##  RSQLite       2.1.1   2018-05-06 [1] CRAN (R 3.5.0)                
    ##  rstudioapi    0.10    2019-03-19 [1] CRAN (R 3.5.2)                
    ##  rvest         0.3.2   2016-06-17 [1] CRAN (R 3.5.0)                
    ##  scales        1.0.0   2018-08-09 [1] CRAN (R 3.5.0)                
    ##  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.5.0)                
    ##  stringi       1.4.3   2019-03-12 [1] CRAN (R 3.5.2)                
    ##  stringr     * 1.4.0   2019-02-10 [1] CRAN (R 3.5.2)                
    ##  testthat      2.1.1   2019-04-23 [1] CRAN (R 3.5.3)                
    ##  tibble      * 2.1.1   2019-03-16 [1] CRAN (R 3.5.3)                
    ##  tidyr       * 0.8.3   2019-03-01 [1] CRAN (R 3.5.2)                
    ##  tidyselect    0.2.5   2018-10-11 [1] CRAN (R 3.5.0)                
    ##  tidyverse   * 1.2.1   2017-11-14 [1] CRAN (R 3.5.0)                
    ##  usethis       1.5.0   2019-04-07 [1] CRAN (R 3.5.2)                
    ##  withr         2.1.2   2018-03-15 [1] CRAN (R 3.5.0)                
    ##  xfun          0.7     2019-05-14 [1] CRAN (R 3.5.2)                
    ##  xml2          1.2.0   2018-01-24 [1] CRAN (R 3.5.0)                
    ##  yaml          2.2.0   2018-07-25 [1] CRAN (R 3.5.0)                
    ## 
    ## [1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library
