Setting up for use of Condor CHTC
================
Frederick Boehm
05/30/2019

## Goals

Goal here is to document the setup & organization of files that I need
for using Condor CHTC to do my two-dimensional scans as part of the
analyses of the four Keller hotspots. Note that they reported 5
hotspots. I previously performed the analyses for Chromosome 2 hotspot.

## Setting

Recall that I have downloaded a Keller 2018 .RData file from DataDryad
that contains many of their results. The .RData file,
“data/Attie\_DO378\_eQTL\_viewer\_v1.Rdata” annotates 139 traits as
belonging to the chromosome 2 hotspot.

Following Karl’s suggestion, I want to look at not only the local trait
that Keller identifies as causal for the hotspot, but also several other
local-ish traits that have reasonable univariate LOD scores.

## Plan for using Condor

I’ll batch the two-dimensional scan jobs by the local trait identity.
So, for each local trait that I choose to examine, I’ll have 139 jobs -
one for each pairing with a hotspot trait.

I’ll organize the data files such that each local trait is its own RData
file, while the 139 nonlocal traits are all in a single RData file. We
can then use the automatically assigned job number to choose a nonlocal
trait for each
job.

## Preparing RDS files for Chr 5 analysis

``` r
load("~/Box Sync/attie/keller2018-chr2-hotspot-chtc/data-to-ignore/Attie_DO378_eQTL_viewer_v1.Rdata")
ls()
```

    ## [1] "dataset.clinical.phenotypes" "dataset.islet.hotspots"     
    ## [3] "dataset.islet.modules"       "dataset.islet.rnaseq"       
    ## [5] "ensembl.version"             "genoprobs"                  
    ## [7] "K"                           "map"                        
    ## [9] "markers"

``` r
library(tidyverse)
```

    ## ── Attaching packages ─────────────────────────────────────────────── tidyverse 1.2.1 ──

    ## ✔ ggplot2 3.1.1       ✔ purrr   0.3.2  
    ## ✔ tibble  2.1.1       ✔ dplyr   0.8.0.1
    ## ✔ tidyr   0.8.3       ✔ stringr 1.4.0  
    ## ✔ readr   1.3.1       ✔ forcats 0.4.0

    ## ── Conflicts ────────────────────────────────────────────────── tidyverse_conflicts() ──
    ## ✖ dplyr::filter() masks stats::filter()
    ## ✖ dplyr::lag()    masks stats::lag()

``` r
dataset.islet.rnaseq$expr %>%
  dim
```

    ## [1]   378 21771

``` r
colnames(dataset.islet.rnaseq$expr) %>%
  head
```

    ## [1] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000037"
    ## [4] "ENSMUSG00000000049" "ENSMUSG00000000056" "ENSMUSG00000000058"

We need to get the names of the nonlocal traits that map to Chromosome 5
hotspot.

``` r
dataset.islet.rnaseq$annots %>%
  colnames
```

    ##  [1] "gene_id"           "symbol"            "chr"              
    ##  [4] "start"             "end"               "strand"           
    ##  [7] "middle"            "nearest.marker.id" "biotype"          
    ## [10] "module"            "hotspot"

``` r
#dataset.islet.rnaseq$annots$hotspot
dataset.islet.rnaseq$annots %>%
  filter(hotspot == "chr5") -> hotspot_annot_tib
dim(hotspot_annot_tib)
```

    ## [1] 180  11

Keller et al., in their supplementary files, reported 182 transcripts,
yet I’ve isolated only 180 here.

I previously documented the issue with the Attie data set’s hotspot
annotations. Namely, when that file was created, they permitted the
“hotspot” field to be only a single value per transcript. Thus, if a
transcript mapped to multiple hotspots, it would still have only a
single value in this “hotspot” field. Because of this, I need to first
redo the mediation analyses that Keller did.

Specifically, I’ll isolate all ~600 or so transcripts that map to at
least one hotspot per the hotspot annotation. I’ll then perform
mediation analyses with each (for the appropriate local transcript).

``` r
dataset.islet.rnaseq$annots %>%
  filter(!is.na(hotspot)) -> hotspot_all_tx
dim(hotspot_all_tx)
```

    ## [1] 671  11

I have 671 transcripts. Let’s examine all 671 at each of the four
hotspots that I have yet to study, ie, those on Chromosomes 5, 7, 11,
and 13. Actually, before I do that, I want to get the genome-wide scan
results for these 671 transcripts to see, for each, which hotspots they
map
to.

``` r
tx671 <- dataset.islet.rnaseq$expr[ , colnames(dataset.islet.rnaseq$expr) %in% hotspot_all_tx$gene_id]
dim(tx671)  
```

    ## [1] 378 671

Now, I need to run `scan1` for each of the 671 traits.

Note that I need to do these with the covariates: sex and the wave
indicators.

``` r
library(qtl2)
```

    ## 
    ## Attaching package: 'qtl2'

    ## The following object is masked from 'package:readr':
    ## 
    ##     read_csv

``` r
fn <- "_cache/s1_671.rds"
if (file.exists(fn)){
  s1_671 <- readRDS(fn)
} else {
  s1_671 <- scan1(genoprobs = genoprobs, pheno = tx671, kinship = K, addcovar = dataset.islet.rnaseq$covar, cores = 5)
  saveRDS(s1_671, file = fn)
}
```

I forgot that the LOD peaks for all traits are saved in
`dataset.islet.rnaseq$lod.peaks`. It has nearly 40,000 peaks recorded. I
really just need to look at it. That is, for now, I don’t really need
the output of `scan1()`.

We want the ensembl ids for the traits that map to Chr 5 hotspot. First,
define the boundaries of the hotspot. The supplemental table in Keller
et al. 2018 lists only the midpoint of the hotspot. I believe that the
half-width is 2 Mb.

``` r
c5hot_lower <- 144
c5hot_upper <- 148
```

Now, filter to get those traits that map to the hotspot. Note that we’re
likely to get local traits, too, in this step.

``` r
dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == 5, pos <= c5hot_upper, pos >= c5hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18)
```

    ##               annot.id   marker.id chrom      pos        lod        symbol
    ## 1   ENSMUSG00000037933 5_147253583     5 147.2536   8.581871         Bicd2
    ## 2   ENSMUSG00000038527 5_147253583     5 147.2536   8.252539          C1rl
    ## 3   ENSMUSG00000038722 5_144716797     5 144.7168   8.787886         Bud31
    ## 4   ENSMUSG00000038780 5_145106694     5 145.1067  10.770852        Smurf1
    ## 5   ENSMUSG00000038855 5_144617027     5 144.6170  15.434906         Itpkb
    ## 6   ENSMUSG00000038963 5_147253583     5 147.2536  11.105587       Slco4a1
    ## 7   ENSMUSG00000038970 5_144348163     5 144.3482  13.813323         Lmtk2
    ## 8   ENSMUSG00000039109 5_147286753     5 147.2868   9.874575         F13a1
    ## 9   ENSMUSG00000039197 5_144617027     5 144.6170   7.433270           Adk
    ## 10  ENSMUSG00000039217 5_147914702     5 147.9147   7.701887          Il18
    ## 11  ENSMUSG00000039419 5_144587968     5 144.5880   9.943503       Cntnap2
    ## 12  ENSMUSG00000039519 5_147275576     5 147.2756   8.966354        Cyp7b1
    ## 13  ENSMUSG00000039835 5_145106694     5 145.1067   7.430985         Nhsl1
    ## 14  ENSMUSG00000040147 5_144727745     5 144.7277   7.545383          Maob
    ## 15  ENSMUSG00000040219 5_144375504     5 144.3755   7.593639         Ttc12
    ## 16  ENSMUSG00000040412 5_144423461     5 144.4235   8.485057 5330417C22Rik
    ## 17  ENSMUSG00000041112 5_147253583     5 147.2536   7.215660         Elmo1
    ## 18  ENSMUSG00000041202 5_147621507     5 147.6215   7.333873       Pla2g2d
    ## 19  ENSMUSG00000041453 5_146699418     5 146.6994 123.657248         Rpl21
    ## 20  ENSMUSG00000041577 5_144185205     5 144.1852   7.912710         Prelp
    ## 21  ENSMUSG00000042429 5_144367577     5 144.3676  10.118809        Adora1
    ## 22  ENSMUSG00000042659 5_144617027     5 144.6170   8.844420        Arrdc4
    ## 23  ENSMUSG00000043388 5_144705848     5 144.7058  97.429140       Tmem130
    ## 24  ENSMUSG00000044066 5_147299484     5 147.2995   7.368401         Cep68
    ## 25  ENSMUSG00000044177 5_147253583     5 147.2536  10.437395       Wfikkn2
    ## 26  ENSMUSG00000044499 5_147221814     5 147.2218   9.076690        Hs3st5
    ## 27  ENSMUSG00000007038 5_145106694     5 145.1067   7.823214          Neu1
    ## 28  ENSMUSG00000007594 5_146694026     5 146.6940   8.740749        Hapln4
    ## 29  ENSMUSG00000007888 5_144727745     5 144.7277   7.499567         Crlf1
    ## 30  ENSMUSG00000045078 5_144411293     5 144.4113  11.188454        Rnf216
    ## 31  ENSMUSG00000045482 5_144705848     5 144.7058   8.615970         Trrap
    ## 32  ENSMUSG00000046204 5_144558267     5 144.5583   7.613675         Pnma2
    ## 33  ENSMUSG00000046314 5_146865044     5 146.8650   7.977230        Stxbp6
    ## 34  ENSMUSG00000046402 5_147157965     5 147.1580   8.357390          Rbp1
    ## 35  ENSMUSG00000046463 5_144617027     5 144.6170   7.737488 5930403N24Rik
    ## 36  ENSMUSG00000046593 5_144423461     5 144.4235   7.240959       Tmem215
    ## 37  ENSMUSG00000046658 5_146324700     5 146.3247   7.497562        Zfp316
    ## 38  ENSMUSG00000046841 5_147916803     5 147.9168   7.852483         Ckap4
    ## 39  ENSMUSG00000046982 5_144407236     5 144.4072   7.356768         Tshz1
    ## 40  ENSMUSG00000047139 5_144617027     5 144.6170   7.838393         Cd24a
    ## 41  ENSMUSG00000047161 5_146103590     5 146.1036   7.899927         Chst9
    ## 42  ENSMUSG00000048216 5_144897381     5 144.8974   8.014328         Gpr85
    ## 43  ENSMUSG00000048720 5_147089309     5 147.0893   7.299479       Tbc1d12
    ## 44  ENSMUSG00000048910 5_144365599     5 144.3656  11.071207       Fam220a
    ## 45  ENSMUSG00000050761 5_147253583     5 147.2536   7.905369         Gp1bb
    ## 46  ENSMUSG00000050966 5_147278599     5 147.2786   7.879251        Lin28a
    ## 47  ENSMUSG00000051209 5_147287312     5 147.2873   8.541877        Gpr119
    ## 48  ENSMUSG00000052117 5_147253583     5 147.2536   7.491495 D630039A03Rik
    ## 49  ENSMUSG00000052271 5_146283615     5 146.2836  13.256971       Bhlha15
    ## 50  ENSMUSG00000053604 5_146334148     5 146.3341   8.022546          Rpia
    ## 51  ENSMUSG00000053716 5_144442944     5 144.4429   8.501527         Dusp7
    ## 52  ENSMUSG00000054008 5_147253583     5 147.2536   7.690149         Ndst1
    ## 53  ENSMUSG00000054277 5_146704810     5 146.7048   7.821351       Arfgap3
    ## 54  ENSMUSG00000054589 5_146334148     5 146.3341   7.290944        Gm9949
    ## 55  ENSMUSG00000055026 5_147253583     5 147.2536   8.785844        Gabrg3
    ## 56  ENSMUSG00000056379 5_147253583     5 147.2536   8.690595         Taar1
    ## 57  ENSMUSG00000057157 5_147795391     5 147.7954  20.792916        Gm6054
    ## 58  ENSMUSG00000057370 5_146707507     5 146.7075  41.194689    Rpl21-ps12
    ## 59  ENSMUSG00000058700 5_146862885     5 146.8629  16.935324        Gm8724
    ## 60  ENSMUSG00000059991 5_144444976     5 144.4450  22.912418         Nptx2
    ## 61  ENSMUSG00000060566 5_146862885     5 146.8629   8.525833     Rpl21-ps3
    ## 62  ENSMUSG00000060899 5_146322338     5 146.3223   9.080983    Rpl21-ps11
    ## 63  ENSMUSG00000061353 5_146904986     5 146.9050   8.842727        Cxcl12
    ## 64  ENSMUSG00000061684 5_146803810     5 146.8038   8.075689     Rpl21-ps8
    ## 65  ENSMUSG00000061947 5_147281622     5 147.2816  10.044685     Serpina10
    ## 66  ENSMUSG00000062554 5_147080273     5 147.0803 119.080437       Gm12751
    ## 67  ENSMUSG00000014496 5_147157965     5 147.1580   7.250428       Ankrd28
    ## 68  ENSMUSG00000062980 5_147114764     5 147.1148  12.209640         Cped1
    ## 69  ENSMUSG00000064177 5_147327447     5 147.3274  11.819643          Ghrl
    ## 70  ENSMUSG00000064315 5_146803810     5 146.8038  20.269702    Rpl21-ps10
    ## 71  ENSMUSG00000066150 5_147287312     5 147.2873   8.068774       Slc31a1
    ## 72  ENSMUSG00000066180 5_146862885     5 146.8629  58.411621       Gm10155
    ## 73  ENSMUSG00000066443 5_146707507     5 146.7075  84.239736       Gm10163
    ## 74  ENSMUSG00000066621 5_146103590     5 146.1036  10.928649        Tecpr1
    ## 75  ENSMUSG00000068014 5_146862885     5 146.8629  85.160236       Gm14648
    ## 76  ENSMUSG00000068397 5_146707507     5 146.7075 101.219770       Gm10240
    ## 77  ENSMUSG00000070419 5_146103590     5 146.1036  53.661681       Cyp3a57
    ## 78  ENSMUSG00000070420 5_146240909     5 146.2409  17.354346       Zscan25
    ## 79  ENSMUSG00000015401 5_144385418     5 144.3854   9.646061        Tmem27
    ## 80  ENSMUSG00000072884 5_147253583     5 147.2536   7.352495       Gm10433
    ## 81  ENSMUSG00000073910 5_147080273     5 147.0803   8.904103         Mob3b
    ## 82  ENSMUSG00000075569 5_144705848     5 144.7058   9.517501       Rsph10b
    ## 83  ENSMUSG00000016200 5_144361337     5 144.3613   9.183923         Syt14
    ## 84  ENSMUSG00000016503 5_146904986     5 146.9050  86.852231         Gtf3a
    ## 85  ENSMUSG00000016510 5_147064187     5 147.0642  15.236930         Mtif3
    ## 86  ENSMUSG00000016520 5_147064187     5 147.0642  14.880149          Lnx2
    ## 87  ENSMUSG00000078917 5_146863965     5 146.8640  26.846502       Gm11281
    ## 88  ENSMUSG00000078935 5_144617027     5 144.6170   7.540734 1700025C18Rik
    ## 89  ENSMUSG00000080957 5_145342881     5 145.3429  11.340943       Gm15739
    ## 90  ENSMUSG00000081167 5_146862885     5 146.8629  19.607576        Gm8880
    ## 91  ENSMUSG00000081559 5_146727146     5 146.7271  39.425616       Gm12411
    ## 92  ENSMUSG00000082879 5_146696722     5 146.6967  12.863006       Gm13525
    ## 93  ENSMUSG00000083596 5_147069549     5 147.0695  20.144500    Rpl21-ps15
    ## 94  ENSMUSG00000083910 5_146889753     5 146.8898   8.067080       Gm13604
    ## 95  ENSMUSG00000083988 5_147221814     5 147.2218  27.104451     Rps10-ps4
    ## 96  ENSMUSG00000084383 5_146707507     5 146.7075 101.219770       Gm13370
    ## 97  ENSMUSG00000085419 5_147602453     5 147.6025   7.210868       Gm11734
    ## 98  ENSMUSG00000085565 5_147988327     5 147.9883  12.325526       Gm15721
    ## 99  ENSMUSG00000085982 5_144727745     5 144.7277   9.707417 9530051G07Rik
    ## 100 ENSMUSG00000086288 5_144348163     5 144.3482   7.781782       Gm15265
    ## 101 ENSMUSG00000086894 5_144051716     5 144.0517  15.081519       Gm15708
    ## 102 ENSMUSG00000089883 5_146704810     5 146.7048  14.362952        Gm3355
    ## 103 ENSMUSG00000089959 5_147563362     5 147.5634   7.894574       Gm16268
    ## 104 ENSMUSG00000090213 5_147221814     5 147.2218   7.312829       Tmem189
    ## 105 ENSMUSG00000090330 5_144423461     5 144.4235   7.381182 9130221H12Rik
    ## 106 ENSMUSG00000019066 5_146863965     5 146.8640   9.353705         Rab3d
    ## 107 ENSMUSG00000092570 5_147275576     5 147.2756  55.533028 2210019I11Rik
    ## 108 ENSMUSG00000093720 5_144361337     5 144.3613   7.694305       Gm20635
    ## 109 ENSMUSG00000093916 5_146334148     5 146.3341   8.843562         Gm379
    ## 110 ENSMUSG00000094092 5_146803810     5 146.8038  10.465282       Gm15921
    ## 111 ENSMUSG00000094185 5_146734615     5 146.7346  11.950488        Gm5445
    ## 112 ENSMUSG00000094326 5_146803810     5 146.8038  23.636806       Gm10045
    ## 113 ENSMUSG00000094910 5_147286753     5 147.2868   7.670759 D430019H16Rik
    ## 114 ENSMUSG00000094929 5_147989668     5 147.9897   7.512724        Gm3526
    ## 115 ENSMUSG00000095040 5_145916546     5 145.9165  61.707986 1700001J03Rik
    ## 116 ENSMUSG00000095524 5_146863965     5 146.8640  25.443558       Gm15682
    ## 117 ENSMUSG00000095940 5_146734615     5 146.7346  34.551776       Gm16416
    ## 118 ENSMUSG00000098053 5_147286753     5 147.2868  19.162110       Gm27033
    ## 119 ENSMUSG00000020122 5_147253583     5 147.2536   8.256919          Egfr
    ## 120 ENSMUSG00000020427 5_147137425     5 147.1374   7.774476        Igfbp3
    ## 121 ENSMUSG00000020549 5_144415349     5 144.4153   7.258753         Elac2
    ## 122 ENSMUSG00000020609 5_144423461     5 144.4235   7.354533          Apob
    ## 123 ENSMUSG00000020649 5_146904986     5 146.9050   8.935532          Rrm2
    ## 124 ENSMUSG00000020811 5_144367577     5 144.3676   7.554287         Wscd1
    ## 125 ENSMUSG00000020832 5_147960714     5 147.9607   7.528956         Eral1
    ## 126 ENSMUSG00000020964 5_146691330     5 146.6913   8.128923         Sel1l
    ## 127 ENSMUSG00000021033 5_144395332     5 144.3953   8.492531         Gstz1
    ## 128 ENSMUSG00000021111 5_147286753     5 147.2868   8.387395        Papola
    ## 129 ENSMUSG00000021371 5_147253583     5 147.2536  11.743386         Mcur1
    ## 130 ENSMUSG00000021385 5_147272554     5 147.2726   7.791216          Ippk
    ## 131 ENSMUSG00000021390 5_144727745     5 144.7277   7.718384           Ogn
    ## 132 ENSMUSG00000001249 5_144185205     5 144.1852   7.499262           Hpn
    ## 133 ENSMUSG00000021610 5_146696722     5 146.6967   8.572637       Clptm1l
    ## 134 ENSMUSG00000021947 5_144617027     5 144.6170  10.592799         Cryl1
    ## 135 ENSMUSG00000021999 5_146727146     5 146.7271   9.934271          Cpb2
    ## 136 ENSMUSG00000022201 5_144375504     5 144.3755   7.320657           Zfr
    ## 137 ENSMUSG00000022211 5_144423461     5 144.4235   8.236619       Lrrc16b
    ## 138 ENSMUSG00000022769 5_146803810     5 146.8038   8.802891        Sdf2l1
    ## 139 ENSMUSG00000022962 5_146691330     5 146.6913   7.624082          Gart
    ## 140 ENSMUSG00000023074 5_144705848     5 144.7058   7.622915        Mospd1
    ## 141 ENSMUSG00000023150 5_147287312     5 147.2873   7.437818      Ivns1abp
    ## 142 ENSMUSG00000023387 5_144617027     5 144.6170   9.769732        Kcnk16
    ## 143 ENSMUSG00000001542 5_144617027     5 144.6170   7.893952          Ell2
    ## 144 ENSMUSG00000001786 5_144617027     5 144.6170   8.398375         Fbxo7
    ## 145 ENSMUSG00000024150 5_146727146     5 146.7271   7.619496         Mcfd2
    ## 146 ENSMUSG00000024386 5_146904986     5 146.9050   9.398566          Proc
    ## 147 ENSMUSG00000025140 5_146688269     5 146.6883   7.377507         Pycr1
    ## 148 ENSMUSG00000026207 5_147278599     5 147.2786   8.457335          Speg
    ## 149 ENSMUSG00000026692 5_147064187     5 147.0642   9.761132          Fmo4
    ## 150 ENSMUSG00000026959 5_144375504     5 144.3755   7.634801         Grin1
    ## 151 ENSMUSG00000027030 5_146696722     5 146.6967   7.792112         Stk39
    ## 152 ENSMUSG00000027111 5_144051716     5 144.0517   7.821916         Itga6
    ## 153 ENSMUSG00000027173 5_147221814     5 147.2218   8.728719        Depdc7
    ## 154 ENSMUSG00000027386 5_147275576     5 147.2756  14.026827         Fbln7
    ## 155 ENSMUSG00000027499 5_147275576     5 147.2756   9.043104          Pkia
    ## 156 ENSMUSG00000027525 5_147253583     5 147.2536   9.828070       Phactr3
    ## 157 ENSMUSG00000027577 5_144617027     5 144.6170   7.693758        Chrna4
    ## 158 ENSMUSG00000027651 5_145729503     5 145.7295   7.190356        Rprd1b
    ## 159 ENSMUSG00000027690 5_144727745     5 144.7277   7.404044        Slc2a2
    ## 160 ENSMUSG00000027834 5_144367577     5 144.3676   7.697699      Serpini1
    ## 161 ENSMUSG00000002778 5_146710838     5 146.7108   7.404865        Kdelr1
    ## 162 ENSMUSG00000028044 5_147286194     5 147.2862   9.617151         Cks1b
    ## 163 ENSMUSG00000028172 5_144617027     5 144.6170  10.736109         Tacr3
    ## 164 ENSMUSG00000028444 5_144407236     5 144.4072   7.663459         Cntfr
    ## 165 ENSMUSG00000028753 5_144407236     5 144.4072   8.439876        Vwa5b1
    ## 166 ENSMUSG00000029001 5_147074911     5 147.0749   7.370698        Fbxo44
    ## 167 ENSMUSG00000029073 5_147137425     5 147.1374   7.361337        Gltpd1
    ## 168 ENSMUSG00000029291 5_145729503     5 145.7295   7.383410         Rufy3
    ## 169 ENSMUSG00000029335 5_144407236     5 144.4072   7.813955          Bmp3
    ## 170 ENSMUSG00000029513 5_144437408     5 144.4374   8.809791        Prkab1
    ## 171 ENSMUSG00000029534 5_146696722     5 146.6967   7.356098           St7
    ## 172 ENSMUSG00000029617 5_144348163     5 144.3482  11.773271          Ccz1
    ## 173 ENSMUSG00000029621 5_145106694     5 145.1067  42.572439        Arpc1a
    ## 174 ENSMUSG00000029624 5_146696722     5 146.6967   9.056399         Ptcd1
    ## 175 ENSMUSG00000029625 5_145224788     5 145.2248  70.495562         Cpsf4
    ## 176 ENSMUSG00000029634 5_145916546     5 145.9165  34.920782          Rnf6
    ## 177 ENSMUSG00000029635 5_144437408     5 144.4374  15.379182          Cdk8
    ## 178 ENSMUSG00000029642 5_147178504     5 147.1785  75.216682        Polr1d
    ## 179 ENSMUSG00000029644 5_147137425     5 147.1374  15.085382          Pdx1
    ## 180 ENSMUSG00000029647 5_147383773     5 147.3838  30.743414          Pan3
    ## 181 ENSMUSG00000029650 5_147914702     5 147.9147  32.430087       Slc46a3
    ## 182 ENSMUSG00000029761 5_144587968     5 144.5880   8.280596         Cald1
    ## 183 ENSMUSG00000029802 5_144365599     5 144.3656   7.322109         Abcg2
    ## 184 ENSMUSG00000029859 5_147327447     5 147.3274   9.455112         Epha1
    ## 185 ENSMUSG00000003379 5_147221814     5 147.2218  15.743862         Cd79a
    ## 186 ENSMUSG00000003731 5_146904986     5 146.9050   7.840695         Kpna6
    ## 187 ENSMUSG00000003746 5_144423461     5 144.4235   7.235668         Man1a
    ## 188 ENSMUSG00000029999 5_146699418     5 146.6994  10.970663          Tgfa
    ## 189 ENSMUSG00000030088 5_144727745     5 144.7277  10.961672       Aldh1l1
    ## 190 ENSMUSG00000030094 5_144449040     5 144.4490   7.259672           Xpc
    ## 191 ENSMUSG00000030189 5_146889753     5 146.8898   7.732855          Ybx3
    ## 192 ENSMUSG00000030282 5_146694026     5 146.6940   7.278547          Cmas
    ## 193 ENSMUSG00000030350 5_147535843     5 147.5358   8.820113         Prmt8
    ## 194 ENSMUSG00000030849 5_144558267     5 144.5583   8.777407         Fgfr2
    ## 195 ENSMUSG00000030970 5_147221814     5 147.2218   7.295517         Ctbp2
    ## 196 ENSMUSG00000031066 5_147253583     5 147.2536   8.303372         Usp11
    ## 197 ENSMUSG00000031255 5_145224788     5 145.2248   8.492608         Sytl4
    ## 198 ENSMUSG00000031271 5_147221814     5 147.2218  10.490251      Serpina7
    ## 199 ENSMUSG00000031543 5_147278599     5 147.2786   9.507512          Ank1
    ## 200 ENSMUSG00000031570 5_147221814     5 147.2218   8.501210      Ppapdc1b
    ## 201 ENSMUSG00000031594 5_147383773     5 147.3838   7.289363          Fgl1
    ## 202 ENSMUSG00000031770 5_147278599     5 147.2786   8.619167       Herpud1
    ## 203 ENSMUSG00000031845 5_147221814     5 147.2218   8.182011         Bcmo1
    ## 204 ENSMUSG00000031934 5_144558267     5 144.5583   7.191923         Panx1
    ## 205 ENSMUSG00000004110 5_147272554     5 147.2726   7.711139       Cacna1e
    ## 206 ENSMUSG00000004317 5_147278599     5 147.2786   9.452286         Clcn5
    ## 207 ENSMUSG00000032356 5_147069549     5 147.0695  10.641763       Rasgrf1
    ## 208 ENSMUSG00000032392 5_147253583     5 147.2536  11.297259        Parp16
    ## 209 ENSMUSG00000032477 5_147137425     5 147.1374  10.412520        Cdc25a
    ## 210 ENSMUSG00000032788 5_147253583     5 147.2536   7.344954          Pdxk
    ## 211 ENSMUSG00000032890 5_144617027     5 144.6170   8.768680         Rims3
    ## 212 ENSMUSG00000033453 5_144365599     5 144.3656   9.278663      Adamts15
    ## 213 ENSMUSG00000033578 5_147278599     5 147.2786  10.382306        Tmem35
    ## 214 ENSMUSG00000033722 5_147284644     5 147.2846   8.621518      BC034090
    ## 215 ENSMUSG00000034115 5_147535843     5 147.5358   7.355213        Scn11a
    ## 216 ENSMUSG00000034235 5_144367577     5 144.3676   7.848889         Usp54
    ## 217 ENSMUSG00000034910 5_144348163     5 144.3482   8.797877         Pygo1
    ## 218 ENSMUSG00000035165 5_146677056     5 146.6771   7.188194         Kcne3
    ## 219 ENSMUSG00000035413 5_147986316     5 147.9863   8.545728        Tmem98
    ## 220 ENSMUSG00000035596 5_147278599     5 147.2786   8.861248        Mboat7
    ## 221 ENSMUSG00000035614 5_147221814     5 147.2218   9.034829       Fam179b
    ## 222 ENSMUSG00000035666 5_144437408     5 144.4374   9.426411        Gtf3c4
    ## 223 ENSMUSG00000036198 5_147089309     5 147.0893  12.977377      Arhgap36
    ## 224 ENSMUSG00000036655 5_144367577     5 144.3676   7.377243       Colec11
    ## 225 ENSMUSG00000037341 5_144375504     5 144.3755   7.251872        Slc9a7
    ## 226 ENSMUSG00000037492 5_147221814     5 147.2218   7.780117         Zmat4
    ##     chr      start        end strand     middle nearest.marker.id
    ## 1    13  49.341549  49.387025      1  49.364287       13_49373360
    ## 2     6 124.493113 124.510643      1 124.501878       6_124490637
    ## 3     5 145.140362 145.148078      1 145.144220       5_145106694
    ## 4     5 144.876495 144.965847     -1 144.921171       5_144916914
    ## 5     1 180.330485 180.424802      1 180.377644       1_180384864
    ## 6     2 180.456245 180.474867      1 180.465556       2_180495183
    ## 7     5 144.100436 144.188204      1 144.144320       5_144128706
    ## 8    13  36.867178  37.050244     -1  36.958711       13_36938062
    ## 9    14  21.052574  21.448568      1  21.250571       14_21228637
    ## 10    9  50.575273  50.581837      1  50.578555        9_50557512
    ## 11    6  45.060061  47.301369      1  46.180715        6_46238899
    ## 12    3  18.071950  18.243338     -1  18.157644        3_18158455
    ## 13   10  18.390322  18.533892      1  18.462107       10_18522964
    ## 14    X  16.709282  16.817366     -1  16.763324        X_16753166
    ## 15    9  49.436963  49.486225     -1  49.461594        9_49494008
    ## 16    3 108.455694 108.536536     -1 108.496115       3_108474452
    ## 17   13  20.090507  20.608353      1  20.349430       13_20434845
    ## 18    4 138.775735 138.782046      1 138.778890       4_138750051
    ## 19    5 146.832890 146.837032      1 146.834961       5_146862885
    ## 20    1 133.910304 133.921414     -1 133.915859       1_134039931
    ## 21    1 134.199223 134.235431     -1 134.217327       1_134214194
    ## 22    7  68.737045  68.749241     -1  68.743143        7_68751017
    ## 23    5 144.735915 144.761649     -1 144.748782       5_144727745
    ## 24   11  20.227037  20.249429     -1  20.238233       11_20122632
    ## 25   11  94.235956  94.246005     -1  94.240980       11_94235885
    ## 26   10  36.506814  36.834397      1  36.670606       10_36674858
    ## 27   17  34.931253  34.935953      1  34.933603       17_34961498
    ## 28    8  70.083529  70.090862      1  70.087196        8_70169904
    ## 29    8  70.493158  70.504081      1  70.498620        8_70489966
    ## 30    5 142.990893 143.113020     -1 143.051956       5_143073701
    ## 31    5 144.768092 144.859778      1 144.813935       5_144897381
    ## 32   14  66.911170  66.920054      1  66.915612       14_66935076
    ## 33   12  44.852488  45.074493     -1  44.963490       12_44964892
    ## 34    9  98.422961  98.446551      1  98.434756        9_98421908
    ## 35   10  37.139558  37.141679      1  37.140618       10_37074650
    ## 36    4  40.472180  40.477168      1  40.474674        4_40526031
    ## 37    5 143.249697 143.270022     -1 143.259860       5_143240265
    ## 38   10  84.526305  84.534036     -1  84.530170       10_84532456
    ## 39   18  84.011628  84.086404     -1  84.049016       18_84056074
    ## 40   10  43.579169  43.584262      1  43.581716       10_43558464
    ## 41   18  15.451924  15.760157     -1  15.606040       18_15638256
    ## 42    6  13.834458  13.839942     -1  13.837200        6_13829457
    ## 43   19  38.836579  38.919923      1  38.878251       19_38864633
    ## 44    5 143.548706 143.564525      1 143.556616       5_143554489
    ## 45   16  18.620319  18.622403     -1  18.621361       16_18603149
    ## 46    4 134.003330 134.018841     -1 134.011086       4_133904386
    ## 47    X  48.667979  48.674478     -1  48.671228        X_48669439
    ## 48    4  57.908483  57.916297     -1  57.912390        4_57896973
    ## 49    5 144.190286 144.194441      1 144.192364       5_144185205
    ## 50    6  70.765720  70.792232     -1  70.778976        6_70761345
    ## 51    9 106.368632 106.375724      1 106.372178       9_106387870
    ## 52   18  60.685978  60.713389     -1  60.699684       18_60604486
    ## 53   15  83.299740  83.350247     -1  83.324994       15_83225104
    ## 54   18  62.180126  62.184405      1  62.182266       18_62183580
    ## 55    7  56.716465  57.387188     -1  57.051826        7_56851082
    ## 56   10  23.920356  23.921469      1  23.920912       10_23920500
    ## 57    5 147.408982 147.409359      1 147.409170       5_147406811
    ## 58    6  90.844978  90.845460     -1  90.845219        6_90840253
    ## 59    6 128.530918 128.531304     -1 128.531111       6_128530829
    ## 60    5 144.545887 144.557478      1 144.551682       5_144558267
    ## 61   12  59.094055  59.094537     -1  59.094296       12_59095072
    ## 62    3  96.666515  96.666982     -1  96.666748        3_96673140
    ## 63    6 117.168535 117.181367      1 117.174951       6_117238405
    ## 64   18  82.523539  82.524021     -1  82.523780       18_82520261
    ## 65   12 103.616675 103.631444     -1 103.624060      12_103663553
    ## 66    4  34.353980  34.354854      1  34.354417        4_34320123
    ## 67   14  31.700015  31.830415     -1  31.765215       14_31741983
    ## 68    6  21.985916  22.256404      1  22.121160        6_22125632
    ## 69    6 113.716119 113.719880     -1 113.718000       6_113764552
    ## 70    3  38.107440  38.107922     -1  38.107681        3_38196744
    ## 71    4  62.360727  62.391769      1  62.376248        4_62386280
    ## 72    7 126.299434 126.299916     -1 126.299675       7_126318055
    ## 73    9  88.390807  88.391351      1  88.391079        9_88417508
    ## 74    5 144.194442 144.223615     -1 144.209028       5_144185205
    ## 75    X  47.900345  47.900827      1  47.900586        X_47903725
    ## 76   15  68.309819  68.310301      1  68.310060       15_68313152
    ## 77    5 145.345279 145.390512      1 145.367896       5_145342881
    ## 78    5 145.283343 145.291469      1 145.287406       5_145342881
    ## 79    X 164.088830 164.118860      1 164.103845       X_164101533
    ## 80   12 100.187968 100.193540      1 100.190754      12_100194506
    ## 81    4  34.949074  35.157484     -1  35.053279        4_35061296
    ## 82    5 143.933035 143.985719      1 143.959377       5_143944917
    ## 83    1 192.897307 193.035698     -1 192.966502       1_192958624
    ## 84    5 146.948657 146.955614      1 146.952136       5_146904986
    ## 85    5 146.951573 146.963800     -1 146.957686       5_146904986
    ## 86    5 147.016655 147.076586     -1 147.046620       5_147064187
    ## 87   13  21.953053  21.953628     -1  21.953340       13_21948789
    ## 88    2 165.078693 165.090750     -1 165.084722       2_165053217
    ## 89    5 146.248551 146.249422      1 146.248986       5_146240909
    ## 90    X  88.959217  88.959670      1  88.959444        X_88853790
    ## 91    4  45.550017  45.550499      1  45.550258        4_45610838
    ## 92    2  49.680499  49.680976      1  49.680738        2_49650224
    ## 93    X 151.215135 151.215687     -1 151.215411       X_151227466
    ## 94    2  67.707290  67.707771     -1  67.707530        2_67700861
    ## 95    5 147.100562 147.101053     -1 147.100808       5_147089309
    ## 96    2  24.985984  24.986466      1  24.986225        2_24966197
    ## 97   11 117.523603 117.532205      1 117.527904      11_117527080
    ## 98    5 147.949882 147.957166     -1 147.953524       5_147953121
    ## 99    X 153.037629 153.072915      1 153.055272       X_152944135
    ## 100   3  94.998800  95.004869     -1  95.001834        3_94941131
    ## 101   5 144.277061 144.280513      1 144.278787       5_144348163
    ## 102  17  13.862317  13.862803     -1  13.862560       17_13894862
    ## 103  10  86.839094  86.874954      1  86.857024       10_86869387
    ## 104   2 167.642608 167.689822     -1 167.666215       2_167647654
    ## 105   7  25.228687  25.230257      1  25.229472        7_25227932
    ## 106   9  21.907491  21.918192     -1  21.912842        9_21878582
    ## 107   5 147.228596 147.271521     -1 147.250058       5_147253583
    ## 108   5 143.390730 143.403748     -1 143.397239       5_143409462
    ## 109   X 108.664004 108.664891     -1 108.664448       X_108537425
    ## 110   5 130.208080 130.208562     -1 130.208321       5_130238343
    ## 111  13  12.378158  12.378640      1  12.378399       13_12540218
    ## 112  17  29.308138  29.308620      1  29.308379       17_29296322
    ## 113  12 105.453856 105.493095      1 105.473476      12_105473437
    ## 114  14   5.253907   5.260580     -1   5.257244        14_5246221
    ## 115   5 146.182450 146.185304     -1 146.183877       5_146240909
    ## 116   5  88.739247  88.739729     -1  88.739488        5_88735139
    ## 117  13 103.397109 103.397591      1 103.397350      13_103361771
    ## 118   5 147.280720 147.281263      1 147.280992       5_147281622
    ## 119  11  16.752203  16.918158      1  16.835180       11_16805656
    ## 120  11   7.206086   7.213923     -1   7.210004        11_7143572
    ## 121  11  64.979038  65.002069      1  64.990554       11_64989953
    ## 122  12   7.977648   8.016835      1   7.997242        12_7959520
    ## 123  12  24.708241  24.714146      1  24.711194       12_24634906
    ## 124  11  71.749920  71.789647      1  71.769784       11_71782692
    ## 125  11  78.073376  78.080383     -1  78.076880       11_78062799
    ## 126  12  91.806043  91.849157     -1  91.827600       12_91812028
    ## 127  12  87.147707  87.164721      1  87.156214       12_87149702
    ## 128  12 105.784694 105.838944      1 105.811819      12_105822211
    ## 129  13  43.538412  43.560191     -1  43.549302       13_43569860
    ## 130  13  49.421311  49.462990      1  49.442150       13_49505606
    ## 131  13  49.608071  49.624500      1  49.616286       13_49637853
    ## 132   7  31.098725  31.115290     -1  31.107008        7_31090237
    ## 133  13  73.604002  73.620639      1  73.612320       13_73587073
    ## 134  14  57.275034  57.398483     -1  57.336758       14_57329302
    ## 135  14  75.242287  75.283551      1  75.262919       14_75278220
    ## 136  15  12.117831  12.185683      1  12.151757       15_12118336
    ## 137  14  55.491093  55.508246      1  55.499670       14_55489547
    ## 138  16  17.130138  17.132383     -1  17.131260       16_17168727
    ## 139  16  91.621186  91.646952     -1  91.634069       16_91619448
    ## 140   X  53.344598  53.370502     -1  53.357550        X_53358219
    ## 141   1 151.344498 151.364445      1 151.354472       1_151323553
    ## 142  14  20.262756  20.269162     -1  20.265959       14_20256918
    ## 143  13  75.707484  75.772356      1  75.739920       13_75694789
    ## 144  10  86.021972  86.051873      1  86.036922       10_86013381
    ## 145  17  87.254658  87.265935     -1  87.260296       17_87267756
    ## 146  18  32.123129  32.139570     -1  32.131350       18_32101915
    ## 147  11 120.635712 120.643769     -1 120.639740      11_120645657
    ## 148   1  75.375297  75.432320      1  75.403808        1_75420221
    ## 149   1 162.793883 162.813972     -1 162.803928       1_162805534
    ## 150   2  25.291181  25.319187     -1  25.305184        2_25350212
    ## 151   2  68.210445  68.472268     -1  68.341356        2_68346231
    ## 152   2  71.745616  71.858416      1  71.802016        2_71876558
    ## 153   2 104.721784 104.742878     -1 104.732331       2_104752033
    ## 154   2 128.862981 128.897034      1 128.880008       2_128880153
    ## 155   3   7.366604   7.445366      1   7.405985         3_7530799
    ## 156   2 178.118975 178.338492      1 178.228734       2_178227821
    ## 157   2 181.018380 181.043546     -1 181.030963       2_181081025
    ## 158   2 158.028497 158.132297      1 158.080397       2_157983121
    ## 159   3  28.697903  28.728356      1  28.713130        3_28688021
    ## 160   3  75.557547  75.643495      1  75.600521        3_75497151
    ## 161   7  45.872772  45.883731      1  45.878252        7_46000542
    ## 162   3  89.415472  89.418383     -1  89.416928        3_89393239
    ## 163   3 134.829007 134.934581      1 134.881794       3_134947378
    ## 164   4  41.657498  41.697089     -1  41.677294        4_41681004
    ## 165   4 138.565360 138.635884     -1 138.600622       4_138608123
    ## 166   4 148.152800 148.160482     -1 148.156641       4_148156547
    ## 167   4 155.864723 155.869440     -1 155.867082       4_155866226
    ## 168   5  88.583574  88.642671      1  88.613122        5_88657556
    ## 169   5  98.854439  98.880960      1  98.867700        5_98872372
    ## 170   5 116.013586 116.024508     -1 116.019047       5_115858970
    ## 171   6  17.692933  17.943025      1  17.817979        6_17806817
    ## 172   5 143.987916 144.014853     -1 144.001384       5_144013221
    ## 173   5 145.083830 145.108761      1 145.096296       5_145106694
    ## 174   5 145.147514 145.167108     -1 145.157311       5_145106694
    ## 175   5 145.167213 145.182041      1 145.174627       5_145224788
    ## 176   5 146.209192 146.221555     -1 146.215374       5_146240909
    ## 177   5 146.231230 146.302874      1 146.267052       5_146283615
    ## 178   5 147.077050 147.111361      1 147.094206       5_147089309
    ## 179   5 147.269959 147.278008      1 147.273984       5_147272554
    ## 180   5 147.430161 147.548502      1 147.489332       5_147535843
    ## 181   5 147.878437 147.894815     -1 147.886626       5_147884560
    ## 182   6  34.598500  34.775473      1  34.686986        6_34683828
    ## 183   6  58.584523  58.692869      1  58.638696        6_58488862
    ## 184   6  42.358487  42.373268     -1  42.365878        6_42364697
    ## 185   7  24.897381  24.902197      1  24.899789        7_24880009
    ## 186   4 129.643980 129.672767     -1 129.658374       4_129694131
    ## 187  10  53.904788  54.075796     -1  53.990292       10_53983416
    ## 188   6  86.195251  86.275449      1  86.235350        6_86247465
    ## 189   6  90.486427  90.600203      1  90.543315        6_90548751
    ## 190   6  91.489305  91.515884     -1  91.502594        6_91511406
    ## 191   6 131.364858 131.388450     -1 131.376654       6_131417347
    ## 192   6 142.756686 142.775714      1 142.766200       6_142770993
    ## 193   6 127.689011 127.769472     -1 127.729242       6_127722321
    ## 194   7 130.162451 133.123350     -1 131.642900       7_131677149
    ## 195   7 132.987563 133.124354     -1 133.055958       7_133014427
    ## 196   X  20.703906  20.720539      1  20.712222        X_20715226
    ## 197   X 133.936385 133.981812     -1 133.959098       X_133928183
    ## 198   X 139.079257 139.085236     -1 139.082246       X_139082524
    ## 199   8  22.974844  23.150497      1  23.062670        8_23030262
    ## 200   8  25.720061  25.724887      1  25.722474        8_25694654
    ## 201   8  41.191434  41.215156     -1  41.203295        8_41203352
    ## 202   8  94.386438  94.395377      1  94.390908        8_94397839
    ## 203   8 117.095854 117.133720      1 117.114787       8_117113822
    ## 204   9  15.005161  15.045478     -1  15.025320        9_15054736
    ## 205   1 154.392519 154.725920     -1 154.559220       1_154569760
    ## 206   X   7.153810   7.319358     -1   7.236584         X_7250880
    ## 207   9  89.909775  90.026976      1  89.968376        9_89976570
    ## 208   9  65.214690  65.239219      1  65.226954        9_65155658
    ## 209   9 109.875579 109.893888      1 109.884734       9_109867448
    ## 210  10  78.436744  78.464975     -1  78.450860       10_78441604
    ## 211   4 120.854816 120.896579      1 120.875698       4_120849899
    ## 212   9  30.899155  30.922452     -1  30.910804        9_30902691
    ## 213   X 134.295225 134.305969      1 134.300597       X_134296564
    ## 214   1 155.212474 155.232724     -1 155.222599       1_155218414
    ## 215   9 119.753763 119.825456     -1 119.789610       9_119738766
    ## 216  14  20.548912  20.618354     -1  20.583633       14_20642601
    ## 217   9  72.925645  72.952181      1  72.938913        9_72950347
    ## 218   7 100.176670 100.184869      1 100.180770       7_100187613
    ## 219  11  80.810175  80.822033      1  80.816104       11_80802203
    ## 220   7   3.677790   3.693523     -1   3.685656         7_3586798
    ## 221  12  64.965742  65.022573      1  64.994158       12_64985175
    ## 222   2  28.822299  28.840360     -1  28.831330        2_28803574
    ## 223   X  49.463945  49.500244      1  49.482094        X_49481750
    ## 224  12  28.594173  28.623290     -1  28.608732       12_28609576
    ## 225   X  20.105755  20.291807     -1  20.198781        X_20217307
    ## 226   8  23.636019  24.156585      1  23.896302        8_23852535
    ##                  biotype        module hotspot
    ## 1         protein_coding           red    chr5
    ## 2         protein_coding         brown    chr5
    ## 3         protein_coding        salmon    <NA>
    ## 4         protein_coding         black    <NA>
    ## 5         protein_coding     steelblue    chr5
    ## 6         protein_coding           red    chr5
    ## 7         protein_coding  midnightblue    <NA>
    ## 8         protein_coding         brown    chr5
    ## 9         protein_coding         brown    chr5
    ## 10        protein_coding     turquoise    chr5
    ## 11        protein_coding          grey    chr5
    ## 12        protein_coding           tan    chr5
    ## 13        protein_coding   greenyellow    chr5
    ## 14        protein_coding     turquoise    chr5
    ## 15        protein_coding         brown    chr5
    ## 16        protein_coding        yellow    chr5
    ## 17        protein_coding       skyblue    chr5
    ## 18        protein_coding          grey    chr5
    ## 19        protein_coding          grey    <NA>
    ## 20        protein_coding          cyan    chr5
    ## 21        protein_coding     steelblue    chr5
    ## 22        protein_coding       magenta    chr5
    ## 23        protein_coding          grey    <NA>
    ## 24        protein_coding        orange    chr5
    ## 25        protein_coding          grey    chr5
    ## 26        protein_coding          grey    chr5
    ## 27        protein_coding          grey    chr5
    ## 28        protein_coding        yellow    chr5
    ## 29        protein_coding          blue    chr5
    ## 30        protein_coding darkturquoise    <NA>
    ## 31        protein_coding        purple    <NA>
    ## 32        protein_coding          grey    chr5
    ## 33        protein_coding         brown    chr5
    ## 34        protein_coding          grey    chr5
    ## 35        protein_coding        yellow    chr5
    ## 36        protein_coding    lightgreen    chr5
    ## 37        protein_coding        purple    <NA>
    ## 38        protein_coding        yellow    chr5
    ## 39        protein_coding        purple    chr5
    ## 40        protein_coding          grey    chr5
    ## 41        protein_coding          grey    chr5
    ## 42        protein_coding       skyblue    chr5
    ## 43        protein_coding          pink    chr5
    ## 44        protein_coding    lightgreen    <NA>
    ## 45        protein_coding        yellow    chr5
    ## 46        protein_coding        yellow    chr5
    ## 47        protein_coding     turquoise    chr5
    ## 48        protein_coding           tan    chr5
    ## 49        protein_coding          grey    <NA>
    ## 50        protein_coding           red    chr5
    ## 51        protein_coding     lightcyan    chr5
    ## 52        protein_coding        yellow    chr5
    ## 53        protein_coding   darkmagenta    chr5
    ## 54        protein_coding          grey    chr5
    ## 55        protein_coding   yellowgreen    chr5
    ## 56        protein_coding       skyblue    chr5
    ## 57            pseudogene          grey    <NA>
    ## 58            pseudogene          grey    chr5
    ## 59            pseudogene          grey    chr5
    ## 60        protein_coding   yellowgreen    <NA>
    ## 61            pseudogene          grey    chr5
    ## 62            pseudogene          grey    <NA>
    ## 63        protein_coding           red    chr5
    ## 64            pseudogene          blue    chr5
    ## 65        protein_coding     royalblue    chr5
    ## 66            pseudogene          grey    chr5
    ## 67        protein_coding        orange    chr5
    ## 68        protein_coding          grey    chr5
    ## 69        protein_coding          grey    chr5
    ## 70            pseudogene          grey    <NA>
    ## 71        protein_coding   darkmagenta    chr5
    ## 72            pseudogene          grey    chr5
    ## 73            pseudogene          grey    chr5
    ## 74        protein_coding        yellow    <NA>
    ## 75            pseudogene          grey    chr5
    ## 76            pseudogene          grey    chr5
    ## 77        protein_coding          grey    <NA>
    ## 78        protein_coding          blue    <NA>
    ## 79        protein_coding        salmon    chr5
    ## 80  processed_transcript         brown    chr5
    ## 81        protein_coding        yellow    chr5
    ## 82        protein_coding     turquoise    <NA>
    ## 83        protein_coding          pink    chr5
    ## 84        protein_coding        yellow    <NA>
    ## 85        protein_coding     turquoise    <NA>
    ## 86        protein_coding          blue    <NA>
    ## 87            pseudogene          grey    chr5
    ## 88        protein_coding     steelblue    chr5
    ## 89            pseudogene          grey    <NA>
    ## 90            pseudogene          grey    chr5
    ## 91            pseudogene          grey    chr5
    ## 92            pseudogene          grey    chr5
    ## 93            pseudogene          grey    chr5
    ## 94            pseudogene          grey    chr5
    ## 95            pseudogene         brown    <NA>
    ## 96            pseudogene          grey    chr5
    ## 97               lincRNA   greenyellow    chr5
    ## 98  processed_transcript          grey    <NA>
    ## 99  processed_transcript     turquoise    chr5
    ## 100            antisense       darkred    chr5
    ## 101            antisense          grey    <NA>
    ## 102           pseudogene          grey    chr5
    ## 103            antisense           red    chr5
    ## 104       protein_coding        yellow    chr5
    ## 105       protein_coding           red    chr5
    ## 106       protein_coding        yellow    chr5
    ## 107            antisense          grey    <NA>
    ## 108            antisense          grey    <NA>
    ## 109       protein_coding         brown    chr5
    ## 110           pseudogene          grey    <NA>
    ## 111           pseudogene          grey    chr5
    ## 112           pseudogene          grey    chr5
    ## 113       protein_coding        orange    chr5
    ## 114       protein_coding         plum1    chr5
    ## 115       protein_coding          grey    <NA>
    ## 116           pseudogene          grey    <NA>
    ## 117           pseudogene          grey    chr5
    ## 118              lincRNA        grey60    <NA>
    ## 119       protein_coding           red    chr5
    ## 120       protein_coding   lightyellow    chr5
    ## 121       protein_coding        purple    chr5
    ## 122       protein_coding         brown    chr5
    ## 123       protein_coding         brown    chr5
    ## 124       protein_coding          grey    chr5
    ## 125       protein_coding          grey    chr5
    ## 126       protein_coding        yellow    chr5
    ## 127       protein_coding         brown    chr5
    ## 128       protein_coding          grey    chr5
    ## 129       protein_coding       magenta    chr5
    ## 130       protein_coding           red    chr5
    ## 131       protein_coding          cyan    chr5
    ## 132       protein_coding      darkgrey    chr5
    ## 133       protein_coding   darkmagenta    chr5
    ## 134       protein_coding          grey    chr5
    ## 135       protein_coding          grey    chr5
    ## 136       protein_coding          pink    chr5
    ## 137       protein_coding  midnightblue    chr5
    ## 138       protein_coding   darkmagenta    chr5
    ## 139       protein_coding         white    chr5
    ## 140       protein_coding         brown    chr5
    ## 141       protein_coding          grey    chr5
    ## 142       protein_coding         brown    chr5
    ## 143       protein_coding         brown    chr5
    ## 144       protein_coding          grey    chr5
    ## 145       protein_coding   darkmagenta    chr5
    ## 146       protein_coding         brown    chr5
    ## 147       protein_coding         brown    chr5
    ## 148       protein_coding        purple    chr5
    ## 149       protein_coding          grey    chr5
    ## 150       protein_coding     steelblue    chr5
    ## 151       protein_coding           tan    chr5
    ## 152       protein_coding     royalblue   chr13
    ## 153       protein_coding         brown    chr5
    ## 154       protein_coding           red    chr5
    ## 155       protein_coding         brown    chr5
    ## 156       protein_coding        orange    chr5
    ## 157       protein_coding           red    chr5
    ## 158       protein_coding darkturquoise    chr5
    ## 159       protein_coding          pink    chr5
    ## 160       protein_coding         brown    chr5
    ## 161       protein_coding        yellow    chr5
    ## 162       protein_coding   darkmagenta    chr5
    ## 163       protein_coding          grey    chr5
    ## 164       protein_coding        purple    chr5
    ## 165       protein_coding     steelblue    chr5
    ## 166       protein_coding        purple    chr5
    ## 167       protein_coding         black    chr5
    ## 168       protein_coding          pink    <NA>
    ## 169       protein_coding          grey    <NA>
    ## 170       protein_coding           red    <NA>
    ## 171       protein_coding         brown    chr5
    ## 172       protein_coding         brown    <NA>
    ## 173       protein_coding          grey    <NA>
    ## 174       protein_coding          grey    <NA>
    ## 175       protein_coding          grey    <NA>
    ## 176       protein_coding         green    <NA>
    ## 177       protein_coding        orange    <NA>
    ## 178       protein_coding       sienna3    <NA>
    ## 179       protein_coding        grey60    <NA>
    ## 180       protein_coding          pink    <NA>
    ## 181       protein_coding         brown    <NA>
    ## 182       protein_coding         brown    chr5
    ## 183       protein_coding          grey    chr5
    ## 184       protein_coding          grey    chr5
    ## 185       protein_coding     turquoise    chr5
    ## 186       protein_coding          pink    chr5
    ## 187       protein_coding          grey    chr5
    ## 188       protein_coding          grey    chr5
    ## 189       protein_coding       magenta    chr5
    ## 190       protein_coding         white    chr5
    ## 191       protein_coding        yellow    chr5
    ## 192       protein_coding   darkmagenta    chr5
    ## 193       protein_coding          grey    chr5
    ## 194       protein_coding     steelblue    chr5
    ## 195       protein_coding           red    chr5
    ## 196       protein_coding   greenyellow    chr5
    ## 197       protein_coding         white    chr5
    ## 198       protein_coding         brown    chr5
    ## 199       protein_coding     steelblue    chr5
    ## 200       protein_coding         brown    chr5
    ## 201       protein_coding     turquoise    chr5
    ## 202       protein_coding       skyblue    chr5
    ## 203       protein_coding          grey    chr5
    ## 204       protein_coding     turquoise    chr5
    ## 205       protein_coding   yellowgreen    chr5
    ## 206       protein_coding          grey    chr5
    ## 207       protein_coding     steelblue    chr5
    ## 208       protein_coding        yellow    chr5
    ## 209       protein_coding   darkmagenta    chr5
    ## 210       protein_coding        purple    chr5
    ## 211       protein_coding     steelblue    chr5
    ## 212       protein_coding          grey    chr5
    ## 213       protein_coding           red    chr5
    ## 214       protein_coding          grey    chr5
    ## 215       protein_coding   yellowgreen    chr5
    ## 216       protein_coding          pink    chr5
    ## 217       protein_coding          grey    chr5
    ## 218       protein_coding   lightyellow    chr5
    ## 219       protein_coding        yellow    chr5
    ## 220       protein_coding     steelblue    chr5
    ## 221       protein_coding           red    chr5
    ## 222       protein_coding       darkred    chr5
    ## 223       protein_coding          grey    chr7
    ## 224       protein_coding          cyan    chr5
    ## 225       protein_coding          pink    chr5
    ## 226       protein_coding          grey    chr5

The above yields 226 expression traits that map to the hotspot on Chr 5.
The list includes both local and nonlocal traits. We’ll create two
distinct objects, one for locals and one for nonlocal traits.

``` r
local_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == 5, pos <= c5hot_upper, pos >= c5hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>%
  mutate(local_tx = chr ==5) %>%
  filter(local_tx)
```

``` r
dim(local_annot)
```

    ## [1] 42 16

``` r
nonlocal_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == 5, pos <= c5hot_upper, pos >= c5hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>% #genomewide threshold
  mutate(local_tx = chr == 5) %>%
  filter(!local_tx) %>% 
  filter(!is.na(hotspot))
```

``` r
dim(nonlocal_annot)
```

    ## [1] 182  16

We’ll use only those nonlocal traits that Keller et al. annotated as
mapping to a hotspot.

This gives us 182 traits for Chr 5 hotspot, which is what Keller et
al. reported.

We also want to look at the local traits for Chr5 hotspot. We have 42
local traits. We want to identify a subset of those for two-dimensional
scans. Let’s first limit to those near the hotspot, ie, within 5Mb of
either end.

``` r
local_annot2 <- local_annot %>%
  filter(middle <= c5hot_upper + 5, middle >= c5hot_lower - 5) %>%
  arrange(desc(lod))
```

Recall that *Pdx1* is the gene that Keller et al. identified as the key
mediator for Chr 5 hotspot. Its LOD is only 15, which seems to indicate
that I should be looking at a broad set of local traits, not just those,
eg., with LOD \> 40\!

Also, I’m limiting consideration of local genes to those with middle
within 5Mb of the hotspot.

Now, I make the two expression trait matrices that I need for use with
Condor.

``` r
Chr5hot_local <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% local_annot2$annot.id]
Chr5hot_nonlocal <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% nonlocal_annot$annot.id]
saveRDS(Chr5hot_local, "../data/Chr5hot_local.rds")
saveRDS(Chr5hot_nonlocal, "../data/Chr5hot_nonlocal.rds")
saveRDS(K$`5`, "../data/Chr5_kinship.rds")
saveRDS(genoprobs$`5`, "../data/Chr5_aprobs.rds")
```

Lastly, we need to identify the scan region for Chr 5.

We consider the physical map.

``` r
(start_index <- which(map$`5`== map$`5`[map$`5` > c5hot_lower][1]) - 50)
```

    ## 5_144013221 
    ##        3498

``` r
(stop_index <- which(map$`5`== map$`5`[map$`5` > c5hot_upper][1]) + 50)
```

    ## 5_148006053 
    ##        3766

``` r
(nsnp <- stop_index - start_index + 1)
```

    ## 5_148006053 
    ##         269

Lastly, we copy the covariates rds file from the earlier
analysis.

``` bash
cp ~/Box\ Sync/attie/keller2018-chr2-hotspot-chtc/data/addcovar.rds ../data/.
```

## References

Keller et al. 2018. Genetics. Genetic drivers of pancreatic islet
function.

## Session info

``` r
devtools::session_info()
```

    ## ─ Session info ──────────────────────────────────────────────────────────
    ##  setting  value                       
    ##  version  R version 3.5.3 (2019-03-11)
    ##  os       macOS Mojave 10.14.5        
    ##  system   x86_64, darwin15.6.0        
    ##  ui       X11                         
    ##  language (EN)                        
    ##  collate  en_US.UTF-8                 
    ##  ctype    en_US.UTF-8                 
    ##  tz       America/Chicago             
    ##  date     2019-05-31                  
    ## 
    ## ─ Packages ──────────────────────────────────────────────────────────────
    ##  package     * version date       lib source                        
    ##  assertthat    0.2.1   2019-03-21 [1] CRAN (R 3.5.2)                
    ##  backports     1.1.4   2019-04-10 [1] CRAN (R 3.5.2)                
    ##  bit           1.1-14  2018-05-29 [1] CRAN (R 3.5.0)                
    ##  bit64         0.9-7   2017-05-08 [1] CRAN (R 3.5.0)                
    ##  blob          1.1.1   2018-03-25 [1] CRAN (R 3.5.0)                
    ##  broom         0.5.1   2018-12-05 [1] CRAN (R 3.5.0)                
    ##  callr         3.2.0   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  cellranger    1.1.0   2016-07-27 [1] CRAN (R 3.5.0)                
    ##  cli           1.1.0   2019-03-19 [1] CRAN (R 3.5.3)                
    ##  colorspace    1.4-1   2019-03-18 [1] CRAN (R 3.5.3)                
    ##  crayon        1.3.4   2017-09-16 [1] CRAN (R 3.5.0)                
    ##  data.table    1.12.0  2019-01-13 [1] CRAN (R 3.5.2)                
    ##  DBI           1.0.0   2018-05-02 [1] CRAN (R 3.5.0)                
    ##  desc          1.2.0   2018-05-01 [1] CRAN (R 3.5.0)                
    ##  devtools      2.0.2   2019-04-08 [1] CRAN (R 3.5.2)                
    ##  digest        0.6.19  2019-05-20 [1] CRAN (R 3.5.2)                
    ##  dplyr       * 0.8.0.1 2019-02-15 [1] CRAN (R 3.5.2)                
    ##  evaluate      0.14    2019-05-28 [1] CRAN (R 3.5.3)                
    ##  forcats     * 0.4.0   2019-02-17 [1] CRAN (R 3.5.2)                
    ##  fs            1.3.1   2019-05-06 [1] CRAN (R 3.5.2)                
    ##  generics      0.0.2   2018-11-29 [1] CRAN (R 3.5.0)                
    ##  ggplot2     * 3.1.1   2019-04-07 [1] CRAN (R 3.5.3)                
    ##  glue          1.3.1   2019-03-12 [1] CRAN (R 3.5.2)                
    ##  gtable        0.3.0   2019-03-25 [1] CRAN (R 3.5.2)                
    ##  haven         2.1.0   2019-02-19 [1] CRAN (R 3.5.2)                
    ##  hms           0.4.2   2018-03-10 [1] CRAN (R 3.5.0)                
    ##  htmltools     0.3.6   2017-04-28 [1] CRAN (R 3.5.0)                
    ##  httr          1.4.0   2018-12-11 [1] CRAN (R 3.5.0)                
    ##  jsonlite      1.6     2018-12-07 [1] CRAN (R 3.5.0)                
    ##  knitr         1.23    2019-05-18 [1] CRAN (R 3.5.2)                
    ##  lattice       0.20-38 2018-11-04 [1] CRAN (R 3.5.3)                
    ##  lazyeval      0.2.2   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  lubridate     1.7.4   2018-04-11 [1] CRAN (R 3.5.0)                
    ##  magrittr      1.5     2014-11-22 [1] CRAN (R 3.5.0)                
    ##  memoise       1.1.0   2017-04-21 [1] CRAN (R 3.5.0)                
    ##  modelr        0.1.4   2019-02-18 [1] CRAN (R 3.5.2)                
    ##  munsell       0.5.0   2018-06-12 [1] CRAN (R 3.5.0)                
    ##  nlme          3.1-137 2018-04-07 [1] CRAN (R 3.5.3)                
    ##  pillar        1.3.1   2018-12-15 [1] CRAN (R 3.5.0)                
    ##  pkgbuild      1.0.3   2019-03-20 [1] CRAN (R 3.5.2)                
    ##  pkgconfig     2.0.2   2018-08-16 [1] CRAN (R 3.5.0)                
    ##  pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.5.0)                
    ##  plyr          1.8.4   2016-06-08 [1] CRAN (R 3.5.0)                
    ##  prettyunits   1.0.2   2015-07-13 [1] CRAN (R 3.5.0)                
    ##  processx      3.3.1   2019-05-08 [1] CRAN (R 3.5.2)                
    ##  ps            1.3.0   2018-12-21 [1] CRAN (R 3.5.0)                
    ##  purrr       * 0.3.2   2019-03-15 [1] CRAN (R 3.5.3)                
    ##  qtl2        * 0.18    2019-02-08 [1] local                         
    ##  R6            2.4.0   2019-02-14 [1] CRAN (R 3.5.2)                
    ##  Rcpp          1.0.1.3 2019-05-29 [1] Github (RcppCore/Rcpp@6062d56)
    ##  readr       * 1.3.1   2018-12-21 [1] CRAN (R 3.5.0)                
    ##  readxl        1.3.0   2019-02-15 [1] CRAN (R 3.5.2)                
    ##  remotes       2.0.4   2019-04-10 [1] CRAN (R 3.5.2)                
    ##  rlang         0.3.4   2019-04-07 [1] CRAN (R 3.5.3)                
    ##  rmarkdown     1.13    2019-05-22 [1] CRAN (R 3.5.2)                
    ##  rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.5.0)                
    ##  RSQLite       2.1.1   2018-05-06 [1] CRAN (R 3.5.0)                
    ##  rstudioapi    0.10    2019-03-19 [1] CRAN (R 3.5.2)                
    ##  rvest         0.3.2   2016-06-17 [1] CRAN (R 3.5.0)                
    ##  scales        1.0.0   2018-08-09 [1] CRAN (R 3.5.0)                
    ##  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.5.0)                
    ##  stringi       1.4.3   2019-03-12 [1] CRAN (R 3.5.2)                
    ##  stringr     * 1.4.0   2019-02-10 [1] CRAN (R 3.5.2)                
    ##  testthat      2.1.1   2019-04-23 [1] CRAN (R 3.5.3)                
    ##  tibble      * 2.1.1   2019-03-16 [1] CRAN (R 3.5.3)                
    ##  tidyr       * 0.8.3   2019-03-01 [1] CRAN (R 3.5.2)                
    ##  tidyselect    0.2.5   2018-10-11 [1] CRAN (R 3.5.0)                
    ##  tidyverse   * 1.2.1   2017-11-14 [1] CRAN (R 3.5.0)                
    ##  usethis       1.5.0   2019-04-07 [1] CRAN (R 3.5.2)                
    ##  withr         2.1.2   2018-03-15 [1] CRAN (R 3.5.0)                
    ##  xfun          0.7     2019-05-14 [1] CRAN (R 3.5.2)                
    ##  xml2          1.2.0   2018-01-24 [1] CRAN (R 3.5.0)                
    ##  yaml          2.2.0   2018-07-25 [1] CRAN (R 3.5.0)                
    ## 
    ## [1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library
